<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4.1 Benchmark ML Methods for Prediction | Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning</title>
  <meta name="description" content="This is a book-style code sharing." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="4.1 Benchmark ML Methods for Prediction | Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a book-style code sharing." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4.1 Benchmark ML Methods for Prediction | Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning" />
  
  <meta name="twitter:description" content="This is a book-style code sharing." />
  

<meta name="author" content="Peng Jin, Qiqi Jin, Kankan Wang" />


<meta name="date" content="2025-07-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="figure-3.html"/>
<link rel="next" href="predict-class-labels-on-the-test-set-and-plot-roc.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Background</a></li>
<li class="chapter" data-level="2" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>2</b> Figure 1</a>
<ul>
<li class="chapter" data-level="2.1" data-path="process-the-data-from-functional_precision_medicine_tumor_board_aml.html"><a href="process-the-data-from-functional_precision_medicine_tumor_board_aml.html"><i class="fa fa-check"></i><b>2.1</b> Process the data from Functional_Precision_Medicine_Tumor_Board_AML</a></li>
<li class="chapter" data-level="2.2" data-path="functional_precision_medicine_tumor_board_aml.html"><a href="functional_precision_medicine_tumor_board_aml.html"><i class="fa fa-check"></i><b>2.2</b> Functional_Precision_Medicine_Tumor_Board_AML</a></li>
<li class="chapter" data-level="2.3" data-path="boxplot-for-drug-sensitivity-between-resistant-and-sensitive.html"><a href="boxplot-for-drug-sensitivity-between-resistant-and-sensitive.html"><i class="fa fa-check"></i><b>2.3</b> Boxplot for drug sensitivity between resistant and sensitive</a></li>
<li class="chapter" data-level="2.4" data-path="de-analysis-functional_medicine.html"><a href="de-analysis-functional_medicine.html"><i class="fa fa-check"></i><b>2.4</b> DE analysis Functional_medicine</a></li>
<li class="chapter" data-level="2.5" data-path="valcanoplot-plot-ex-vivo.html"><a href="valcanoplot-plot-ex-vivo.html"><i class="fa fa-check"></i><b>2.5</b> Valcanoplot plot (Ex vivo)</a></li>
<li class="chapter" data-level="2.6" data-path="heatmap-for-representative-de-genes.html"><a href="heatmap-for-representative-de-genes.html"><i class="fa fa-check"></i><b>2.6</b> Heatmap for representative DE genes</a></li>
<li class="chapter" data-level="2.7" data-path="cibersortx-results-visualization-ex-vivo.html"><a href="cibersortx-results-visualization-ex-vivo.html"><i class="fa fa-check"></i><b>2.7</b> CIBERSORTx Results Visualization (ex vivo)</a></li>
<li class="chapter" data-level="2.8" data-path="cibersortx-results-visualization-in-vivo.html"><a href="cibersortx-results-visualization-in-vivo.html"><i class="fa fa-check"></i><b>2.8</b> CIBERSORTx Results Visualization (in vivo)</a></li>
<li class="chapter" data-level="2.9" data-path="ssgsea-analysis-in-vivo.html"><a href="ssgsea-analysis-in-vivo.html"><i class="fa fa-check"></i><b>2.9</b> ssGSEA analysis in vivo</a></li>
<li class="chapter" data-level="2.10" data-path="go-enrichment-dot-plot.html"><a href="go-enrichment-dot-plot.html"><i class="fa fa-check"></i><b>2.10</b> GO Enrichment Dot Plot</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>3</b> Figure 2</a>
<ul>
<li class="chapter" data-level="3.1" data-path="de-analysis-between-in-vivo-sensitive-and-resistant.html"><a href="de-analysis-between-in-vivo-sensitive-and-resistant.html"><i class="fa fa-check"></i><b>3.1</b> DE analysis between in vivo sensitive and resistant</a></li>
<li class="chapter" data-level="3.2" data-path="de-analysis-rjaml-azaven-cohort.html"><a href="de-analysis-rjaml-azaven-cohort.html"><i class="fa fa-check"></i><b>3.2</b> DE analysis RJAML AZA+VEN cohort</a></li>
<li class="chapter" data-level="3.3" data-path="valcanoplot-plot-ex-vivo-1.html"><a href="valcanoplot-plot-ex-vivo-1.html"><i class="fa fa-check"></i><b>3.3</b> Valcanoplot plot (Ex vivo)</a></li>
<li class="chapter" data-level="3.4" data-path="valcanoplot-plot-in-vivo.html"><a href="valcanoplot-plot-in-vivo.html"><i class="fa fa-check"></i><b>3.4</b> Valcanoplot plot (In vivo)</a></li>
<li class="chapter" data-level="3.5" data-path="venn-plot.html"><a href="venn-plot.html"><i class="fa fa-check"></i><b>3.5</b> Venn plot</a></li>
<li class="chapter" data-level="3.6" data-path="integration-of-degs-ex-vivo-and-in-vivo.html"><a href="integration-of-degs-ex-vivo-and-in-vivo.html"><i class="fa fa-check"></i><b>3.6</b> Integration of DEGs ex vivo and in vivo</a></li>
<li class="chapter" data-level="3.7" data-path="gsea-analysis-for-crispri-drug-sensitivity.html"><a href="gsea-analysis-for-crispri-drug-sensitivity.html"><i class="fa fa-check"></i><b>3.7</b> GSEA analysis for CRISPRi drug sensitivity</a></li>
<li class="chapter" data-level="3.8" data-path="differential-expression-analysis-for-shpink1.html"><a href="differential-expression-analysis-for-shpink1.html"><i class="fa fa-check"></i><b>3.8</b> Differential Expression Analysis for shPINK1</a></li>
<li class="chapter" data-level="3.9" data-path="de-analysis-in-cell-lines.html"><a href="de-analysis-in-cell-lines.html"><i class="fa fa-check"></i><b>3.9</b> DE analysis in cell lines</a></li>
<li class="chapter" data-level="3.10" data-path="valcano-plots-molm13.html"><a href="valcano-plots-molm13.html"><i class="fa fa-check"></i><b>3.10</b> Valcano plots (MOLM13)</a></li>
<li class="chapter" data-level="3.11" data-path="valcano-plots-oa3.html"><a href="valcano-plots-oa3.html"><i class="fa fa-check"></i><b>3.11</b> Valcano plots (OA3)</a></li>
<li class="chapter" data-level="3.12" data-path="gsea-analysis-molm13.html"><a href="gsea-analysis-molm13.html"><i class="fa fa-check"></i><b>3.12</b> GSEA analysis (MOLM13)</a></li>
<li class="chapter" data-level="3.13" data-path="gsea-analysis-oa3.html"><a href="gsea-analysis-oa3.html"><i class="fa fa-check"></i><b>3.13</b> GSEA analysis (OA3)</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>4</b> Figure 3</a>
<ul>
<li class="chapter" data-level="4.1" data-path="benchmark-ml-methods-for-prediction.html"><a href="benchmark-ml-methods-for-prediction.html"><i class="fa fa-check"></i><b>4.1</b> Benchmark ML Methods for Prediction</a></li>
<li class="chapter" data-level="4.2" data-path="predict-class-labels-on-the-test-set-and-plot-roc.html"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html"><i class="fa fa-check"></i><b>4.2</b> Predict Class Labels on the Test Set and Plot ROC</a></li>
<li class="chapter" data-level="4.3" data-path="auc-and-auprc-heatmap-for-the-entire-and-test-sets.html"><a href="auc-and-auprc-heatmap-for-the-entire-and-test-sets.html"><i class="fa fa-check"></i><b>4.3</b> AUC and AUPRC Heatmap for the Entire and Test Sets</a></li>
<li class="chapter" data-level="4.4" data-path="model-performance-plot.html"><a href="model-performance-plot.html"><i class="fa fa-check"></i><b>4.4</b> Model Performance Plot</a></li>
<li class="chapter" data-level="4.5" data-path="feature-importance.html"><a href="feature-importance.html"><i class="fa fa-check"></i><b>4.5</b> Feature importance</a></li>
<li class="chapter" data-level="4.6" data-path="heatmap-for-confusion-matrices-defaut-cutoff-0.html"><a href="heatmap-for-confusion-matrices-defaut-cutoff-0.html"><i class="fa fa-check"></i><b>4.6</b> Heatmap for Confusion Matrices (defaut cutoff 0.5)</a></li>
<li class="chapter" data-level="4.7" data-path="barplot-for-predicted-response-0.html"><a href="barplot-for-predicted-response-0.html"><i class="fa fa-check"></i><b>4.7</b> Barplot for predicted Response (0.5)</a></li>
<li class="chapter" data-level="4.8" data-path="heatmap-for-confusion-matrices-optimal-cutoff-0.html"><a href="heatmap-for-confusion-matrices-optimal-cutoff-0.html"><i class="fa fa-check"></i><b>4.8</b> Heatmap for Confusion Matrices (optimal cutoff 0.630)</a></li>
<li class="chapter" data-level="4.9" data-path="barplot-for-predicted-response-0.html"><a href="barplot-for-predicted-response-0.html#barplot-for-predicted-response-0.630"><i class="fa fa-check"></i><b>4.9</b> Barplot for predicted Response (0.630)</a></li>
<li class="chapter" data-level="4.10" data-path="density-plot-for-response-probability.html"><a href="density-plot-for-response-probability.html"><i class="fa fa-check"></i><b>4.10</b> Density Plot for Response Probability</a></li>
<li class="chapter" data-level="4.11" data-path="violin-and-boxplots-for-response-probability-rf8.html"><a href="violin-and-boxplots-for-response-probability-rf8.html"><i class="fa fa-check"></i><b>4.11</b> Violin and Boxplots for Response Probability (RF8)</a></li>
<li class="chapter" data-level="4.12" data-path="barplot-for-response.html"><a href="barplot-for-response.html"><i class="fa fa-check"></i><b>4.12</b> Barplot for Response</a></li>
<li class="chapter" data-level="4.13" data-path="rf8-for-mdr-status.html"><a href="rf8-for-mdr-status.html"><i class="fa fa-check"></i><b>4.13</b> RF8 for MDR status</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>5</b> Figure 4</a>
<ul>
<li class="chapter" data-level="5.1" data-path="validation-of-rf8-model-rjaml-cohort-2.html"><a href="validation-of-rf8-model-rjaml-cohort-2.html"><i class="fa fa-check"></i><b>5.1</b> Validation of RF8 model (RJAML-Cohort 2)</a></li>
<li class="chapter" data-level="5.2" data-path="validation-of-rf8-model-fpmtb.html"><a href="validation-of-rf8-model-fpmtb.html"><i class="fa fa-check"></i><b>5.2</b> Validation of RF8 model (FPMTB)</a></li>
<li class="chapter" data-level="5.3" data-path="validation-of-rf8-model-beataml.html"><a href="validation-of-rf8-model-beataml.html"><i class="fa fa-check"></i><b>5.3</b> Validation of RF8 model (BeatAML)</a></li>
<li class="chapter" data-level="5.4" data-path="violin-and-boxplots-for-response-probability-rf8-1.html"><a href="violin-and-boxplots-for-response-probability-rf8-1.html"><i class="fa fa-check"></i><b>5.4</b> Violin and Boxplots for Response Probability (RF8)</a></li>
<li class="chapter" data-level="5.5" data-path="rf8-score-among-clusters.html"><a href="rf8-score-among-clusters.html"><i class="fa fa-check"></i><b>5.5</b> RF8 score among clusters</a></li>
<li class="chapter" data-level="5.6" data-path="integration-of-ex-vivo-drug-response-in-beataml.html"><a href="integration-of-ex-vivo-drug-response-in-beataml.html"><i class="fa fa-check"></i><b>5.6</b> Integration of ex vivo drug-response in BeatAML</a></li>
<li class="chapter" data-level="5.7" data-path="beataml-waves-1-4-heatmap-based-on-drug-profiling.html"><a href="beataml-waves-1-4-heatmap-based-on-drug-profiling.html"><i class="fa fa-check"></i><b>5.7</b> BeatAML Waves 1-4 heatmap based on drug profiling</a></li>
<li class="chapter" data-level="5.8" data-path="boxplot-for-drug-sensitivitity-between-resistant-and-sensitive.html"><a href="boxplot-for-drug-sensitivitity-between-resistant-and-sensitive.html"><i class="fa fa-check"></i><b>5.8</b> Boxplot for drug sensitivitity between resistant and sensitive</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>6</b> Figure 5</a>
<ul>
<li class="chapter" data-level="6.1" data-path="integration-with-aml-genomics.html"><a href="integration-with-aml-genomics.html"><i class="fa fa-check"></i><b>6.1</b> Integration with AML genomics</a></li>
<li class="chapter" data-level="6.2" data-path="mutation-pairs.html"><a href="mutation-pairs.html"><i class="fa fa-check"></i><b>6.2</b> Mutation pairs</a></li>
<li class="chapter" data-level="6.3" data-path="complexheatmap-for-mutation.html"><a href="complexheatmap-for-mutation.html"><i class="fa fa-check"></i><b>6.3</b> Complexheatmap for mutation</a></li>
<li class="chapter" data-level="6.4" data-path="fisher-test.html"><a href="fisher-test.html"><i class="fa fa-check"></i><b>6.4</b> Fisher test</a></li>
<li class="chapter" data-level="6.5" data-path="barplot-in-rjaml-cohort.html"><a href="barplot-in-rjaml-cohort.html"><i class="fa fa-check"></i><b>6.5</b> Barplot in RJAML cohort</a></li>
<li class="chapter" data-level="6.6" data-path="table-one-for-multi-cohorts.html"><a href="table-one-for-multi-cohorts.html"><i class="fa fa-check"></i><b>6.6</b> Table one for multi-cohorts</a></li>
<li class="chapter" data-level="6.7" data-path="barplot-for-real-response.html"><a href="barplot-for-real-response.html"><i class="fa fa-check"></i><b>6.7</b> Barplot for real response</a></li>
<li class="chapter" data-level="6.8" data-path="multivariable-logisticreg.html"><a href="multivariable-logisticreg.html"><i class="fa fa-check"></i><b>6.8</b> Multivariable LogisticReg</a></li>
<li class="chapter" data-level="6.9" data-path="identification-of-rf8-correlated-genes.html"><a href="identification-of-rf8-correlated-genes.html"><i class="fa fa-check"></i><b>6.9</b> Identification of RF8-correlated genes</a></li>
<li class="chapter" data-level="6.10" data-path="differential-expression-analysis-for-mut-vs.-wt.html"><a href="differential-expression-analysis-for-mut-vs.-wt.html"><i class="fa fa-check"></i><b>6.10</b> Differential Expression Analysis for Mut vs. WT</a></li>
<li class="chapter" data-level="6.11" data-path="venn-diagram-for-target-genes.html"><a href="venn-diagram-for-target-genes.html"><i class="fa fa-check"></i><b>6.11</b> Venn Diagram for target genes</a></li>
<li class="chapter" data-level="6.12" data-path="pathway-overlap-analysis-of-genes.html"><a href="pathway-overlap-analysis-of-genes.html"><i class="fa fa-check"></i><b>6.12</b> Pathway overlap analysis of genes</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="figure-6.html"><a href="figure-6.html"><i class="fa fa-check"></i><b>7</b> Figure 6</a>
<ul>
<li class="chapter" data-level="7.1" data-path="relationship-between-rf8-score-vs.-venaza-response-probability.html"><a href="relationship-between-rf8-score-vs.-venaza-response-probability.html"><i class="fa fa-check"></i><b>7.1</b> Relationship between RF8 score vs. VEN/AZA response probability</a></li>
<li class="chapter" data-level="7.2" data-path="monotonic-response-prediction-plot.html"><a href="monotonic-response-prediction-plot.html"><i class="fa fa-check"></i><b>7.2</b> Monotonic Response Prediction Plot</a></li>
<li class="chapter" data-level="7.3" data-path="survival-plot.html"><a href="survival-plot.html"><i class="fa fa-check"></i><b>7.3</b> Survival plot</a></li>
<li class="chapter" data-level="7.4" data-path="brier-score-analysis.html"><a href="brier-score-analysis.html"><i class="fa fa-check"></i><b>7.4</b> Brier Score Analysis</a></li>
<li class="chapter" data-level="7.5" data-path="c-index-comparison.html"><a href="c-index-comparison.html"><i class="fa fa-check"></i><b>7.5</b> C-index Comparison</a></li>
<li class="chapter" data-level="7.6" data-path="km-plots-for-rjaml-cohort-2.html"><a href="km-plots-for-rjaml-cohort-2.html"><i class="fa fa-check"></i><b>7.6</b> KM-Plots for RJAML Cohort 2</a></li>
<li class="chapter" data-level="7.7" data-path="barplot-for-crcri.html"><a href="barplot-for-crcri.html"><i class="fa fa-check"></i><b>7.7</b> Barplot for CR/CRi</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="benchmark-ml-methods-for-prediction" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Benchmark ML Methods for Prediction<a href="benchmark-ml-methods-for-prediction.html#benchmark-ml-methods-for-prediction" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="benchmark-ml-methods-for-prediction.html#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mlr3)</span>
<span id="cb67-2"><a href="benchmark-ml-methods-for-prediction.html#cb67-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mlr3learners)</span>
<span id="cb67-3"><a href="benchmark-ml-methods-for-prediction.html#cb67-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mlr3verse)</span>
<span id="cb67-4"><a href="benchmark-ml-methods-for-prediction.html#cb67-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mlr3tuning)</span>
<span id="cb67-5"><a href="benchmark-ml-methods-for-prediction.html#cb67-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mlr3measures)</span>
<span id="cb67-6"><a href="benchmark-ml-methods-for-prediction.html#cb67-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb67-7"><a href="benchmark-ml-methods-for-prediction.html#cb67-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mlr3benchmark)</span>
<span id="cb67-8"><a href="benchmark-ml-methods-for-prediction.html#cb67-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Boruta)</span>
<span id="cb67-9"><a href="benchmark-ml-methods-for-prediction.html#cb67-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(caret)</span>
<span id="cb67-10"><a href="benchmark-ml-methods-for-prediction.html#cb67-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(randomForest)</span>
<span id="cb67-11"><a href="benchmark-ml-methods-for-prediction.html#cb67-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pROC)</span>
<span id="cb67-12"><a href="benchmark-ml-methods-for-prediction.html#cb67-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(PRROC)</span>
<span id="cb67-13"><a href="benchmark-ml-methods-for-prediction.html#cb67-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb67-14"><a href="benchmark-ml-methods-for-prediction.html#cb67-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-15"><a href="benchmark-ml-methods-for-prediction.html#cb67-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Set seed for reproducibility</span></span>
<span id="cb67-16"><a href="benchmark-ml-methods-for-prediction.html#cb67-16" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123456</span>)</span>
<span id="cb67-17"><a href="benchmark-ml-methods-for-prediction.html#cb67-17" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb67-18"><a href="benchmark-ml-methods-for-prediction.html#cb67-18" aria-hidden="true" tabindex="-1"></a><span class="co"># ################################################</span></span>
<span id="cb67-19"><a href="benchmark-ml-methods-for-prediction.html#cb67-19" aria-hidden="true" tabindex="-1"></a><span class="co"># ## Load Dataset and Prepare Classification Task</span></span>
<span id="cb67-20"><a href="benchmark-ml-methods-for-prediction.html#cb67-20" aria-hidden="true" tabindex="-1"></a><span class="co"># ################################################</span></span>
<span id="cb67-21"><a href="benchmark-ml-methods-for-prediction.html#cb67-21" aria-hidden="true" tabindex="-1"></a><span class="co"># # Read gene expression data</span></span>
<span id="cb67-22"><a href="benchmark-ml-methods-for-prediction.html#cb67-22" aria-hidden="true" tabindex="-1"></a><span class="co"># data &lt;- readxl::read_excel(&quot;../02.data/TrainingTest_RJAML_Cohort1_84_genes.xlsx&quot;)</span></span>
<span id="cb67-23"><a href="benchmark-ml-methods-for-prediction.html#cb67-23" aria-hidden="true" tabindex="-1"></a><span class="co"># data &lt;- data[, -1]  # Remove the first column if unnecessary (PatientID column)</span></span>
<span id="cb67-24"><a href="benchmark-ml-methods-for-prediction.html#cb67-24" aria-hidden="true" tabindex="-1"></a><span class="co"># data$Response &lt;- as.factor(data$Response)  # Convert &#39;Response&#39; to factor for classification</span></span>
<span id="cb67-25"><a href="benchmark-ml-methods-for-prediction.html#cb67-25" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb67-26"><a href="benchmark-ml-methods-for-prediction.html#cb67-26" aria-hidden="true" tabindex="-1"></a><span class="co"># # Create classification task for mlr3</span></span>
<span id="cb67-27"><a href="benchmark-ml-methods-for-prediction.html#cb67-27" aria-hidden="true" tabindex="-1"></a><span class="co"># task &lt;- TaskClassif$new(id = &quot;AML&quot;, backend = data, target = &quot;Response&quot;)</span></span>
<span id="cb67-28"><a href="benchmark-ml-methods-for-prediction.html#cb67-28" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb67-29"><a href="benchmark-ml-methods-for-prediction.html#cb67-29" aria-hidden="true" tabindex="-1"></a><span class="co"># ################################################</span></span>
<span id="cb67-30"><a href="benchmark-ml-methods-for-prediction.html#cb67-30" aria-hidden="true" tabindex="-1"></a><span class="co"># ## Define Learners (Models) and Resampling Strategy</span></span>
<span id="cb67-31"><a href="benchmark-ml-methods-for-prediction.html#cb67-31" aria-hidden="true" tabindex="-1"></a><span class="co"># ################################################</span></span>
<span id="cb67-32"><a href="benchmark-ml-methods-for-prediction.html#cb67-32" aria-hidden="true" tabindex="-1"></a><span class="co"># # Define a list of learners (classification models)</span></span>
<span id="cb67-33"><a href="benchmark-ml-methods-for-prediction.html#cb67-33" aria-hidden="true" tabindex="-1"></a><span class="co"># learners &lt;- list(</span></span>
<span id="cb67-34"><a href="benchmark-ml-methods-for-prediction.html#cb67-34" aria-hidden="true" tabindex="-1"></a><span class="co">#   lrn(&quot;classif.rpart&quot;, predict_type = &quot;prob&quot;),       # Decision Tree</span></span>
<span id="cb67-35"><a href="benchmark-ml-methods-for-prediction.html#cb67-35" aria-hidden="true" tabindex="-1"></a><span class="co">#   lrn(&quot;classif.ranger&quot;, predict_type = &quot;prob&quot;),      # Random Forest</span></span>
<span id="cb67-36"><a href="benchmark-ml-methods-for-prediction.html#cb67-36" aria-hidden="true" tabindex="-1"></a><span class="co">#   lrn(&quot;classif.xgboost&quot;, predict_type = &quot;prob&quot;),     # XGBoost</span></span>
<span id="cb67-37"><a href="benchmark-ml-methods-for-prediction.html#cb67-37" aria-hidden="true" tabindex="-1"></a><span class="co">#   lrn(&quot;classif.log_reg&quot;, predict_type = &quot;prob&quot;),     # Logistic Regression</span></span>
<span id="cb67-38"><a href="benchmark-ml-methods-for-prediction.html#cb67-38" aria-hidden="true" tabindex="-1"></a><span class="co">#   lrn(&quot;classif.svm&quot;, predict_type = &quot;prob&quot;),         # Support Vector Machine</span></span>
<span id="cb67-39"><a href="benchmark-ml-methods-for-prediction.html#cb67-39" aria-hidden="true" tabindex="-1"></a><span class="co">#   lrn(&quot;classif.nnet&quot;, predict_type = &quot;prob&quot;),        # Neural Network</span></span>
<span id="cb67-40"><a href="benchmark-ml-methods-for-prediction.html#cb67-40" aria-hidden="true" tabindex="-1"></a><span class="co">#   lrn(&quot;classif.kknn&quot;, predict_type = &quot;prob&quot;),        # K-Nearest Neighbors</span></span>
<span id="cb67-41"><a href="benchmark-ml-methods-for-prediction.html#cb67-41" aria-hidden="true" tabindex="-1"></a><span class="co">#   lrn(&quot;classif.naive_bayes&quot;, predict_type = &quot;prob&quot;), # Naive Bayes</span></span>
<span id="cb67-42"><a href="benchmark-ml-methods-for-prediction.html#cb67-42" aria-hidden="true" tabindex="-1"></a><span class="co">#   lrn(&quot;classif.glmnet&quot;, predict_type = &quot;prob&quot;),      # ElasticNet</span></span>
<span id="cb67-43"><a href="benchmark-ml-methods-for-prediction.html#cb67-43" aria-hidden="true" tabindex="-1"></a><span class="co">#   lrn(&quot;classif.cv_glmnet&quot;, predict_type = &quot;prob&quot;)    # Cross-validated ElasticNet</span></span>
<span id="cb67-44"><a href="benchmark-ml-methods-for-prediction.html#cb67-44" aria-hidden="true" tabindex="-1"></a><span class="co"># )</span></span>
<span id="cb67-45"><a href="benchmark-ml-methods-for-prediction.html#cb67-45" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb67-46"><a href="benchmark-ml-methods-for-prediction.html#cb67-46" aria-hidden="true" tabindex="-1"></a><span class="co"># # Define a 5-fold cross-validation resampling strategy repeated 100 times</span></span>
<span id="cb67-47"><a href="benchmark-ml-methods-for-prediction.html#cb67-47" aria-hidden="true" tabindex="-1"></a><span class="co"># resampling_repeated &lt;- rsmp(&quot;repeated_cv&quot;, folds = 5, repeats = 100)</span></span>
<span id="cb67-48"><a href="benchmark-ml-methods-for-prediction.html#cb67-48" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb67-49"><a href="benchmark-ml-methods-for-prediction.html#cb67-49" aria-hidden="true" tabindex="-1"></a><span class="co"># ################################################</span></span>
<span id="cb67-50"><a href="benchmark-ml-methods-for-prediction.html#cb67-50" aria-hidden="true" tabindex="-1"></a><span class="co"># ## Benchmarking Setup and Execution</span></span>
<span id="cb67-51"><a href="benchmark-ml-methods-for-prediction.html#cb67-51" aria-hidden="true" tabindex="-1"></a><span class="co"># ################################################</span></span>
<span id="cb67-52"><a href="benchmark-ml-methods-for-prediction.html#cb67-52" aria-hidden="true" tabindex="-1"></a><span class="co"># # Create benchmarking design grid</span></span>
<span id="cb67-53"><a href="benchmark-ml-methods-for-prediction.html#cb67-53" aria-hidden="true" tabindex="-1"></a><span class="co"># design &lt;- benchmark_grid(</span></span>
<span id="cb67-54"><a href="benchmark-ml-methods-for-prediction.html#cb67-54" aria-hidden="true" tabindex="-1"></a><span class="co">#   tasks = task,</span></span>
<span id="cb67-55"><a href="benchmark-ml-methods-for-prediction.html#cb67-55" aria-hidden="true" tabindex="-1"></a><span class="co">#   learners = learners,</span></span>
<span id="cb67-56"><a href="benchmark-ml-methods-for-prediction.html#cb67-56" aria-hidden="true" tabindex="-1"></a><span class="co">#   resamplings = resampling_repeated</span></span>
<span id="cb67-57"><a href="benchmark-ml-methods-for-prediction.html#cb67-57" aria-hidden="true" tabindex="-1"></a><span class="co"># )</span></span>
<span id="cb67-58"><a href="benchmark-ml-methods-for-prediction.html#cb67-58" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb67-59"><a href="benchmark-ml-methods-for-prediction.html#cb67-59" aria-hidden="true" tabindex="-1"></a><span class="co"># # Execute benchmark</span></span>
<span id="cb67-60"><a href="benchmark-ml-methods-for-prediction.html#cb67-60" aria-hidden="true" tabindex="-1"></a><span class="co"># bmr &lt;- benchmark(design)</span></span>
<span id="cb67-61"><a href="benchmark-ml-methods-for-prediction.html#cb67-61" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb67-62"><a href="benchmark-ml-methods-for-prediction.html#cb67-62" aria-hidden="true" tabindex="-1"></a><span class="co"># # Define performance measures</span></span>
<span id="cb67-63"><a href="benchmark-ml-methods-for-prediction.html#cb67-63" aria-hidden="true" tabindex="-1"></a><span class="co"># measures &lt;- list(msr(&quot;classif.auc&quot;), msr(&quot;classif.acc&quot;))</span></span>
<span id="cb67-64"><a href="benchmark-ml-methods-for-prediction.html#cb67-64" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb67-65"><a href="benchmark-ml-methods-for-prediction.html#cb67-65" aria-hidden="true" tabindex="-1"></a><span class="co"># # Aggregate performance results</span></span>
<span id="cb67-66"><a href="benchmark-ml-methods-for-prediction.html#cb67-66" aria-hidden="true" tabindex="-1"></a><span class="co"># mean_score &lt;- bmr$aggregate(measures)</span></span>
<span id="cb67-67"><a href="benchmark-ml-methods-for-prediction.html#cb67-67" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb67-68"><a href="benchmark-ml-methods-for-prediction.html#cb67-68" aria-hidden="true" tabindex="-1"></a><span class="co"># # Extract benchmark scores for each iteration</span></span>
<span id="cb67-69"><a href="benchmark-ml-methods-for-prediction.html#cb67-69" aria-hidden="true" tabindex="-1"></a><span class="co"># score &lt;- bmr$score(measures)</span></span>
<span id="cb67-70"><a href="benchmark-ml-methods-for-prediction.html#cb67-70" aria-hidden="true" tabindex="-1"></a><span class="co"># score &lt;- score[, c(&quot;learner_id&quot;, &quot;iteration&quot;, &quot;classif.auc&quot;, &quot;classif.acc&quot;)]</span></span>
<span id="cb67-71"><a href="benchmark-ml-methods-for-prediction.html#cb67-71" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb67-72"><a href="benchmark-ml-methods-for-prediction.html#cb67-72" aria-hidden="true" tabindex="-1"></a><span class="co"># # Calculate mean and median AUC</span></span>
<span id="cb67-73"><a href="benchmark-ml-methods-for-prediction.html#cb67-73" aria-hidden="true" tabindex="-1"></a><span class="co"># mean_auc &lt;- mean(score$classif.auc, na.rm = TRUE)</span></span>
<span id="cb67-74"><a href="benchmark-ml-methods-for-prediction.html#cb67-74" aria-hidden="true" tabindex="-1"></a><span class="co"># median_auc &lt;- median(score$classif.auc, na.rm = TRUE)</span></span>
<span id="cb67-75"><a href="benchmark-ml-methods-for-prediction.html#cb67-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-76"><a href="benchmark-ml-methods-for-prediction.html#cb67-76" aria-hidden="true" tabindex="-1"></a><span class="co"># Save benchmark results to a CSV file</span></span>
<span id="cb67-77"><a href="benchmark-ml-methods-for-prediction.html#cb67-77" aria-hidden="true" tabindex="-1"></a><span class="co"># write.csv(score, &quot;../03.out/figure3/ML_Results_GeneExp.csv&quot;)</span></span>
<span id="cb67-78"><a href="benchmark-ml-methods-for-prediction.html#cb67-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-79"><a href="benchmark-ml-methods-for-prediction.html#cb67-79" aria-hidden="true" tabindex="-1"></a><span class="do">################################################</span></span>
<span id="cb67-80"><a href="benchmark-ml-methods-for-prediction.html#cb67-80" aria-hidden="true" tabindex="-1"></a><span class="do">## Boruta Feature Selection</span></span>
<span id="cb67-81"><a href="benchmark-ml-methods-for-prediction.html#cb67-81" aria-hidden="true" tabindex="-1"></a><span class="do">################################################</span></span>
<span id="cb67-82"><a href="benchmark-ml-methods-for-prediction.html#cb67-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-83"><a href="benchmark-ml-methods-for-prediction.html#cb67-83" aria-hidden="true" tabindex="-1"></a><span class="co"># Reload dataset and prepare for feature selection</span></span>
<span id="cb67-84"><a href="benchmark-ml-methods-for-prediction.html#cb67-84" aria-hidden="true" tabindex="-1"></a><span class="co"># data &lt;- read_excel(&quot;../02.data/TrainingTest_RJAML_Cohort1_84_genes.xlsx&quot;)</span></span>
<span id="cb67-85"><a href="benchmark-ml-methods-for-prediction.html#cb67-85" aria-hidden="true" tabindex="-1"></a><span class="co"># data &lt;- data[, -1]  # Remove the first column if unnecessary</span></span>
<span id="cb67-86"><a href="benchmark-ml-methods-for-prediction.html#cb67-86" aria-hidden="true" tabindex="-1"></a><span class="co"># data$Response &lt;- as.factor(data$Response)  # Convert &#39;Response&#39; to factor</span></span>
<span id="cb67-87"><a href="benchmark-ml-methods-for-prediction.html#cb67-87" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb67-88"><a href="benchmark-ml-methods-for-prediction.html#cb67-88" aria-hidden="true" tabindex="-1"></a><span class="co"># # Perform Boruta feature selection</span></span>
<span id="cb67-89"><a href="benchmark-ml-methods-for-prediction.html#cb67-89" aria-hidden="true" tabindex="-1"></a><span class="co"># set.seed(123)</span></span>
<span id="cb67-90"><a href="benchmark-ml-methods-for-prediction.html#cb67-90" aria-hidden="true" tabindex="-1"></a><span class="co"># boruta_result &lt;- Boruta(Response ~ ., data = data, pValue = 0.01, maxRuns = 1000, doTrace = 2)</span></span>
<span id="cb67-91"><a href="benchmark-ml-methods-for-prediction.html#cb67-91" aria-hidden="true" tabindex="-1"></a><span class="co"># boruta_selected &lt;- TentativeRoughFix(boruta_result)</span></span>
<span id="cb67-92"><a href="benchmark-ml-methods-for-prediction.html#cb67-92" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb67-93"><a href="benchmark-ml-methods-for-prediction.html#cb67-93" aria-hidden="true" tabindex="-1"></a><span class="co"># # Extract selected features (excluding tentative features)</span></span>
<span id="cb67-94"><a href="benchmark-ml-methods-for-prediction.html#cb67-94" aria-hidden="true" tabindex="-1"></a><span class="co"># selected_features &lt;- getSelectedAttributes(boruta_selected, withTentative = FALSE)</span></span>
<span id="cb67-95"><a href="benchmark-ml-methods-for-prediction.html#cb67-95" aria-hidden="true" tabindex="-1"></a><span class="co"># selected_features</span></span>
<span id="cb67-96"><a href="benchmark-ml-methods-for-prediction.html#cb67-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-97"><a href="benchmark-ml-methods-for-prediction.html#cb67-97" aria-hidden="true" tabindex="-1"></a><span class="do">#######################################################</span></span>
<span id="cb67-98"><a href="benchmark-ml-methods-for-prediction.html#cb67-98" aria-hidden="true" tabindex="-1"></a><span class="do">## Train a Random Forest Model with Selected Features</span></span>
<span id="cb67-99"><a href="benchmark-ml-methods-for-prediction.html#cb67-99" aria-hidden="true" tabindex="-1"></a><span class="do">#######################################################</span></span>
<span id="cb67-100"><a href="benchmark-ml-methods-for-prediction.html#cb67-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-101"><a href="benchmark-ml-methods-for-prediction.html#cb67-101" aria-hidden="true" tabindex="-1"></a><span class="co"># Reload dataset and prepare with selected features</span></span>
<span id="cb67-102"><a href="benchmark-ml-methods-for-prediction.html#cb67-102" aria-hidden="true" tabindex="-1"></a><span class="co"># data &lt;- read_excel(&quot;../02.data/TrainingTest_RJAML_Cohort1_84_genes.xlsx&quot;)</span></span>
<span id="cb67-103"><a href="benchmark-ml-methods-for-prediction.html#cb67-103" aria-hidden="true" tabindex="-1"></a><span class="co"># data &lt;- data[, -1]  # Remove PatientID column</span></span>
<span id="cb67-104"><a href="benchmark-ml-methods-for-prediction.html#cb67-104" aria-hidden="true" tabindex="-1"></a><span class="co"># data_selected &lt;- data[, c(&quot;Response&quot;, selected_features)]  # Select important features</span></span>
<span id="cb67-105"><a href="benchmark-ml-methods-for-prediction.html#cb67-105" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb67-106"><a href="benchmark-ml-methods-for-prediction.html#cb67-106" aria-hidden="true" tabindex="-1"></a><span class="co"># # Log-transform feature values (excluding the Response column)</span></span>
<span id="cb67-107"><a href="benchmark-ml-methods-for-prediction.html#cb67-107" aria-hidden="true" tabindex="-1"></a><span class="co"># data_selected[, -1] &lt;- log2(data_selected[, -1] + 1)</span></span>
<span id="cb67-108"><a href="benchmark-ml-methods-for-prediction.html#cb67-108" aria-hidden="true" tabindex="-1"></a><span class="co"># data_selected$Response &lt;- as.factor(data_selected$Response)  # Ensure &#39;Response&#39; is a factor</span></span>
<span id="cb67-109"><a href="benchmark-ml-methods-for-prediction.html#cb67-109" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb67-110"><a href="benchmark-ml-methods-for-prediction.html#cb67-110" aria-hidden="true" tabindex="-1"></a><span class="co"># # Split the dataset: 70% for training, 30% for testing</span></span>
<span id="cb67-111"><a href="benchmark-ml-methods-for-prediction.html#cb67-111" aria-hidden="true" tabindex="-1"></a><span class="co"># set.seed(12345)</span></span>
<span id="cb67-112"><a href="benchmark-ml-methods-for-prediction.html#cb67-112" aria-hidden="true" tabindex="-1"></a><span class="co"># trainIndex &lt;- createDataPartition(data_selected$Response, p = 0.7, list = FALSE)</span></span>
<span id="cb67-113"><a href="benchmark-ml-methods-for-prediction.html#cb67-113" aria-hidden="true" tabindex="-1"></a><span class="co"># trainData &lt;- data_selected[trainIndex, ]</span></span>
<span id="cb67-114"><a href="benchmark-ml-methods-for-prediction.html#cb67-114" aria-hidden="true" tabindex="-1"></a><span class="co"># testData &lt;- data_selected[-trainIndex, ]</span></span>
<span id="cb67-115"><a href="benchmark-ml-methods-for-prediction.html#cb67-115" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb67-116"><a href="benchmark-ml-methods-for-prediction.html#cb67-116" aria-hidden="true" tabindex="-1"></a><span class="co"># # Define training control with 5-fold cross-validation repeated 50 times</span></span>
<span id="cb67-117"><a href="benchmark-ml-methods-for-prediction.html#cb67-117" aria-hidden="true" tabindex="-1"></a><span class="co"># control &lt;- trainControl(method = &quot;repeatedcv&quot;, number = 5, repeats = 50,</span></span>
<span id="cb67-118"><a href="benchmark-ml-methods-for-prediction.html#cb67-118" aria-hidden="true" tabindex="-1"></a><span class="co">#                         classProbs = TRUE, summaryFunction = twoClassSummary)</span></span>
<span id="cb67-119"><a href="benchmark-ml-methods-for-prediction.html#cb67-119" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb67-120"><a href="benchmark-ml-methods-for-prediction.html#cb67-120" aria-hidden="true" tabindex="-1"></a><span class="co"># # Define hyperparameter tuning grid</span></span>
<span id="cb67-121"><a href="benchmark-ml-methods-for-prediction.html#cb67-121" aria-hidden="true" tabindex="-1"></a><span class="co"># tuneGrid &lt;- expand.grid(.mtry = c(1, 3, 5))</span></span>
<span id="cb67-122"><a href="benchmark-ml-methods-for-prediction.html#cb67-122" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb67-123"><a href="benchmark-ml-methods-for-prediction.html#cb67-123" aria-hidden="true" tabindex="-1"></a><span class="co"># # Train a Random Forest model with hyperparameter tuning</span></span>
<span id="cb67-124"><a href="benchmark-ml-methods-for-prediction.html#cb67-124" aria-hidden="true" tabindex="-1"></a><span class="co"># rf_tuned &lt;- train(</span></span>
<span id="cb67-125"><a href="benchmark-ml-methods-for-prediction.html#cb67-125" aria-hidden="true" tabindex="-1"></a><span class="co">#   Response ~ ., </span></span>
<span id="cb67-126"><a href="benchmark-ml-methods-for-prediction.html#cb67-126" aria-hidden="true" tabindex="-1"></a><span class="co">#   data = trainData, </span></span>
<span id="cb67-127"><a href="benchmark-ml-methods-for-prediction.html#cb67-127" aria-hidden="true" tabindex="-1"></a><span class="co">#   method = &quot;rf&quot;, </span></span>
<span id="cb67-128"><a href="benchmark-ml-methods-for-prediction.html#cb67-128" aria-hidden="true" tabindex="-1"></a><span class="co">#   trControl = control, </span></span>
<span id="cb67-129"><a href="benchmark-ml-methods-for-prediction.html#cb67-129" aria-hidden="true" tabindex="-1"></a><span class="co">#   tuneGrid = tuneGrid, </span></span>
<span id="cb67-130"><a href="benchmark-ml-methods-for-prediction.html#cb67-130" aria-hidden="true" tabindex="-1"></a><span class="co">#   ntree = 50,  # Number of trees</span></span>
<span id="cb67-131"><a href="benchmark-ml-methods-for-prediction.html#cb67-131" aria-hidden="true" tabindex="-1"></a><span class="co">#   metric = &quot;ROC&quot;  # Use ROC as the evaluation metric</span></span>
<span id="cb67-132"><a href="benchmark-ml-methods-for-prediction.html#cb67-132" aria-hidden="true" tabindex="-1"></a><span class="co"># )</span></span>
<span id="cb67-133"><a href="benchmark-ml-methods-for-prediction.html#cb67-133" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb67-134"><a href="benchmark-ml-methods-for-prediction.html#cb67-134" aria-hidden="true" tabindex="-1"></a><span class="co"># # Save the trained Random Forest model to an RDS file</span></span>
<span id="cb67-135"><a href="benchmark-ml-methods-for-prediction.html#cb67-135" aria-hidden="true" tabindex="-1"></a><span class="co"># saveRDS(rf_tuned, file = &quot;../03.out/figure3/AZA_VEN_RandomForest8.rds&quot;)</span></span>
<span id="cb67-136"><a href="benchmark-ml-methods-for-prediction.html#cb67-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-137"><a href="benchmark-ml-methods-for-prediction.html#cb67-137" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract variable importance from the model</span></span>
<span id="cb67-138"><a href="benchmark-ml-methods-for-prediction.html#cb67-138" aria-hidden="true" tabindex="-1"></a><span class="co"># importance &lt;- varImp(rf_tuned, scale = FALSE)</span></span>
<span id="cb67-139"><a href="benchmark-ml-methods-for-prediction.html#cb67-139" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb67-140"><a href="benchmark-ml-methods-for-prediction.html#cb67-140" aria-hidden="true" tabindex="-1"></a><span class="co"># # Convert importance to a data frame for easier visualization</span></span>
<span id="cb67-141"><a href="benchmark-ml-methods-for-prediction.html#cb67-141" aria-hidden="true" tabindex="-1"></a><span class="co"># importance_df &lt;- data.frame(</span></span>
<span id="cb67-142"><a href="benchmark-ml-methods-for-prediction.html#cb67-142" aria-hidden="true" tabindex="-1"></a><span class="co">#   Feature = rownames(importance$importance),</span></span>
<span id="cb67-143"><a href="benchmark-ml-methods-for-prediction.html#cb67-143" aria-hidden="true" tabindex="-1"></a><span class="co">#   Importance = importance$importance$Overall</span></span>
<span id="cb67-144"><a href="benchmark-ml-methods-for-prediction.html#cb67-144" aria-hidden="true" tabindex="-1"></a><span class="co"># )</span></span>
<span id="cb67-145"><a href="benchmark-ml-methods-for-prediction.html#cb67-145" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb67-146"><a href="benchmark-ml-methods-for-prediction.html#cb67-146" aria-hidden="true" tabindex="-1"></a><span class="co"># # Sort features by importance</span></span>
<span id="cb67-147"><a href="benchmark-ml-methods-for-prediction.html#cb67-147" aria-hidden="true" tabindex="-1"></a><span class="co"># importance_df &lt;- importance_df[order(importance_df$Importance, decreasing = TRUE), ]</span></span>
<span id="cb67-148"><a href="benchmark-ml-methods-for-prediction.html#cb67-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-149"><a href="benchmark-ml-methods-for-prediction.html#cb67-149" aria-hidden="true" tabindex="-1"></a><span class="co"># Save feature importance to a CSV file</span></span>
<span id="cb67-150"><a href="benchmark-ml-methods-for-prediction.html#cb67-150" aria-hidden="true" tabindex="-1"></a><span class="co"># write.csv(importance_df, &quot;../03.out/figure3/RF_FeatureImportance.csv&quot;)</span></span></code></pre></div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="figure-3.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="predict-class-labels-on-the-test-set-and-plot-roc.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/03-results3.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
