<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3.8 Differential Expression Analysis for shPINK1 | Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning</title>
  <meta name="description" content="This is a book-style code sharing." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="3.8 Differential Expression Analysis for shPINK1 | Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a book-style code sharing." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3.8 Differential Expression Analysis for shPINK1 | Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning" />
  
  <meta name="twitter:description" content="This is a book-style code sharing." />
  

<meta name="author" content="Peng Jin, Qiqi Jin, Kankan Wang" />


<meta name="date" content="2025-07-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="gsea-analysis-for-crispri-drug-sensitivity.html"/>
<link rel="next" href="de-analysis-in-cell-lines.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Background</a></li>
<li class="chapter" data-level="2" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>2</b> Figure 1</a>
<ul>
<li class="chapter" data-level="2.1" data-path="process-the-data-from-functional_precision_medicine_tumor_board_aml.html"><a href="process-the-data-from-functional_precision_medicine_tumor_board_aml.html"><i class="fa fa-check"></i><b>2.1</b> Process the data from Functional_Precision_Medicine_Tumor_Board_AML</a></li>
<li class="chapter" data-level="2.2" data-path="functional_precision_medicine_tumor_board_aml.html"><a href="functional_precision_medicine_tumor_board_aml.html"><i class="fa fa-check"></i><b>2.2</b> Functional_Precision_Medicine_Tumor_Board_AML</a></li>
<li class="chapter" data-level="2.3" data-path="boxplot-for-drug-sensitivity-between-resistant-and-sensitive.html"><a href="boxplot-for-drug-sensitivity-between-resistant-and-sensitive.html"><i class="fa fa-check"></i><b>2.3</b> Boxplot for drug sensitivity between resistant and sensitive</a></li>
<li class="chapter" data-level="2.4" data-path="de-analysis-functional_medicine.html"><a href="de-analysis-functional_medicine.html"><i class="fa fa-check"></i><b>2.4</b> DE analysis Functional_medicine</a></li>
<li class="chapter" data-level="2.5" data-path="valcanoplot-plot-ex-vivo.html"><a href="valcanoplot-plot-ex-vivo.html"><i class="fa fa-check"></i><b>2.5</b> Valcanoplot plot (Ex vivo)</a></li>
<li class="chapter" data-level="2.6" data-path="heatmap-for-representative-de-genes.html"><a href="heatmap-for-representative-de-genes.html"><i class="fa fa-check"></i><b>2.6</b> Heatmap for representative DE genes</a></li>
<li class="chapter" data-level="2.7" data-path="cibersortx-results-visualization-ex-vivo.html"><a href="cibersortx-results-visualization-ex-vivo.html"><i class="fa fa-check"></i><b>2.7</b> CIBERSORTx Results Visualization (ex vivo)</a></li>
<li class="chapter" data-level="2.8" data-path="cibersortx-results-visualization-in-vivo.html"><a href="cibersortx-results-visualization-in-vivo.html"><i class="fa fa-check"></i><b>2.8</b> CIBERSORTx Results Visualization (in vivo)</a></li>
<li class="chapter" data-level="2.9" data-path="ssgsea-analysis-in-vivo.html"><a href="ssgsea-analysis-in-vivo.html"><i class="fa fa-check"></i><b>2.9</b> ssGSEA analysis in vivo</a></li>
<li class="chapter" data-level="2.10" data-path="go-enrichment-dot-plot.html"><a href="go-enrichment-dot-plot.html"><i class="fa fa-check"></i><b>2.10</b> GO Enrichment Dot Plot</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>3</b> Figure 2</a>
<ul>
<li class="chapter" data-level="3.1" data-path="de-analysis-between-in-vivo-sensitive-and-resistant.html"><a href="de-analysis-between-in-vivo-sensitive-and-resistant.html"><i class="fa fa-check"></i><b>3.1</b> DE analysis between in vivo sensitive and resistant</a></li>
<li class="chapter" data-level="3.2" data-path="de-analysis-rjaml-azaven-cohort.html"><a href="de-analysis-rjaml-azaven-cohort.html"><i class="fa fa-check"></i><b>3.2</b> DE analysis RJAML AZA+VEN cohort</a></li>
<li class="chapter" data-level="3.3" data-path="valcanoplot-plot-ex-vivo-1.html"><a href="valcanoplot-plot-ex-vivo-1.html"><i class="fa fa-check"></i><b>3.3</b> Valcanoplot plot (Ex vivo)</a></li>
<li class="chapter" data-level="3.4" data-path="valcanoplot-plot-in-vivo.html"><a href="valcanoplot-plot-in-vivo.html"><i class="fa fa-check"></i><b>3.4</b> Valcanoplot plot (In vivo)</a></li>
<li class="chapter" data-level="3.5" data-path="venn-plot.html"><a href="venn-plot.html"><i class="fa fa-check"></i><b>3.5</b> Venn plot</a></li>
<li class="chapter" data-level="3.6" data-path="integration-of-degs-ex-vivo-and-in-vivo.html"><a href="integration-of-degs-ex-vivo-and-in-vivo.html"><i class="fa fa-check"></i><b>3.6</b> Integration of DEGs ex vivo and in vivo</a></li>
<li class="chapter" data-level="3.7" data-path="gsea-analysis-for-crispri-drug-sensitivity.html"><a href="gsea-analysis-for-crispri-drug-sensitivity.html"><i class="fa fa-check"></i><b>3.7</b> GSEA analysis for CRISPRi drug sensitivity</a></li>
<li class="chapter" data-level="3.8" data-path="differential-expression-analysis-for-shpink1.html"><a href="differential-expression-analysis-for-shpink1.html"><i class="fa fa-check"></i><b>3.8</b> Differential Expression Analysis for shPINK1</a></li>
<li class="chapter" data-level="3.9" data-path="de-analysis-in-cell-lines.html"><a href="de-analysis-in-cell-lines.html"><i class="fa fa-check"></i><b>3.9</b> DE analysis in cell lines</a></li>
<li class="chapter" data-level="3.10" data-path="valcano-plots-molm13.html"><a href="valcano-plots-molm13.html"><i class="fa fa-check"></i><b>3.10</b> Valcano plots (MOLM13)</a></li>
<li class="chapter" data-level="3.11" data-path="valcano-plots-oa3.html"><a href="valcano-plots-oa3.html"><i class="fa fa-check"></i><b>3.11</b> Valcano plots (OA3)</a></li>
<li class="chapter" data-level="3.12" data-path="gsea-analysis-molm13.html"><a href="gsea-analysis-molm13.html"><i class="fa fa-check"></i><b>3.12</b> GSEA analysis (MOLM13)</a></li>
<li class="chapter" data-level="3.13" data-path="gsea-analysis-oa3.html"><a href="gsea-analysis-oa3.html"><i class="fa fa-check"></i><b>3.13</b> GSEA analysis (OA3)</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>4</b> Figure 3</a>
<ul>
<li class="chapter" data-level="4.1" data-path="benchmark-ml-methods-for-prediction.html"><a href="benchmark-ml-methods-for-prediction.html"><i class="fa fa-check"></i><b>4.1</b> Benchmark ML Methods for Prediction</a></li>
<li class="chapter" data-level="4.2" data-path="predict-class-labels-on-the-test-set-and-plot-roc.html"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html"><i class="fa fa-check"></i><b>4.2</b> Predict Class Labels on the Test Set and Plot ROC</a></li>
<li class="chapter" data-level="4.3" data-path="auc-and-auprc-heatmap-for-the-entire-and-test-sets.html"><a href="auc-and-auprc-heatmap-for-the-entire-and-test-sets.html"><i class="fa fa-check"></i><b>4.3</b> AUC and AUPRC Heatmap for the Entire and Test Sets</a></li>
<li class="chapter" data-level="4.4" data-path="model-performance-plot.html"><a href="model-performance-plot.html"><i class="fa fa-check"></i><b>4.4</b> Model Performance Plot</a></li>
<li class="chapter" data-level="4.5" data-path="feature-importance.html"><a href="feature-importance.html"><i class="fa fa-check"></i><b>4.5</b> Feature importance</a></li>
<li class="chapter" data-level="4.6" data-path="heatmap-for-confusion-matrices-defaut-cutoff-0.html"><a href="heatmap-for-confusion-matrices-defaut-cutoff-0.html"><i class="fa fa-check"></i><b>4.6</b> Heatmap for Confusion Matrices (defaut cutoff 0.5)</a></li>
<li class="chapter" data-level="4.7" data-path="barplot-for-predicted-response-0.html"><a href="barplot-for-predicted-response-0.html"><i class="fa fa-check"></i><b>4.7</b> Barplot for predicted Response (0.5)</a></li>
<li class="chapter" data-level="4.8" data-path="heatmap-for-confusion-matrices-optimal-cutoff-0.html"><a href="heatmap-for-confusion-matrices-optimal-cutoff-0.html"><i class="fa fa-check"></i><b>4.8</b> Heatmap for Confusion Matrices (optimal cutoff 0.630)</a></li>
<li class="chapter" data-level="4.9" data-path="barplot-for-predicted-response-0.html"><a href="barplot-for-predicted-response-0.html#barplot-for-predicted-response-0.630"><i class="fa fa-check"></i><b>4.9</b> Barplot for predicted Response (0.630)</a></li>
<li class="chapter" data-level="4.10" data-path="density-plot-for-response-probability.html"><a href="density-plot-for-response-probability.html"><i class="fa fa-check"></i><b>4.10</b> Density Plot for Response Probability</a></li>
<li class="chapter" data-level="4.11" data-path="violin-and-boxplots-for-response-probability-rf8.html"><a href="violin-and-boxplots-for-response-probability-rf8.html"><i class="fa fa-check"></i><b>4.11</b> Violin and Boxplots for Response Probability (RF8)</a></li>
<li class="chapter" data-level="4.12" data-path="barplot-for-response.html"><a href="barplot-for-response.html"><i class="fa fa-check"></i><b>4.12</b> Barplot for Response</a></li>
<li class="chapter" data-level="4.13" data-path="rf8-for-mdr-status.html"><a href="rf8-for-mdr-status.html"><i class="fa fa-check"></i><b>4.13</b> RF8 for MDR status</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>5</b> Figure 4</a>
<ul>
<li class="chapter" data-level="5.1" data-path="validation-of-rf8-model-rjaml-cohort-2.html"><a href="validation-of-rf8-model-rjaml-cohort-2.html"><i class="fa fa-check"></i><b>5.1</b> Validation of RF8 model (RJAML-Cohort 2)</a></li>
<li class="chapter" data-level="5.2" data-path="validation-of-rf8-model-fpmtb.html"><a href="validation-of-rf8-model-fpmtb.html"><i class="fa fa-check"></i><b>5.2</b> Validation of RF8 model (FPMTB)</a></li>
<li class="chapter" data-level="5.3" data-path="validation-of-rf8-model-beataml.html"><a href="validation-of-rf8-model-beataml.html"><i class="fa fa-check"></i><b>5.3</b> Validation of RF8 model (BeatAML)</a></li>
<li class="chapter" data-level="5.4" data-path="violin-and-boxplots-for-response-probability-rf8-1.html"><a href="violin-and-boxplots-for-response-probability-rf8-1.html"><i class="fa fa-check"></i><b>5.4</b> Violin and Boxplots for Response Probability (RF8)</a></li>
<li class="chapter" data-level="5.5" data-path="rf8-score-among-clusters.html"><a href="rf8-score-among-clusters.html"><i class="fa fa-check"></i><b>5.5</b> RF8 score among clusters</a></li>
<li class="chapter" data-level="5.6" data-path="integration-of-ex-vivo-drug-response-in-beataml.html"><a href="integration-of-ex-vivo-drug-response-in-beataml.html"><i class="fa fa-check"></i><b>5.6</b> Integration of ex vivo drug-response in BeatAML</a></li>
<li class="chapter" data-level="5.7" data-path="beataml-waves-1-4-heatmap-based-on-drug-profiling.html"><a href="beataml-waves-1-4-heatmap-based-on-drug-profiling.html"><i class="fa fa-check"></i><b>5.7</b> BeatAML Waves 1-4 heatmap based on drug profiling</a></li>
<li class="chapter" data-level="5.8" data-path="boxplot-for-drug-sensitivitity-between-resistant-and-sensitive.html"><a href="boxplot-for-drug-sensitivitity-between-resistant-and-sensitive.html"><i class="fa fa-check"></i><b>5.8</b> Boxplot for drug sensitivitity between resistant and sensitive</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>6</b> Figure 5</a>
<ul>
<li class="chapter" data-level="6.1" data-path="integration-with-aml-genomics.html"><a href="integration-with-aml-genomics.html"><i class="fa fa-check"></i><b>6.1</b> Integration with AML genomics</a></li>
<li class="chapter" data-level="6.2" data-path="mutation-pairs.html"><a href="mutation-pairs.html"><i class="fa fa-check"></i><b>6.2</b> Mutation pairs</a></li>
<li class="chapter" data-level="6.3" data-path="complexheatmap-for-mutation.html"><a href="complexheatmap-for-mutation.html"><i class="fa fa-check"></i><b>6.3</b> Complexheatmap for mutation</a></li>
<li class="chapter" data-level="6.4" data-path="fisher-test.html"><a href="fisher-test.html"><i class="fa fa-check"></i><b>6.4</b> Fisher test</a></li>
<li class="chapter" data-level="6.5" data-path="barplot-in-rjaml-cohort.html"><a href="barplot-in-rjaml-cohort.html"><i class="fa fa-check"></i><b>6.5</b> Barplot in RJAML cohort</a></li>
<li class="chapter" data-level="6.6" data-path="table-one-for-multi-cohorts.html"><a href="table-one-for-multi-cohorts.html"><i class="fa fa-check"></i><b>6.6</b> Table one for multi-cohorts</a></li>
<li class="chapter" data-level="6.7" data-path="barplot-for-real-response.html"><a href="barplot-for-real-response.html"><i class="fa fa-check"></i><b>6.7</b> Barplot for real response</a></li>
<li class="chapter" data-level="6.8" data-path="multivariable-logisticreg.html"><a href="multivariable-logisticreg.html"><i class="fa fa-check"></i><b>6.8</b> Multivariable LogisticReg</a></li>
<li class="chapter" data-level="6.9" data-path="identification-of-rf8-correlated-genes.html"><a href="identification-of-rf8-correlated-genes.html"><i class="fa fa-check"></i><b>6.9</b> Identification of RF8-correlated genes</a></li>
<li class="chapter" data-level="6.10" data-path="differential-expression-analysis-for-mut-vs.-wt.html"><a href="differential-expression-analysis-for-mut-vs.-wt.html"><i class="fa fa-check"></i><b>6.10</b> Differential Expression Analysis for Mut vs. WT</a></li>
<li class="chapter" data-level="6.11" data-path="venn-diagram-for-target-genes.html"><a href="venn-diagram-for-target-genes.html"><i class="fa fa-check"></i><b>6.11</b> Venn Diagram for target genes</a></li>
<li class="chapter" data-level="6.12" data-path="pathway-overlap-analysis-of-genes.html"><a href="pathway-overlap-analysis-of-genes.html"><i class="fa fa-check"></i><b>6.12</b> Pathway overlap analysis of genes</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="figure-6.html"><a href="figure-6.html"><i class="fa fa-check"></i><b>7</b> Figure 6</a>
<ul>
<li class="chapter" data-level="7.1" data-path="relationship-between-rf8-score-vs.-venaza-response-probability.html"><a href="relationship-between-rf8-score-vs.-venaza-response-probability.html"><i class="fa fa-check"></i><b>7.1</b> Relationship between RF8 score vs. VEN/AZA response probability</a></li>
<li class="chapter" data-level="7.2" data-path="monotonic-response-prediction-plot.html"><a href="monotonic-response-prediction-plot.html"><i class="fa fa-check"></i><b>7.2</b> Monotonic Response Prediction Plot</a></li>
<li class="chapter" data-level="7.3" data-path="survival-plot.html"><a href="survival-plot.html"><i class="fa fa-check"></i><b>7.3</b> Survival plot</a></li>
<li class="chapter" data-level="7.4" data-path="brier-score-analysis.html"><a href="brier-score-analysis.html"><i class="fa fa-check"></i><b>7.4</b> Brier Score Analysis</a></li>
<li class="chapter" data-level="7.5" data-path="c-index-comparison.html"><a href="c-index-comparison.html"><i class="fa fa-check"></i><b>7.5</b> C-index Comparison</a></li>
<li class="chapter" data-level="7.6" data-path="km-plots-for-rjaml-cohort-2.html"><a href="km-plots-for-rjaml-cohort-2.html"><i class="fa fa-check"></i><b>7.6</b> KM-Plots for RJAML Cohort 2</a></li>
<li class="chapter" data-level="7.7" data-path="barplot-for-crcri.html"><a href="barplot-for-crcri.html"><i class="fa fa-check"></i><b>7.7</b> Barplot for CR/CRi</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="differential-expression-analysis-for-shpink1" class="section level2 hasAnchor" number="3.8">
<h2><span class="header-section-number">3.8</span> Differential Expression Analysis for shPINK1<a href="differential-expression-analysis-for-shpink1.html#differential-expression-analysis-for-shpink1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="differential-expression-analysis-for-shpink1.html#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################</span></span>
<span id="cb53-2"><a href="differential-expression-analysis-for-shpink1.html#cb53-2" aria-hidden="true" tabindex="-1"></a><span class="do">### Differential Expression Analysis for shPINK1</span></span>
<span id="cb53-3"><a href="differential-expression-analysis-for-shpink1.html#cb53-3" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################</span></span>
<span id="cb53-4"><a href="differential-expression-analysis-for-shpink1.html#cb53-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-5"><a href="differential-expression-analysis-for-shpink1.html#cb53-5" aria-hidden="true" tabindex="-1"></a><span class="co"># library(GenomicFeatures)</span></span>
<span id="cb53-6"><a href="differential-expression-analysis-for-shpink1.html#cb53-6" aria-hidden="true" tabindex="-1"></a><span class="co"># library(stringr)</span></span>
<span id="cb53-7"><a href="differential-expression-analysis-for-shpink1.html#cb53-7" aria-hidden="true" tabindex="-1"></a><span class="co"># dir &lt;- &quot;../02.data/&quot;</span></span>
<span id="cb53-8"><a href="differential-expression-analysis-for-shpink1.html#cb53-8" aria-hidden="true" tabindex="-1"></a><span class="co"># setwd(dir)</span></span>
<span id="cb53-9"><a href="differential-expression-analysis-for-shpink1.html#cb53-9" aria-hidden="true" tabindex="-1"></a><span class="co"># files &lt;- list.files(dir, pattern = &quot;.count&quot;, recursive=TRUE)</span></span>
<span id="cb53-10"><a href="differential-expression-analysis-for-shpink1.html#cb53-10" aria-hidden="true" tabindex="-1"></a><span class="co"># count.files &lt;- list()</span></span>
<span id="cb53-11"><a href="differential-expression-analysis-for-shpink1.html#cb53-11" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb53-12"><a href="differential-expression-analysis-for-shpink1.html#cb53-12" aria-hidden="true" tabindex="-1"></a><span class="co"># for(i in 1:length(files)){</span></span>
<span id="cb53-13"><a href="differential-expression-analysis-for-shpink1.html#cb53-13" aria-hidden="true" tabindex="-1"></a><span class="co">#   count.files[[i]] &lt;- read.table(files[[i]], header=FALSE, sep=&quot;\t&quot;, row.names=1)</span></span>
<span id="cb53-14"><a href="differential-expression-analysis-for-shpink1.html#cb53-14" aria-hidden="true" tabindex="-1"></a><span class="co"># }</span></span>
<span id="cb53-15"><a href="differential-expression-analysis-for-shpink1.html#cb53-15" aria-hidden="true" tabindex="-1"></a><span class="co"># count.matrix &lt;- do.call(cbind, count.files)</span></span>
<span id="cb53-16"><a href="differential-expression-analysis-for-shpink1.html#cb53-16" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames &lt;- gsub(&quot;.count&quot;, &quot;&quot;, files)</span></span>
<span id="cb53-17"><a href="differential-expression-analysis-for-shpink1.html#cb53-17" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(count.matrix) &lt;- colnames</span></span>
<span id="cb53-18"><a href="differential-expression-analysis-for-shpink1.html#cb53-18" aria-hidden="true" tabindex="-1"></a><span class="co"># count.matrix &lt;- count.matrix[1:60483, ]</span></span>
<span id="cb53-19"><a href="differential-expression-analysis-for-shpink1.html#cb53-19" aria-hidden="true" tabindex="-1"></a><span class="co"># ## Save the count table</span></span>
<span id="cb53-20"><a href="differential-expression-analysis-for-shpink1.html#cb53-20" aria-hidden="true" tabindex="-1"></a><span class="co"># write.table(count.matrix, &quot;../03.out/figure2/shPINK1_count.matrix.txt&quot;, sep=&quot;\t&quot;, row.names=T, col.names=NA, quote=FALSE)</span></span>
<span id="cb53-21"><a href="differential-expression-analysis-for-shpink1.html#cb53-21" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb53-22"><a href="differential-expression-analysis-for-shpink1.html#cb53-22" aria-hidden="true" tabindex="-1"></a><span class="co"># gtffile &lt;- &quot;gencode.v22.annotation.gtf.gz&quot;</span></span>
<span id="cb53-23"><a href="differential-expression-analysis-for-shpink1.html#cb53-23" aria-hidden="true" tabindex="-1"></a><span class="co"># txdb &lt;- makeTxDbFromGFF(gtffile, format=&quot;gtf&quot;)</span></span>
<span id="cb53-24"><a href="differential-expression-analysis-for-shpink1.html#cb53-24" aria-hidden="true" tabindex="-1"></a><span class="co"># ebg &lt;- exonsBy(txdb, by=&quot;gene&quot;)</span></span>
<span id="cb53-25"><a href="differential-expression-analysis-for-shpink1.html#cb53-25" aria-hidden="true" tabindex="-1"></a><span class="co"># ebgList &lt;- sum(width(reduce(ebg)))</span></span>
<span id="cb53-26"><a href="differential-expression-analysis-for-shpink1.html#cb53-26" aria-hidden="true" tabindex="-1"></a><span class="co"># genes &lt;- intersect(rownames(count.matrix), names(ebgList))</span></span>
<span id="cb53-27"><a href="differential-expression-analysis-for-shpink1.html#cb53-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Length &lt;- as.vector(ebgList[genes])</span></span>
<span id="cb53-28"><a href="differential-expression-analysis-for-shpink1.html#cb53-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Length &lt;- as.vector(Length)</span></span>
<span id="cb53-29"><a href="differential-expression-analysis-for-shpink1.html#cb53-29" aria-hidden="true" tabindex="-1"></a><span class="co"># ## Normalize</span></span>
<span id="cb53-30"><a href="differential-expression-analysis-for-shpink1.html#cb53-30" aria-hidden="true" tabindex="-1"></a><span class="co"># TPM &lt;- t(t(count.matrix / t(Length)) * 1e6 / colSums(count.matrix / t(Length)))</span></span>
<span id="cb53-31"><a href="differential-expression-analysis-for-shpink1.html#cb53-31" aria-hidden="true" tabindex="-1"></a><span class="co"># TPM &lt;- data.frame(TPM)</span></span>
<span id="cb53-32"><a href="differential-expression-analysis-for-shpink1.html#cb53-32" aria-hidden="true" tabindex="-1"></a><span class="co"># TPM$ID &lt;- row.names(TPM)</span></span>
<span id="cb53-33"><a href="differential-expression-analysis-for-shpink1.html#cb53-33" aria-hidden="true" tabindex="-1"></a><span class="co"># ## gene anno</span></span>
<span id="cb53-34"><a href="differential-expression-analysis-for-shpink1.html#cb53-34" aria-hidden="true" tabindex="-1"></a><span class="co"># anno &lt;- read.table(&quot;geneAnnotation.txt&quot;, header = F)</span></span>
<span id="cb53-35"><a href="differential-expression-analysis-for-shpink1.html#cb53-35" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(anno) &lt;- c(&quot;ID&quot;, &quot;Symbol&quot;, &quot;Type&quot;)</span></span>
<span id="cb53-36"><a href="differential-expression-analysis-for-shpink1.html#cb53-36" aria-hidden="true" tabindex="-1"></a><span class="co"># tpm &lt;- merge(anno, TPM, by = &quot;ID&quot;)</span></span>
<span id="cb53-37"><a href="differential-expression-analysis-for-shpink1.html#cb53-37" aria-hidden="true" tabindex="-1"></a><span class="co"># write.table(tpm, &quot;../03.out/figure2/shPINK1_ID_matched_TPM_matrix.txt&quot;)</span></span>
<span id="cb53-38"><a href="differential-expression-analysis-for-shpink1.html#cb53-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-39"><a href="differential-expression-analysis-for-shpink1.html#cb53-39" aria-hidden="true" tabindex="-1"></a><span class="do">####</span></span>
<span id="cb53-40"><a href="differential-expression-analysis-for-shpink1.html#cb53-40" aria-hidden="true" tabindex="-1"></a><span class="do">#### PCA analysis</span></span>
<span id="cb53-41"><a href="differential-expression-analysis-for-shpink1.html#cb53-41" aria-hidden="true" tabindex="-1"></a><span class="do">####</span></span>
<span id="cb53-42"><a href="differential-expression-analysis-for-shpink1.html#cb53-42" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb53-43"><a href="differential-expression-analysis-for-shpink1.html#cb53-43" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb53-44"><a href="differential-expression-analysis-for-shpink1.html#cb53-44" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggfortify)</span>
<span id="cb53-45"><a href="differential-expression-analysis-for-shpink1.html#cb53-45" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb53-46"><a href="differential-expression-analysis-for-shpink1.html#cb53-46" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scatterplot3d)</span>
<span id="cb53-47"><a href="differential-expression-analysis-for-shpink1.html#cb53-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-48"><a href="differential-expression-analysis-for-shpink1.html#cb53-48" aria-hidden="true" tabindex="-1"></a>tpm <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../03.out/figure2/shPINK1_ID_matched_TPM_matrix.txt&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">check.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb53-49"><a href="differential-expression-analysis-for-shpink1.html#cb53-49" aria-hidden="true" tabindex="-1"></a><span class="do">## MOLM13 cell lines</span></span>
<span id="cb53-50"><a href="differential-expression-analysis-for-shpink1.html#cb53-50" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> tpm[, <span class="dv">4</span><span class="sc">:</span><span class="dv">9</span>]</span>
<span id="cb53-51"><a href="differential-expression-analysis-for-shpink1.html#cb53-51" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(dat) <span class="ot">&lt;-</span> tpm<span class="sc">$</span>ID</span>
<span id="cb53-52"><a href="differential-expression-analysis-for-shpink1.html#cb53-52" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> dat[<span class="fu">which</span>(<span class="fu">rowMeans</span>(dat[, <span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)]) <span class="sc">&gt;</span> <span class="dv">0</span>), ]</span>
<span id="cb53-53"><a href="differential-expression-analysis-for-shpink1.html#cb53-53" aria-hidden="true" tabindex="-1"></a>scaled_matrix <span class="ot">&lt;-</span> <span class="fu">scale</span>(<span class="fu">t</span>(dat))</span>
<span id="cb53-54"><a href="differential-expression-analysis-for-shpink1.html#cb53-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-55"><a href="differential-expression-analysis-for-shpink1.html#cb53-55" aria-hidden="true" tabindex="-1"></a>pca_result <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(scaled_matrix, <span class="at">scale. =</span> <span class="cn">TRUE</span>)</span>
<span id="cb53-56"><a href="differential-expression-analysis-for-shpink1.html#cb53-56" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a grouping vector</span></span>
<span id="cb53-57"><a href="differential-expression-analysis-for-shpink1.html#cb53-57" aria-hidden="true" tabindex="-1"></a>groups <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;Scramble&quot;</span>, <span class="dv">2</span>), <span class="fu">rep</span>(<span class="st">&quot;shPINK1-1&quot;</span>, <span class="dv">2</span>), <span class="fu">rep</span>(<span class="st">&quot;shPINK1-2&quot;</span>, <span class="dv">2</span>))</span>
<span id="cb53-58"><a href="differential-expression-analysis-for-shpink1.html#cb53-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-59"><a href="differential-expression-analysis-for-shpink1.html#cb53-59" aria-hidden="true" tabindex="-1"></a>pca_2d_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pca_result<span class="sc">$</span>x)</span>
<span id="cb53-60"><a href="differential-expression-analysis-for-shpink1.html#cb53-60" aria-hidden="true" tabindex="-1"></a>pca_2d_df<span class="sc">$</span>Group <span class="ot">&lt;-</span> groups</span>
<span id="cb53-61"><a href="differential-expression-analysis-for-shpink1.html#cb53-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-62"><a href="differential-expression-analysis-for-shpink1.html#cb53-62" aria-hidden="true" tabindex="-1"></a><span class="co"># Define colors for groups</span></span>
<span id="cb53-63"><a href="differential-expression-analysis-for-shpink1.html#cb53-63" aria-hidden="true" tabindex="-1"></a>group_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;green&quot;</span>, <span class="st">&quot;blue&quot;</span>)</span>
<span id="cb53-64"><a href="differential-expression-analysis-for-shpink1.html#cb53-64" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(group_colors) <span class="ot">&lt;-</span> <span class="fu">unique</span>(groups)</span>
<span id="cb53-65"><a href="differential-expression-analysis-for-shpink1.html#cb53-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-66"><a href="differential-expression-analysis-for-shpink1.html#cb53-66" aria-hidden="true" tabindex="-1"></a><span class="co"># Add variance explained for axis labels</span></span>
<span id="cb53-67"><a href="differential-expression-analysis-for-shpink1.html#cb53-67" aria-hidden="true" tabindex="-1"></a>explained_variance <span class="ot">&lt;-</span> <span class="fu">summary</span>(pca_result)<span class="sc">$</span>importance[<span class="dv">2</span>, ]</span>
<span id="cb53-68"><a href="differential-expression-analysis-for-shpink1.html#cb53-68" aria-hidden="true" tabindex="-1"></a>x_label <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;PC1 (&quot;</span>, <span class="fu">round</span>(explained_variance[<span class="dv">1</span>] <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">2</span>), <span class="st">&quot;%)&quot;</span>)</span>
<span id="cb53-69"><a href="differential-expression-analysis-for-shpink1.html#cb53-69" aria-hidden="true" tabindex="-1"></a>y_label <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;PC2 (&quot;</span>, <span class="fu">round</span>(explained_variance[<span class="dv">2</span>] <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">2</span>), <span class="st">&quot;%)&quot;</span>)</span>
<span id="cb53-70"><a href="differential-expression-analysis-for-shpink1.html#cb53-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-71"><a href="differential-expression-analysis-for-shpink1.html#cb53-71" aria-hidden="true" tabindex="-1"></a><span class="co"># pdf(&quot;../03.out/figure2/shPINK1_M13_2D_PCA.pdf&quot;, width = 5.3, height = 4)</span></span>
<span id="cb53-72"><a href="differential-expression-analysis-for-shpink1.html#cb53-72" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_2d_df, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> Group)) <span class="sc">+</span></span>
<span id="cb53-73"><a href="differential-expression-analysis-for-shpink1.html#cb53-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb53-74"><a href="differential-expression-analysis-for-shpink1.html#cb53-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> group_colors) <span class="sc">+</span></span>
<span id="cb53-75"><a href="differential-expression-analysis-for-shpink1.html#cb53-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb53-76"><a href="differential-expression-analysis-for-shpink1.html#cb53-76" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">paste0</span>(<span class="st">&quot;PC1 (&quot;</span>, <span class="fu">round</span>(explained_variance[<span class="dv">1</span>] <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">2</span>), <span class="st">&quot;%)&quot;</span>),</span>
<span id="cb53-77"><a href="differential-expression-analysis-for-shpink1.html#cb53-77" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">paste0</span>(<span class="st">&quot;PC2 (&quot;</span>, <span class="fu">round</span>(explained_variance[<span class="dv">2</span>] <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">2</span>), <span class="st">&quot;%)&quot;</span>),</span>
<span id="cb53-78"><a href="differential-expression-analysis-for-shpink1.html#cb53-78" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;MOLM13 cells&quot;</span></span>
<span id="cb53-79"><a href="differential-expression-analysis-for-shpink1.html#cb53-79" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb53-80"><a href="differential-expression-analysis-for-shpink1.html#cb53-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb53-81"><a href="differential-expression-analysis-for-shpink1.html#cb53-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.title =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb53-82"><a href="differential-expression-analysis-for-shpink1.html#cb53-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> <span class="st">&quot;Groups&quot;</span>))</span>
<span id="cb53-83"><a href="differential-expression-analysis-for-shpink1.html#cb53-83" aria-hidden="true" tabindex="-1"></a><span class="co"># dev.off()</span></span>
<span id="cb53-84"><a href="differential-expression-analysis-for-shpink1.html#cb53-84" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p)</span></code></pre></div>
<p><img src="figures/unnamed-chunk-33-1.png" width="672" /></p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="differential-expression-analysis-for-shpink1.html#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="do">## OA3 cell lines</span></span>
<span id="cb54-2"><a href="differential-expression-analysis-for-shpink1.html#cb54-2" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> tpm[, <span class="dv">10</span><span class="sc">:</span><span class="dv">15</span>]</span>
<span id="cb54-3"><a href="differential-expression-analysis-for-shpink1.html#cb54-3" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(dat) <span class="ot">&lt;-</span> tpm<span class="sc">$</span>ID</span>
<span id="cb54-4"><a href="differential-expression-analysis-for-shpink1.html#cb54-4" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> dat[<span class="fu">which</span>(<span class="fu">rowMeans</span>(dat[, <span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)]) <span class="sc">&gt;</span> <span class="dv">0</span>), ]</span>
<span id="cb54-5"><a href="differential-expression-analysis-for-shpink1.html#cb54-5" aria-hidden="true" tabindex="-1"></a>scaled_matrix <span class="ot">&lt;-</span> <span class="fu">scale</span>(<span class="fu">t</span>(dat))</span>
<span id="cb54-6"><a href="differential-expression-analysis-for-shpink1.html#cb54-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-7"><a href="differential-expression-analysis-for-shpink1.html#cb54-7" aria-hidden="true" tabindex="-1"></a>pca_result <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(scaled_matrix, <span class="at">scale. =</span> <span class="cn">TRUE</span>)</span>
<span id="cb54-8"><a href="differential-expression-analysis-for-shpink1.html#cb54-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a grouping vector</span></span>
<span id="cb54-9"><a href="differential-expression-analysis-for-shpink1.html#cb54-9" aria-hidden="true" tabindex="-1"></a>groups <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;Scramble&quot;</span>, <span class="dv">2</span>), <span class="fu">rep</span>(<span class="st">&quot;shPINK1-1&quot;</span>, <span class="dv">2</span>), <span class="fu">rep</span>(<span class="st">&quot;shPINK1-2&quot;</span>, <span class="dv">2</span>))</span>
<span id="cb54-10"><a href="differential-expression-analysis-for-shpink1.html#cb54-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-11"><a href="differential-expression-analysis-for-shpink1.html#cb54-11" aria-hidden="true" tabindex="-1"></a>pca_2d_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pca_result<span class="sc">$</span>x)</span>
<span id="cb54-12"><a href="differential-expression-analysis-for-shpink1.html#cb54-12" aria-hidden="true" tabindex="-1"></a>pca_2d_df<span class="sc">$</span>Group <span class="ot">&lt;-</span> groups</span>
<span id="cb54-13"><a href="differential-expression-analysis-for-shpink1.html#cb54-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-14"><a href="differential-expression-analysis-for-shpink1.html#cb54-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Define colors for groups</span></span>
<span id="cb54-15"><a href="differential-expression-analysis-for-shpink1.html#cb54-15" aria-hidden="true" tabindex="-1"></a>group_colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;green&quot;</span>, <span class="st">&quot;blue&quot;</span>)</span>
<span id="cb54-16"><a href="differential-expression-analysis-for-shpink1.html#cb54-16" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(group_colors) <span class="ot">&lt;-</span> <span class="fu">unique</span>(groups)</span>
<span id="cb54-17"><a href="differential-expression-analysis-for-shpink1.html#cb54-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-18"><a href="differential-expression-analysis-for-shpink1.html#cb54-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Add variance explained for axis labels</span></span>
<span id="cb54-19"><a href="differential-expression-analysis-for-shpink1.html#cb54-19" aria-hidden="true" tabindex="-1"></a>explained_variance <span class="ot">&lt;-</span> <span class="fu">summary</span>(pca_result)<span class="sc">$</span>importance[<span class="dv">2</span>, ]</span>
<span id="cb54-20"><a href="differential-expression-analysis-for-shpink1.html#cb54-20" aria-hidden="true" tabindex="-1"></a>x_label <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;PC1 (&quot;</span>, <span class="fu">round</span>(explained_variance[<span class="dv">1</span>] <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">2</span>), <span class="st">&quot;%)&quot;</span>)</span>
<span id="cb54-21"><a href="differential-expression-analysis-for-shpink1.html#cb54-21" aria-hidden="true" tabindex="-1"></a>y_label <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;PC2 (&quot;</span>, <span class="fu">round</span>(explained_variance[<span class="dv">2</span>] <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">2</span>), <span class="st">&quot;%)&quot;</span>)</span>
<span id="cb54-22"><a href="differential-expression-analysis-for-shpink1.html#cb54-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-23"><a href="differential-expression-analysis-for-shpink1.html#cb54-23" aria-hidden="true" tabindex="-1"></a><span class="co"># pdf(&quot;../03.out/figure2/shPINK1_OA3_2D_PCA.pdf&quot;, width = 5.3, height = 4)</span></span>
<span id="cb54-24"><a href="differential-expression-analysis-for-shpink1.html#cb54-24" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_2d_df, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> Group)) <span class="sc">+</span></span>
<span id="cb54-25"><a href="differential-expression-analysis-for-shpink1.html#cb54-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb54-26"><a href="differential-expression-analysis-for-shpink1.html#cb54-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> group_colors) <span class="sc">+</span></span>
<span id="cb54-27"><a href="differential-expression-analysis-for-shpink1.html#cb54-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb54-28"><a href="differential-expression-analysis-for-shpink1.html#cb54-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="fu">paste0</span>(<span class="st">&quot;PC1 (&quot;</span>, <span class="fu">round</span>(explained_variance[<span class="dv">1</span>] <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">2</span>), <span class="st">&quot;%)&quot;</span>),</span>
<span id="cb54-29"><a href="differential-expression-analysis-for-shpink1.html#cb54-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">paste0</span>(<span class="st">&quot;PC2 (&quot;</span>, <span class="fu">round</span>(explained_variance[<span class="dv">2</span>] <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">2</span>), <span class="st">&quot;%)&quot;</span>),</span>
<span id="cb54-30"><a href="differential-expression-analysis-for-shpink1.html#cb54-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">&quot;OA3 cells&quot;</span></span>
<span id="cb54-31"><a href="differential-expression-analysis-for-shpink1.html#cb54-31" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb54-32"><a href="differential-expression-analysis-for-shpink1.html#cb54-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb54-33"><a href="differential-expression-analysis-for-shpink1.html#cb54-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.title =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb54-34"><a href="differential-expression-analysis-for-shpink1.html#cb54-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> <span class="st">&quot;Groups&quot;</span>))</span>
<span id="cb54-35"><a href="differential-expression-analysis-for-shpink1.html#cb54-35" aria-hidden="true" tabindex="-1"></a><span class="co"># dev.off()</span></span>
<span id="cb54-36"><a href="differential-expression-analysis-for-shpink1.html#cb54-36" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p1)</span></code></pre></div>
<p><img src="figures/unnamed-chunk-33-2.png" width="672" /></p>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="gsea-analysis-for-crispri-drug-sensitivity.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="de-analysis-in-cell-lines.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/02-results2.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
