<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6.10 Differential Expression Analysis for Mut vs. WT | Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning</title>
  <meta name="description" content="This is a book-style code sharing." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="6.10 Differential Expression Analysis for Mut vs. WT | Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a book-style code sharing." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6.10 Differential Expression Analysis for Mut vs. WT | Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning" />
  
  <meta name="twitter:description" content="This is a book-style code sharing." />
  

<meta name="author" content="Peng Jin, Qiqi Jin, Kankan Wang" />


<meta name="date" content="2025-07-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="identification-of-rf8-correlated-genes.html"/>
<link rel="next" href="venn-diagram-for-target-genes.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Background</a></li>
<li class="chapter" data-level="2" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>2</b> Figure 1</a>
<ul>
<li class="chapter" data-level="2.1" data-path="process-the-data-from-functional_precision_medicine_tumor_board_aml.html"><a href="process-the-data-from-functional_precision_medicine_tumor_board_aml.html"><i class="fa fa-check"></i><b>2.1</b> Process the data from Functional_Precision_Medicine_Tumor_Board_AML</a></li>
<li class="chapter" data-level="2.2" data-path="functional_precision_medicine_tumor_board_aml.html"><a href="functional_precision_medicine_tumor_board_aml.html"><i class="fa fa-check"></i><b>2.2</b> Functional_Precision_Medicine_Tumor_Board_AML</a></li>
<li class="chapter" data-level="2.3" data-path="boxplot-for-drug-sensitivity-between-resistant-and-sensitive.html"><a href="boxplot-for-drug-sensitivity-between-resistant-and-sensitive.html"><i class="fa fa-check"></i><b>2.3</b> Boxplot for drug sensitivity between resistant and sensitive</a></li>
<li class="chapter" data-level="2.4" data-path="de-analysis-functional_medicine.html"><a href="de-analysis-functional_medicine.html"><i class="fa fa-check"></i><b>2.4</b> DE analysis Functional_medicine</a></li>
<li class="chapter" data-level="2.5" data-path="valcanoplot-plot-ex-vivo.html"><a href="valcanoplot-plot-ex-vivo.html"><i class="fa fa-check"></i><b>2.5</b> Valcanoplot plot (Ex vivo)</a></li>
<li class="chapter" data-level="2.6" data-path="heatmap-for-representative-de-genes.html"><a href="heatmap-for-representative-de-genes.html"><i class="fa fa-check"></i><b>2.6</b> Heatmap for representative DE genes</a></li>
<li class="chapter" data-level="2.7" data-path="cibersortx-results-visualization-ex-vivo.html"><a href="cibersortx-results-visualization-ex-vivo.html"><i class="fa fa-check"></i><b>2.7</b> CIBERSORTx Results Visualization (ex vivo)</a></li>
<li class="chapter" data-level="2.8" data-path="cibersortx-results-visualization-in-vivo.html"><a href="cibersortx-results-visualization-in-vivo.html"><i class="fa fa-check"></i><b>2.8</b> CIBERSORTx Results Visualization (in vivo)</a></li>
<li class="chapter" data-level="2.9" data-path="ssgsea-analysis-in-vivo.html"><a href="ssgsea-analysis-in-vivo.html"><i class="fa fa-check"></i><b>2.9</b> ssGSEA analysis in vivo</a></li>
<li class="chapter" data-level="2.10" data-path="go-enrichment-dot-plot.html"><a href="go-enrichment-dot-plot.html"><i class="fa fa-check"></i><b>2.10</b> GO Enrichment Dot Plot</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>3</b> Figure 2</a>
<ul>
<li class="chapter" data-level="3.1" data-path="de-analysis-between-in-vivo-sensitive-and-resistant.html"><a href="de-analysis-between-in-vivo-sensitive-and-resistant.html"><i class="fa fa-check"></i><b>3.1</b> DE analysis between in vivo sensitive and resistant</a></li>
<li class="chapter" data-level="3.2" data-path="de-analysis-rjaml-azaven-cohort.html"><a href="de-analysis-rjaml-azaven-cohort.html"><i class="fa fa-check"></i><b>3.2</b> DE analysis RJAML AZA+VEN cohort</a></li>
<li class="chapter" data-level="3.3" data-path="valcanoplot-plot-ex-vivo-1.html"><a href="valcanoplot-plot-ex-vivo-1.html"><i class="fa fa-check"></i><b>3.3</b> Valcanoplot plot (Ex vivo)</a></li>
<li class="chapter" data-level="3.4" data-path="valcanoplot-plot-in-vivo.html"><a href="valcanoplot-plot-in-vivo.html"><i class="fa fa-check"></i><b>3.4</b> Valcanoplot plot (In vivo)</a></li>
<li class="chapter" data-level="3.5" data-path="venn-plot.html"><a href="venn-plot.html"><i class="fa fa-check"></i><b>3.5</b> Venn plot</a></li>
<li class="chapter" data-level="3.6" data-path="integration-of-degs-ex-vivo-and-in-vivo.html"><a href="integration-of-degs-ex-vivo-and-in-vivo.html"><i class="fa fa-check"></i><b>3.6</b> Integration of DEGs ex vivo and in vivo</a></li>
<li class="chapter" data-level="3.7" data-path="gsea-analysis-for-crispri-drug-sensitivity.html"><a href="gsea-analysis-for-crispri-drug-sensitivity.html"><i class="fa fa-check"></i><b>3.7</b> GSEA analysis for CRISPRi drug sensitivity</a></li>
<li class="chapter" data-level="3.8" data-path="differential-expression-analysis-for-shpink1.html"><a href="differential-expression-analysis-for-shpink1.html"><i class="fa fa-check"></i><b>3.8</b> Differential Expression Analysis for shPINK1</a></li>
<li class="chapter" data-level="3.9" data-path="de-analysis-in-cell-lines.html"><a href="de-analysis-in-cell-lines.html"><i class="fa fa-check"></i><b>3.9</b> DE analysis in cell lines</a></li>
<li class="chapter" data-level="3.10" data-path="valcano-plots-molm13.html"><a href="valcano-plots-molm13.html"><i class="fa fa-check"></i><b>3.10</b> Valcano plots (MOLM13)</a></li>
<li class="chapter" data-level="3.11" data-path="valcano-plots-oa3.html"><a href="valcano-plots-oa3.html"><i class="fa fa-check"></i><b>3.11</b> Valcano plots (OA3)</a></li>
<li class="chapter" data-level="3.12" data-path="gsea-analysis-molm13.html"><a href="gsea-analysis-molm13.html"><i class="fa fa-check"></i><b>3.12</b> GSEA analysis (MOLM13)</a></li>
<li class="chapter" data-level="3.13" data-path="gsea-analysis-oa3.html"><a href="gsea-analysis-oa3.html"><i class="fa fa-check"></i><b>3.13</b> GSEA analysis (OA3)</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>4</b> Figure 3</a>
<ul>
<li class="chapter" data-level="4.1" data-path="benchmark-ml-methods-for-prediction.html"><a href="benchmark-ml-methods-for-prediction.html"><i class="fa fa-check"></i><b>4.1</b> Benchmark ML Methods for Prediction</a></li>
<li class="chapter" data-level="4.2" data-path="predict-class-labels-on-the-test-set-and-plot-roc.html"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html"><i class="fa fa-check"></i><b>4.2</b> Predict Class Labels on the Test Set and Plot ROC</a></li>
<li class="chapter" data-level="4.3" data-path="auc-and-auprc-heatmap-for-the-entire-and-test-sets.html"><a href="auc-and-auprc-heatmap-for-the-entire-and-test-sets.html"><i class="fa fa-check"></i><b>4.3</b> AUC and AUPRC Heatmap for the Entire and Test Sets</a></li>
<li class="chapter" data-level="4.4" data-path="model-performance-plot.html"><a href="model-performance-plot.html"><i class="fa fa-check"></i><b>4.4</b> Model Performance Plot</a></li>
<li class="chapter" data-level="4.5" data-path="feature-importance.html"><a href="feature-importance.html"><i class="fa fa-check"></i><b>4.5</b> Feature importance</a></li>
<li class="chapter" data-level="4.6" data-path="heatmap-for-confusion-matrices-defaut-cutoff-0.html"><a href="heatmap-for-confusion-matrices-defaut-cutoff-0.html"><i class="fa fa-check"></i><b>4.6</b> Heatmap for Confusion Matrices (defaut cutoff 0.5)</a></li>
<li class="chapter" data-level="4.7" data-path="barplot-for-predicted-response-0.html"><a href="barplot-for-predicted-response-0.html"><i class="fa fa-check"></i><b>4.7</b> Barplot for predicted Response (0.5)</a></li>
<li class="chapter" data-level="4.8" data-path="heatmap-for-confusion-matrices-optimal-cutoff-0.html"><a href="heatmap-for-confusion-matrices-optimal-cutoff-0.html"><i class="fa fa-check"></i><b>4.8</b> Heatmap for Confusion Matrices (optimal cutoff 0.630)</a></li>
<li class="chapter" data-level="4.9" data-path="barplot-for-predicted-response-0.html"><a href="barplot-for-predicted-response-0.html#barplot-for-predicted-response-0.630"><i class="fa fa-check"></i><b>4.9</b> Barplot for predicted Response (0.630)</a></li>
<li class="chapter" data-level="4.10" data-path="density-plot-for-response-probability.html"><a href="density-plot-for-response-probability.html"><i class="fa fa-check"></i><b>4.10</b> Density Plot for Response Probability</a></li>
<li class="chapter" data-level="4.11" data-path="violin-and-boxplots-for-response-probability-rf8.html"><a href="violin-and-boxplots-for-response-probability-rf8.html"><i class="fa fa-check"></i><b>4.11</b> Violin and Boxplots for Response Probability (RF8)</a></li>
<li class="chapter" data-level="4.12" data-path="barplot-for-response.html"><a href="barplot-for-response.html"><i class="fa fa-check"></i><b>4.12</b> Barplot for Response</a></li>
<li class="chapter" data-level="4.13" data-path="rf8-for-mdr-status.html"><a href="rf8-for-mdr-status.html"><i class="fa fa-check"></i><b>4.13</b> RF8 for MDR status</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>5</b> Figure 4</a>
<ul>
<li class="chapter" data-level="5.1" data-path="validation-of-rf8-model-rjaml-cohort-2.html"><a href="validation-of-rf8-model-rjaml-cohort-2.html"><i class="fa fa-check"></i><b>5.1</b> Validation of RF8 model (RJAML-Cohort 2)</a></li>
<li class="chapter" data-level="5.2" data-path="validation-of-rf8-model-fpmtb.html"><a href="validation-of-rf8-model-fpmtb.html"><i class="fa fa-check"></i><b>5.2</b> Validation of RF8 model (FPMTB)</a></li>
<li class="chapter" data-level="5.3" data-path="validation-of-rf8-model-beataml.html"><a href="validation-of-rf8-model-beataml.html"><i class="fa fa-check"></i><b>5.3</b> Validation of RF8 model (BeatAML)</a></li>
<li class="chapter" data-level="5.4" data-path="violin-and-boxplots-for-response-probability-rf8-1.html"><a href="violin-and-boxplots-for-response-probability-rf8-1.html"><i class="fa fa-check"></i><b>5.4</b> Violin and Boxplots for Response Probability (RF8)</a></li>
<li class="chapter" data-level="5.5" data-path="rf8-score-among-clusters.html"><a href="rf8-score-among-clusters.html"><i class="fa fa-check"></i><b>5.5</b> RF8 score among clusters</a></li>
<li class="chapter" data-level="5.6" data-path="integration-of-ex-vivo-drug-response-in-beataml.html"><a href="integration-of-ex-vivo-drug-response-in-beataml.html"><i class="fa fa-check"></i><b>5.6</b> Integration of ex vivo drug-response in BeatAML</a></li>
<li class="chapter" data-level="5.7" data-path="beataml-waves-1-4-heatmap-based-on-drug-profiling.html"><a href="beataml-waves-1-4-heatmap-based-on-drug-profiling.html"><i class="fa fa-check"></i><b>5.7</b> BeatAML Waves 1-4 heatmap based on drug profiling</a></li>
<li class="chapter" data-level="5.8" data-path="boxplot-for-drug-sensitivitity-between-resistant-and-sensitive.html"><a href="boxplot-for-drug-sensitivitity-between-resistant-and-sensitive.html"><i class="fa fa-check"></i><b>5.8</b> Boxplot for drug sensitivitity between resistant and sensitive</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>6</b> Figure 5</a>
<ul>
<li class="chapter" data-level="6.1" data-path="integration-with-aml-genomics.html"><a href="integration-with-aml-genomics.html"><i class="fa fa-check"></i><b>6.1</b> Integration with AML genomics</a></li>
<li class="chapter" data-level="6.2" data-path="mutation-pairs.html"><a href="mutation-pairs.html"><i class="fa fa-check"></i><b>6.2</b> Mutation pairs</a></li>
<li class="chapter" data-level="6.3" data-path="complexheatmap-for-mutation.html"><a href="complexheatmap-for-mutation.html"><i class="fa fa-check"></i><b>6.3</b> Complexheatmap for mutation</a></li>
<li class="chapter" data-level="6.4" data-path="fisher-test.html"><a href="fisher-test.html"><i class="fa fa-check"></i><b>6.4</b> Fisher test</a></li>
<li class="chapter" data-level="6.5" data-path="barplot-in-rjaml-cohort.html"><a href="barplot-in-rjaml-cohort.html"><i class="fa fa-check"></i><b>6.5</b> Barplot in RJAML cohort</a></li>
<li class="chapter" data-level="6.6" data-path="table-one-for-multi-cohorts.html"><a href="table-one-for-multi-cohorts.html"><i class="fa fa-check"></i><b>6.6</b> Table one for multi-cohorts</a></li>
<li class="chapter" data-level="6.7" data-path="barplot-for-real-response.html"><a href="barplot-for-real-response.html"><i class="fa fa-check"></i><b>6.7</b> Barplot for real response</a></li>
<li class="chapter" data-level="6.8" data-path="multivariable-logisticreg.html"><a href="multivariable-logisticreg.html"><i class="fa fa-check"></i><b>6.8</b> Multivariable LogisticReg</a></li>
<li class="chapter" data-level="6.9" data-path="identification-of-rf8-correlated-genes.html"><a href="identification-of-rf8-correlated-genes.html"><i class="fa fa-check"></i><b>6.9</b> Identification of RF8-correlated genes</a></li>
<li class="chapter" data-level="6.10" data-path="differential-expression-analysis-for-mut-vs.-wt.html"><a href="differential-expression-analysis-for-mut-vs.-wt.html"><i class="fa fa-check"></i><b>6.10</b> Differential Expression Analysis for Mut vs. WT</a></li>
<li class="chapter" data-level="6.11" data-path="venn-diagram-for-target-genes.html"><a href="venn-diagram-for-target-genes.html"><i class="fa fa-check"></i><b>6.11</b> Venn Diagram for target genes</a></li>
<li class="chapter" data-level="6.12" data-path="pathway-overlap-analysis-of-genes.html"><a href="pathway-overlap-analysis-of-genes.html"><i class="fa fa-check"></i><b>6.12</b> Pathway overlap analysis of genes</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="figure-6.html"><a href="figure-6.html"><i class="fa fa-check"></i><b>7</b> Figure 6</a>
<ul>
<li class="chapter" data-level="7.1" data-path="relationship-between-rf8-score-vs.-venaza-response-probability.html"><a href="relationship-between-rf8-score-vs.-venaza-response-probability.html"><i class="fa fa-check"></i><b>7.1</b> Relationship between RF8 score vs. VEN/AZA response probability</a></li>
<li class="chapter" data-level="7.2" data-path="monotonic-response-prediction-plot.html"><a href="monotonic-response-prediction-plot.html"><i class="fa fa-check"></i><b>7.2</b> Monotonic Response Prediction Plot</a></li>
<li class="chapter" data-level="7.3" data-path="survival-plot.html"><a href="survival-plot.html"><i class="fa fa-check"></i><b>7.3</b> Survival plot</a></li>
<li class="chapter" data-level="7.4" data-path="brier-score-analysis.html"><a href="brier-score-analysis.html"><i class="fa fa-check"></i><b>7.4</b> Brier Score Analysis</a></li>
<li class="chapter" data-level="7.5" data-path="c-index-comparison.html"><a href="c-index-comparison.html"><i class="fa fa-check"></i><b>7.5</b> C-index Comparison</a></li>
<li class="chapter" data-level="7.6" data-path="km-plots-for-rjaml-cohort-2.html"><a href="km-plots-for-rjaml-cohort-2.html"><i class="fa fa-check"></i><b>7.6</b> KM-Plots for RJAML Cohort 2</a></li>
<li class="chapter" data-level="7.7" data-path="barplot-for-crcri.html"><a href="barplot-for-crcri.html"><i class="fa fa-check"></i><b>7.7</b> Barplot for CR/CRi</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="differential-expression-analysis-for-mut-vs.-wt" class="section level2 hasAnchor" number="6.10">
<h2><span class="header-section-number">6.10</span> Differential Expression Analysis for Mut vs. WT<a href="differential-expression-analysis-for-mut-vs.-wt.html#differential-expression-analysis-for-mut-vs.-wt" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb169"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb169-1"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-1" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################</span></span>
<span id="cb169-2"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-2" aria-hidden="true" tabindex="-1"></a><span class="do">### Differential Expression Analysis for Mut vs. WT</span></span>
<span id="cb169-3"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-3" aria-hidden="true" tabindex="-1"></a><span class="do">##########################################################################</span></span>
<span id="cb169-4"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb169-5"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-5" aria-hidden="true" tabindex="-1"></a><span class="co"># # Filter data for relevant samples</span></span>
<span id="cb169-6"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-6" aria-hidden="true" tabindex="-1"></a><span class="co"># source(&quot;RunDESeq2.R&quot;)</span></span>
<span id="cb169-7"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-7" aria-hidden="true" tabindex="-1"></a><span class="co"># mat &lt;- read.table(&quot;../02.data/ID_matched_count_matrix.txt&quot;, header = T, row.names = 1)</span></span>
<span id="cb169-8"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-8" aria-hidden="true" tabindex="-1"></a><span class="co"># mat &lt;- mat[, -c(1:2)]</span></span>
<span id="cb169-9"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-9" aria-hidden="true" tabindex="-1"></a><span class="co"># tpm &lt;- read.table(&quot;../02.data/ID_matched_TPM_matrix.txt&quot;, header = TRUE, check.names = FALSE)</span></span>
<span id="cb169-10"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-10" aria-hidden="true" tabindex="-1"></a><span class="co"># sample_info &lt;- read.csv(&quot;../02.data/rjaml_rf8_scores.csv&quot;, header = TRUE, check.names = FALSE)</span></span>
<span id="cb169-11"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-11" aria-hidden="true" tabindex="-1"></a><span class="co"># tpm &lt;- tpm[, c(&quot;ID&quot;, &quot;Symbol&quot;, &quot;Type&quot;, as.character(sample_info$PatientID))]</span></span>
<span id="cb169-12"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-12" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb169-13"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-13" aria-hidden="true" tabindex="-1"></a><span class="co"># tpm &lt;- subset(tpm, Type %in% c(&quot;protein_coding&quot;, &quot;miRNA&quot;, &quot;lincRNA&quot;, &quot;antisense&quot;,</span></span>
<span id="cb169-14"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-14" aria-hidden="true" tabindex="-1"></a><span class="co">#                                &quot;processed_transcript&quot;,</span></span>
<span id="cb169-15"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-15" aria-hidden="true" tabindex="-1"></a><span class="co">#                                &quot;transcribed_processed_pseudogene&quot;, &quot;transcribed_unitary_pseudogene&quot;,</span></span>
<span id="cb169-16"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-16" aria-hidden="true" tabindex="-1"></a><span class="co">#                                &quot;transcribed_unprocessed_pseudogene&quot;,&quot;macro_lncRNA&quot;, &quot;TEC&quot;,</span></span>
<span id="cb169-17"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-17" aria-hidden="true" tabindex="-1"></a><span class="co">#                                &quot;3prime_overlapping_ncRNA&quot;,&quot;sense_intronic&quot;, &quot;sense_overlapping&quot;,</span></span>
<span id="cb169-18"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-18" aria-hidden="true" tabindex="-1"></a><span class="co">#                                &quot;unprocessed_pseudogene&quot;))</span></span>
<span id="cb169-19"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-19" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb169-20"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-20" aria-hidden="true" tabindex="-1"></a><span class="co"># tpm &lt;- tpm[which(rowMeans(tpm[, -c(1:3)]) &gt; 0.5), ]</span></span>
<span id="cb169-21"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-21" aria-hidden="true" tabindex="-1"></a><span class="co"># tpm &lt;- tpm[!duplicated(tpm$Symbol), ]</span></span>
<span id="cb169-22"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-22" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb169-23"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-23" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">###</span><span class="co"> TP53 Mutation vs. WT</span></span>
<span id="cb169-24"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-24" aria-hidden="true" tabindex="-1"></a><span class="co"># sample_info &lt;- sample_info[order(sample_info$TP53), ]</span></span>
<span id="cb169-25"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-25" aria-hidden="true" tabindex="-1"></a><span class="co"># table(sample_info$TP53)</span></span>
<span id="cb169-26"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-26" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb169-27"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-27" aria-hidden="true" tabindex="-1"></a><span class="co"># mat &lt;- mat[tpm$ID, sample_info$PatientID]</span></span>
<span id="cb169-28"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-28" aria-hidden="true" tabindex="-1"></a><span class="co"># tpm &lt;- tpm[, c(&quot;ID&quot;, &quot;Symbol&quot;, &quot;Type&quot;, as.character(sample_info$PatientID))]</span></span>
<span id="cb169-29"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-29" aria-hidden="true" tabindex="-1"></a><span class="co"># # Run DESeq2 analysis</span></span>
<span id="cb169-30"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-30" aria-hidden="true" tabindex="-1"></a><span class="co"># RunDESeq2(count_mat = mat, n.cont = 100, n.treat = 10,</span></span>
<span id="cb169-31"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-31" aria-hidden="true" tabindex="-1"></a><span class="co">#           prefix = &quot;../03.out/figure5/DESeq2_out_TP53_Mut_WT&quot;,</span></span>
<span id="cb169-32"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-32" aria-hidden="true" tabindex="-1"></a><span class="co">#           sort.p = FALSE,</span></span>
<span id="cb169-33"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-33" aria-hidden="true" tabindex="-1"></a><span class="co">#           merge.normalized = TRUE,</span></span>
<span id="cb169-34"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-34" aria-hidden="true" tabindex="-1"></a><span class="co">#           normalized_mat = tpm)</span></span>
<span id="cb169-35"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-35" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb169-36"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-36" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">###</span><span class="co"> DNMT3A Mutation vs. WT</span></span>
<span id="cb169-37"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-37" aria-hidden="true" tabindex="-1"></a><span class="co"># sample_info &lt;- sample_info[order(sample_info$DNMT3A), ]</span></span>
<span id="cb169-38"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-38" aria-hidden="true" tabindex="-1"></a><span class="co"># table(sample_info$DNMT3A)</span></span>
<span id="cb169-39"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-39" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb169-40"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-40" aria-hidden="true" tabindex="-1"></a><span class="co"># mat &lt;- mat[tpm$ID, sample_info$PatientID]</span></span>
<span id="cb169-41"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-41" aria-hidden="true" tabindex="-1"></a><span class="co"># tpm &lt;- tpm[, c(&quot;ID&quot;, &quot;Symbol&quot;, &quot;Type&quot;, as.character(sample_info$PatientID))]</span></span>
<span id="cb169-42"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-42" aria-hidden="true" tabindex="-1"></a><span class="co"># # Run DESeq2 analysis</span></span>
<span id="cb169-43"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-43" aria-hidden="true" tabindex="-1"></a><span class="co"># RunDESeq2(count_mat = mat, n.cont = 83, n.treat = 27,</span></span>
<span id="cb169-44"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-44" aria-hidden="true" tabindex="-1"></a><span class="co">#           prefix = &quot;../03.out/figure5/DESeq2_out_DNMT3A_Mut_WT&quot;,</span></span>
<span id="cb169-45"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-45" aria-hidden="true" tabindex="-1"></a><span class="co">#           sort.p = FALSE,</span></span>
<span id="cb169-46"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-46" aria-hidden="true" tabindex="-1"></a><span class="co">#           merge.normalized = TRUE,</span></span>
<span id="cb169-47"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-47" aria-hidden="true" tabindex="-1"></a><span class="co">#           normalized_mat = tpm)</span></span>
<span id="cb169-48"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-48" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb169-49"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-49" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">###</span><span class="co"> IDH2 Mutation vs. WT</span></span>
<span id="cb169-50"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-50" aria-hidden="true" tabindex="-1"></a><span class="co"># sample_info &lt;- sample_info[order(sample_info$IDH2), ]</span></span>
<span id="cb169-51"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-51" aria-hidden="true" tabindex="-1"></a><span class="co"># table(sample_info$IDH2)</span></span>
<span id="cb169-52"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-52" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb169-53"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-53" aria-hidden="true" tabindex="-1"></a><span class="co"># mat &lt;- mat[tpm$ID, sample_info$PatientID]</span></span>
<span id="cb169-54"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-54" aria-hidden="true" tabindex="-1"></a><span class="co"># tpm &lt;- tpm[, c(&quot;ID&quot;, &quot;Symbol&quot;, &quot;Type&quot;, as.character(sample_info$PatientID))]</span></span>
<span id="cb169-55"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-55" aria-hidden="true" tabindex="-1"></a><span class="co"># # Run DESeq2 analysis</span></span>
<span id="cb169-56"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-56" aria-hidden="true" tabindex="-1"></a><span class="co"># RunDESeq2(count_mat = mat, n.cont = 84, n.treat = 26,</span></span>
<span id="cb169-57"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-57" aria-hidden="true" tabindex="-1"></a><span class="co">#           prefix = &quot;../03.out/figure5/DESeq2_out_IDH2_Mut_WT&quot;,</span></span>
<span id="cb169-58"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-58" aria-hidden="true" tabindex="-1"></a><span class="co">#           sort.p = FALSE,</span></span>
<span id="cb169-59"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-59" aria-hidden="true" tabindex="-1"></a><span class="co">#           merge.normalized = TRUE,</span></span>
<span id="cb169-60"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-60" aria-hidden="true" tabindex="-1"></a><span class="co">#           normalized_mat = tpm)</span></span>
<span id="cb169-61"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-61" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb169-62"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-62" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">###</span><span class="co"> CEBPA Mutation vs. WT</span></span>
<span id="cb169-63"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-63" aria-hidden="true" tabindex="-1"></a><span class="co"># sample_info &lt;- sample_info[order(sample_info$CEBPA), ]</span></span>
<span id="cb169-64"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-64" aria-hidden="true" tabindex="-1"></a><span class="co"># table(sample_info$CEBPA)</span></span>
<span id="cb169-65"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-65" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb169-66"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-66" aria-hidden="true" tabindex="-1"></a><span class="co"># mat &lt;- mat[tpm$ID, sample_info$PatientID]</span></span>
<span id="cb169-67"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-67" aria-hidden="true" tabindex="-1"></a><span class="co"># tpm &lt;- tpm[, c(&quot;ID&quot;, &quot;Symbol&quot;, &quot;Type&quot;, as.character(sample_info$PatientID))]</span></span>
<span id="cb169-68"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-68" aria-hidden="true" tabindex="-1"></a><span class="co"># # Run DESeq2 analysis</span></span>
<span id="cb169-69"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-69" aria-hidden="true" tabindex="-1"></a><span class="co"># RunDESeq2(count_mat = mat, n.cont = 100, n.treat = 10,</span></span>
<span id="cb169-70"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-70" aria-hidden="true" tabindex="-1"></a><span class="co">#           prefix = &quot;../03.out/figure5/DESeq2_out_CEBPA_Mut_WT&quot;,</span></span>
<span id="cb169-71"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-71" aria-hidden="true" tabindex="-1"></a><span class="co">#           sort.p = FALSE,</span></span>
<span id="cb169-72"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-72" aria-hidden="true" tabindex="-1"></a><span class="co">#           merge.normalized = TRUE,</span></span>
<span id="cb169-73"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb169-73" aria-hidden="true" tabindex="-1"></a><span class="co">#           normalized_mat = tpm)</span></span></code></pre></div>
<div class="sourceCode" id="cb170"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb170-1"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb170-2"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-2" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;plotVolcanoV3.R&quot;</span>)</span>
<span id="cb170-3"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-4"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-4" aria-hidden="true" tabindex="-1"></a><span class="do">##### TP53</span></span>
<span id="cb170-5"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-5" aria-hidden="true" tabindex="-1"></a>de_results <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../03.out/figure5/DESeq2_out_TP53_Mut_WT_with_normalized_mat.txt&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb170-6"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-6" aria-hidden="true" tabindex="-1"></a>de_results_filtered <span class="ot">&lt;-</span> de_results[, <span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>]</span>
<span id="cb170-7"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-7" aria-hidden="true" tabindex="-1"></a>degs_tp53 <span class="ot">&lt;-</span> <span class="fu">subset</span>(de_results_filtered, <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="fl">0.25</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb170-8"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-9"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Cap extreme values</span></span>
<span id="cb170-10"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-10" aria-hidden="true" tabindex="-1"></a>de_results_filtered<span class="sc">$</span>log2FoldChange[de_results_filtered<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">5</span>] <span class="ot">&lt;-</span> <span class="sc">-</span><span class="dv">5</span></span>
<span id="cb170-11"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-11" aria-hidden="true" tabindex="-1"></a>de_results_filtered<span class="sc">$</span>log2FoldChange[de_results_filtered<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">5</span>] <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb170-12"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-12" aria-hidden="true" tabindex="-1"></a>de_results_filtered<span class="sc">$</span>padj[de_results_filtered<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">1e-6</span>] <span class="ot">&lt;-</span> <span class="fl">1e-6</span></span>
<span id="cb170-13"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-14"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate volcano plot</span></span>
<span id="cb170-15"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-15" aria-hidden="true" tabindex="-1"></a>volcano_plot1 <span class="ot">&lt;-</span> <span class="fu">plotVolcano</span>(<span class="at">mat =</span> de_results_filtered, <span class="at">gene.col =</span> <span class="st">&quot;Symbol&quot;</span>, <span class="at">x.col =</span> <span class="st">&quot;log2FoldChange&quot;</span>, <span class="at">y.col =</span> <span class="st">&quot;padj&quot;</span>, </span>
<span id="cb170-16"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-16" aria-hidden="true" tabindex="-1"></a>                            <span class="at">labx =</span> <span class="st">&quot;log2FoldChange&quot;</span>, <span class="at">laby =</span> <span class="st">&quot;-Log10FDR&quot;</span>,</span>
<span id="cb170-17"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-17" aria-hidden="true" tabindex="-1"></a>                            <span class="at">x_cut1 =</span> <span class="fl">0.5</span>, <span class="at">x_cut2 =</span> <span class="dv">1</span>, <span class="at">y_cut1 =</span> <span class="fl">0.05</span>, <span class="at">y_cut2 =</span> <span class="fl">0.01</span>, <span class="at">x.lim =</span> <span class="dv">5</span>, <span class="at">y.lim =</span> <span class="dv">6</span>, </span>
<span id="cb170-18"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-18" aria-hidden="true" tabindex="-1"></a>                            <span class="at">label =</span> <span class="cn">FALSE</span>, <span class="at">title =</span> <span class="st">&quot;TP53 (Mut vs. WT)&quot;</span>, <span class="at">selected_genes =</span> <span class="st">&quot;NA&quot;</span>) <span class="sc">+</span> <span class="fu">theme_classic</span>()</span>
<span id="cb170-19"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-20"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-21"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-21" aria-hidden="true" tabindex="-1"></a><span class="do">##### DNMT3A</span></span>
<span id="cb170-22"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-22" aria-hidden="true" tabindex="-1"></a>de_results <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../03.out/figure5/DESeq2_out_DNMT3A_Mut_WT_with_normalized_mat.txt&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb170-23"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-23" aria-hidden="true" tabindex="-1"></a>de_results_filtered <span class="ot">&lt;-</span> de_results[, <span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>]</span>
<span id="cb170-24"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-24" aria-hidden="true" tabindex="-1"></a>degs_dnmt3a <span class="ot">&lt;-</span> <span class="fu">subset</span>(de_results_filtered, <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="fl">0.25</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb170-25"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-26"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Cap extreme values</span></span>
<span id="cb170-27"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-27" aria-hidden="true" tabindex="-1"></a>de_results_filtered<span class="sc">$</span>log2FoldChange[de_results_filtered<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="sc">-</span><span class="dv">4</span></span>
<span id="cb170-28"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-28" aria-hidden="true" tabindex="-1"></a>de_results_filtered<span class="sc">$</span>log2FoldChange[de_results_filtered<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="dv">4</span></span>
<span id="cb170-29"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-29" aria-hidden="true" tabindex="-1"></a>de_results_filtered<span class="sc">$</span>padj[de_results_filtered<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">1e-6</span>] <span class="ot">&lt;-</span> <span class="fl">1e-6</span></span>
<span id="cb170-30"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-31"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate volcano plot</span></span>
<span id="cb170-32"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-32" aria-hidden="true" tabindex="-1"></a>volcano_plot2 <span class="ot">&lt;-</span> <span class="fu">plotVolcano</span>(<span class="at">mat =</span> de_results_filtered, <span class="at">gene.col =</span> <span class="st">&quot;Symbol&quot;</span>, <span class="at">x.col =</span> <span class="st">&quot;log2FoldChange&quot;</span>, <span class="at">y.col =</span> <span class="st">&quot;padj&quot;</span>, </span>
<span id="cb170-33"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-33" aria-hidden="true" tabindex="-1"></a>                            <span class="at">labx =</span> <span class="st">&quot;log2FoldChange&quot;</span>, <span class="at">laby =</span> <span class="st">&quot;-Log10FDR&quot;</span>,</span>
<span id="cb170-34"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-34" aria-hidden="true" tabindex="-1"></a>                            <span class="at">x_cut1 =</span> <span class="fl">0.5</span>, <span class="at">x_cut2 =</span> <span class="dv">1</span>, <span class="at">y_cut1 =</span> <span class="fl">0.05</span>, <span class="at">y_cut2 =</span> <span class="fl">0.01</span>, <span class="at">x.lim =</span> <span class="dv">4</span>, <span class="at">y.lim =</span> <span class="dv">6</span>, </span>
<span id="cb170-35"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-35" aria-hidden="true" tabindex="-1"></a>                            <span class="at">label =</span> <span class="cn">FALSE</span>, <span class="at">title =</span> <span class="st">&quot;DNMT3A (Mut vs. WT)&quot;</span>, <span class="at">selected_genes =</span> <span class="st">&quot;NA&quot;</span>) <span class="sc">+</span> <span class="fu">theme_classic</span>()</span>
<span id="cb170-36"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-37"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-37" aria-hidden="true" tabindex="-1"></a><span class="do">##### IDH2</span></span>
<span id="cb170-38"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-38" aria-hidden="true" tabindex="-1"></a>de_results <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../03.out/figure5/DESeq2_out_IDH2_Mut_WT_with_normalized_mat.txt&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb170-39"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-39" aria-hidden="true" tabindex="-1"></a>de_results_filtered <span class="ot">&lt;-</span> de_results[, <span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>]</span>
<span id="cb170-40"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-40" aria-hidden="true" tabindex="-1"></a>degs_idh2 <span class="ot">&lt;-</span> <span class="fu">subset</span>(de_results_filtered, <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="fl">0.25</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb170-41"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-42"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-42" aria-hidden="true" tabindex="-1"></a><span class="co"># Cap extreme values</span></span>
<span id="cb170-43"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-43" aria-hidden="true" tabindex="-1"></a>de_results_filtered<span class="sc">$</span>log2FoldChange[de_results_filtered<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="sc">-</span><span class="dv">4</span></span>
<span id="cb170-44"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-44" aria-hidden="true" tabindex="-1"></a>de_results_filtered<span class="sc">$</span>log2FoldChange[de_results_filtered<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="dv">4</span></span>
<span id="cb170-45"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-45" aria-hidden="true" tabindex="-1"></a>de_results_filtered<span class="sc">$</span>padj[de_results_filtered<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">1e-4</span>] <span class="ot">&lt;-</span> <span class="fl">1e-4</span></span>
<span id="cb170-46"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-47"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-47" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate volcano plot</span></span>
<span id="cb170-48"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-48" aria-hidden="true" tabindex="-1"></a>volcano_plot3 <span class="ot">&lt;-</span> <span class="fu">plotVolcano</span>(<span class="at">mat =</span> de_results_filtered, <span class="at">gene.col =</span> <span class="st">&quot;Symbol&quot;</span>, <span class="at">x.col =</span> <span class="st">&quot;log2FoldChange&quot;</span>, <span class="at">y.col =</span> <span class="st">&quot;padj&quot;</span>, </span>
<span id="cb170-49"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-49" aria-hidden="true" tabindex="-1"></a>                             <span class="at">labx =</span> <span class="st">&quot;log2FoldChange&quot;</span>, <span class="at">laby =</span> <span class="st">&quot;-Log10FDR&quot;</span>,</span>
<span id="cb170-50"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-50" aria-hidden="true" tabindex="-1"></a>                             <span class="at">x_cut1 =</span> <span class="fl">0.5</span>, <span class="at">x_cut2 =</span> <span class="dv">1</span>, <span class="at">y_cut1 =</span> <span class="fl">0.05</span>, <span class="at">y_cut2 =</span> <span class="fl">0.01</span>, <span class="at">x.lim =</span> <span class="dv">4</span>, <span class="at">y.lim =</span> <span class="dv">4</span>, </span>
<span id="cb170-51"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-51" aria-hidden="true" tabindex="-1"></a>                             <span class="at">label =</span> <span class="cn">FALSE</span>, <span class="at">title =</span> <span class="st">&quot;IDH2 (Mut vs. WT)&quot;</span>, <span class="at">selected_genes =</span> <span class="st">&quot;NA&quot;</span>) <span class="sc">+</span> <span class="fu">theme_classic</span>()</span>
<span id="cb170-52"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-53"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-53" aria-hidden="true" tabindex="-1"></a><span class="do">##### CEBPA</span></span>
<span id="cb170-54"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-54" aria-hidden="true" tabindex="-1"></a>de_results <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;../03.out/figure5/DESeq2_out_CEBPA_Mut_WT_with_normalized_mat.txt&quot;</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb170-55"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-55" aria-hidden="true" tabindex="-1"></a>de_results_filtered <span class="ot">&lt;-</span> de_results[, <span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>]</span>
<span id="cb170-56"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-56" aria-hidden="true" tabindex="-1"></a>degs_cebpa <span class="ot">&lt;-</span> <span class="fu">subset</span>(de_results_filtered, <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> <span class="fl">0.25</span> <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb170-57"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-58"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-58" aria-hidden="true" tabindex="-1"></a><span class="co"># Cap extreme values</span></span>
<span id="cb170-59"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-59" aria-hidden="true" tabindex="-1"></a>de_results_filtered<span class="sc">$</span>log2FoldChange[de_results_filtered<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">5</span>] <span class="ot">&lt;-</span> <span class="sc">-</span><span class="dv">5</span></span>
<span id="cb170-60"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-60" aria-hidden="true" tabindex="-1"></a>de_results_filtered<span class="sc">$</span>log2FoldChange[de_results_filtered<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">5</span>] <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb170-61"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-61" aria-hidden="true" tabindex="-1"></a>de_results_filtered<span class="sc">$</span>padj[de_results_filtered<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">1e-6</span>] <span class="ot">&lt;-</span> <span class="fl">1e-6</span></span>
<span id="cb170-62"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-63"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-63" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate volcano plot</span></span>
<span id="cb170-64"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-64" aria-hidden="true" tabindex="-1"></a>volcano_plot4 <span class="ot">&lt;-</span> <span class="fu">plotVolcano</span>(<span class="at">mat =</span> de_results_filtered, <span class="at">gene.col =</span> <span class="st">&quot;Symbol&quot;</span>, <span class="at">x.col =</span> <span class="st">&quot;log2FoldChange&quot;</span>, <span class="at">y.col =</span> <span class="st">&quot;padj&quot;</span>, </span>
<span id="cb170-65"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-65" aria-hidden="true" tabindex="-1"></a>                             <span class="at">labx =</span> <span class="st">&quot;log2FoldChange&quot;</span>, <span class="at">laby =</span> <span class="st">&quot;-Log10FDR&quot;</span>,</span>
<span id="cb170-66"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-66" aria-hidden="true" tabindex="-1"></a>                             <span class="at">x_cut1 =</span> <span class="fl">0.5</span>, <span class="at">x_cut2 =</span> <span class="dv">1</span>, <span class="at">y_cut1 =</span> <span class="fl">0.05</span>, <span class="at">y_cut2 =</span> <span class="fl">0.01</span>, <span class="at">x.lim =</span> <span class="dv">4</span>, <span class="at">y.lim =</span> <span class="dv">4</span>, </span>
<span id="cb170-67"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-67" aria-hidden="true" tabindex="-1"></a>                             <span class="at">label =</span> <span class="cn">FALSE</span>, <span class="at">title =</span> <span class="st">&quot;CEBPA (Mut vs. WT)&quot;</span>, <span class="at">selected_genes =</span> <span class="st">&quot;NA&quot;</span>) <span class="sc">+</span> <span class="fu">theme_classic</span>()</span>
<span id="cb170-68"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-69"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-69" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> volcano_plot1<span class="sc">|</span>volcano_plot2<span class="sc">|</span>volcano_plot3<span class="sc">|</span>volcano_plot4</span>
<span id="cb170-70"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb170-70" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(plots)</span></code></pre></div>
<p><img src="figures/unnamed-chunk-77-1.png" width="672" /></p>
<div class="sourceCode" id="cb171"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb171-1"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb171-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save volcano plot</span></span>
<span id="cb171-2"><a href="differential-expression-analysis-for-mut-vs.-wt.html#cb171-2" aria-hidden="true" tabindex="-1"></a><span class="co"># ggsave(plots, filename = &quot;../03.out/figure5/DESeq2_valcanoplot.pdf&quot;, width = 12, height = 3)</span></span></code></pre></div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="identification-of-rf8-correlated-genes.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="venn-diagram-for-target-genes.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/05-results5.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
