<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6.9 Identification of RF8-correlated genes | Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning</title>
  <meta name="description" content="This is a book-style code sharing." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="6.9 Identification of RF8-correlated genes | Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a book-style code sharing." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6.9 Identification of RF8-correlated genes | Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning" />
  
  <meta name="twitter:description" content="This is a book-style code sharing." />
  

<meta name="author" content="Peng Jin, Qiqi Jin, Kankan Wang" />


<meta name="date" content="2025-07-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="multivariable-logisticreg.html"/>
<link rel="next" href="differential-expression-analysis-for-mut-vs.-wt.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Background</a></li>
<li class="chapter" data-level="2" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>2</b> Figure 1</a>
<ul>
<li class="chapter" data-level="2.1" data-path="process-the-data-from-functional_precision_medicine_tumor_board_aml.html"><a href="process-the-data-from-functional_precision_medicine_tumor_board_aml.html"><i class="fa fa-check"></i><b>2.1</b> Process the data from Functional_Precision_Medicine_Tumor_Board_AML</a></li>
<li class="chapter" data-level="2.2" data-path="functional_precision_medicine_tumor_board_aml.html"><a href="functional_precision_medicine_tumor_board_aml.html"><i class="fa fa-check"></i><b>2.2</b> Functional_Precision_Medicine_Tumor_Board_AML</a></li>
<li class="chapter" data-level="2.3" data-path="boxplot-for-drug-sensitivity-between-resistant-and-sensitive.html"><a href="boxplot-for-drug-sensitivity-between-resistant-and-sensitive.html"><i class="fa fa-check"></i><b>2.3</b> Boxplot for drug sensitivity between resistant and sensitive</a></li>
<li class="chapter" data-level="2.4" data-path="de-analysis-functional_medicine.html"><a href="de-analysis-functional_medicine.html"><i class="fa fa-check"></i><b>2.4</b> DE analysis Functional_medicine</a></li>
<li class="chapter" data-level="2.5" data-path="valcanoplot-plot-ex-vivo.html"><a href="valcanoplot-plot-ex-vivo.html"><i class="fa fa-check"></i><b>2.5</b> Valcanoplot plot (Ex vivo)</a></li>
<li class="chapter" data-level="2.6" data-path="heatmap-for-representative-de-genes.html"><a href="heatmap-for-representative-de-genes.html"><i class="fa fa-check"></i><b>2.6</b> Heatmap for representative DE genes</a></li>
<li class="chapter" data-level="2.7" data-path="cibersortx-results-visualization-ex-vivo.html"><a href="cibersortx-results-visualization-ex-vivo.html"><i class="fa fa-check"></i><b>2.7</b> CIBERSORTx Results Visualization (ex vivo)</a></li>
<li class="chapter" data-level="2.8" data-path="cibersortx-results-visualization-in-vivo.html"><a href="cibersortx-results-visualization-in-vivo.html"><i class="fa fa-check"></i><b>2.8</b> CIBERSORTx Results Visualization (in vivo)</a></li>
<li class="chapter" data-level="2.9" data-path="ssgsea-analysis-in-vivo.html"><a href="ssgsea-analysis-in-vivo.html"><i class="fa fa-check"></i><b>2.9</b> ssGSEA analysis in vivo</a></li>
<li class="chapter" data-level="2.10" data-path="go-enrichment-dot-plot.html"><a href="go-enrichment-dot-plot.html"><i class="fa fa-check"></i><b>2.10</b> GO Enrichment Dot Plot</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>3</b> Figure 2</a>
<ul>
<li class="chapter" data-level="3.1" data-path="de-analysis-between-in-vivo-sensitive-and-resistant.html"><a href="de-analysis-between-in-vivo-sensitive-and-resistant.html"><i class="fa fa-check"></i><b>3.1</b> DE analysis between in vivo sensitive and resistant</a></li>
<li class="chapter" data-level="3.2" data-path="de-analysis-rjaml-azaven-cohort.html"><a href="de-analysis-rjaml-azaven-cohort.html"><i class="fa fa-check"></i><b>3.2</b> DE analysis RJAML AZA+VEN cohort</a></li>
<li class="chapter" data-level="3.3" data-path="valcanoplot-plot-ex-vivo-1.html"><a href="valcanoplot-plot-ex-vivo-1.html"><i class="fa fa-check"></i><b>3.3</b> Valcanoplot plot (Ex vivo)</a></li>
<li class="chapter" data-level="3.4" data-path="valcanoplot-plot-in-vivo.html"><a href="valcanoplot-plot-in-vivo.html"><i class="fa fa-check"></i><b>3.4</b> Valcanoplot plot (In vivo)</a></li>
<li class="chapter" data-level="3.5" data-path="venn-plot.html"><a href="venn-plot.html"><i class="fa fa-check"></i><b>3.5</b> Venn plot</a></li>
<li class="chapter" data-level="3.6" data-path="integration-of-degs-ex-vivo-and-in-vivo.html"><a href="integration-of-degs-ex-vivo-and-in-vivo.html"><i class="fa fa-check"></i><b>3.6</b> Integration of DEGs ex vivo and in vivo</a></li>
<li class="chapter" data-level="3.7" data-path="gsea-analysis-for-crispri-drug-sensitivity.html"><a href="gsea-analysis-for-crispri-drug-sensitivity.html"><i class="fa fa-check"></i><b>3.7</b> GSEA analysis for CRISPRi drug sensitivity</a></li>
<li class="chapter" data-level="3.8" data-path="differential-expression-analysis-for-shpink1.html"><a href="differential-expression-analysis-for-shpink1.html"><i class="fa fa-check"></i><b>3.8</b> Differential Expression Analysis for shPINK1</a></li>
<li class="chapter" data-level="3.9" data-path="de-analysis-in-cell-lines.html"><a href="de-analysis-in-cell-lines.html"><i class="fa fa-check"></i><b>3.9</b> DE analysis in cell lines</a></li>
<li class="chapter" data-level="3.10" data-path="valcano-plots-molm13.html"><a href="valcano-plots-molm13.html"><i class="fa fa-check"></i><b>3.10</b> Valcano plots (MOLM13)</a></li>
<li class="chapter" data-level="3.11" data-path="valcano-plots-oa3.html"><a href="valcano-plots-oa3.html"><i class="fa fa-check"></i><b>3.11</b> Valcano plots (OA3)</a></li>
<li class="chapter" data-level="3.12" data-path="gsea-analysis-molm13.html"><a href="gsea-analysis-molm13.html"><i class="fa fa-check"></i><b>3.12</b> GSEA analysis (MOLM13)</a></li>
<li class="chapter" data-level="3.13" data-path="gsea-analysis-oa3.html"><a href="gsea-analysis-oa3.html"><i class="fa fa-check"></i><b>3.13</b> GSEA analysis (OA3)</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>4</b> Figure 3</a>
<ul>
<li class="chapter" data-level="4.1" data-path="benchmark-ml-methods-for-prediction.html"><a href="benchmark-ml-methods-for-prediction.html"><i class="fa fa-check"></i><b>4.1</b> Benchmark ML Methods for Prediction</a></li>
<li class="chapter" data-level="4.2" data-path="predict-class-labels-on-the-test-set-and-plot-roc.html"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html"><i class="fa fa-check"></i><b>4.2</b> Predict Class Labels on the Test Set and Plot ROC</a></li>
<li class="chapter" data-level="4.3" data-path="auc-and-auprc-heatmap-for-the-entire-and-test-sets.html"><a href="auc-and-auprc-heatmap-for-the-entire-and-test-sets.html"><i class="fa fa-check"></i><b>4.3</b> AUC and AUPRC Heatmap for the Entire and Test Sets</a></li>
<li class="chapter" data-level="4.4" data-path="model-performance-plot.html"><a href="model-performance-plot.html"><i class="fa fa-check"></i><b>4.4</b> Model Performance Plot</a></li>
<li class="chapter" data-level="4.5" data-path="feature-importance.html"><a href="feature-importance.html"><i class="fa fa-check"></i><b>4.5</b> Feature importance</a></li>
<li class="chapter" data-level="4.6" data-path="heatmap-for-confusion-matrices-defaut-cutoff-0.html"><a href="heatmap-for-confusion-matrices-defaut-cutoff-0.html"><i class="fa fa-check"></i><b>4.6</b> Heatmap for Confusion Matrices (defaut cutoff 0.5)</a></li>
<li class="chapter" data-level="4.7" data-path="barplot-for-predicted-response-0.html"><a href="barplot-for-predicted-response-0.html"><i class="fa fa-check"></i><b>4.7</b> Barplot for predicted Response (0.5)</a></li>
<li class="chapter" data-level="4.8" data-path="heatmap-for-confusion-matrices-optimal-cutoff-0.html"><a href="heatmap-for-confusion-matrices-optimal-cutoff-0.html"><i class="fa fa-check"></i><b>4.8</b> Heatmap for Confusion Matrices (optimal cutoff 0.630)</a></li>
<li class="chapter" data-level="4.9" data-path="barplot-for-predicted-response-0.html"><a href="barplot-for-predicted-response-0.html#barplot-for-predicted-response-0.630"><i class="fa fa-check"></i><b>4.9</b> Barplot for predicted Response (0.630)</a></li>
<li class="chapter" data-level="4.10" data-path="density-plot-for-response-probability.html"><a href="density-plot-for-response-probability.html"><i class="fa fa-check"></i><b>4.10</b> Density Plot for Response Probability</a></li>
<li class="chapter" data-level="4.11" data-path="violin-and-boxplots-for-response-probability-rf8.html"><a href="violin-and-boxplots-for-response-probability-rf8.html"><i class="fa fa-check"></i><b>4.11</b> Violin and Boxplots for Response Probability (RF8)</a></li>
<li class="chapter" data-level="4.12" data-path="barplot-for-response.html"><a href="barplot-for-response.html"><i class="fa fa-check"></i><b>4.12</b> Barplot for Response</a></li>
<li class="chapter" data-level="4.13" data-path="rf8-for-mdr-status.html"><a href="rf8-for-mdr-status.html"><i class="fa fa-check"></i><b>4.13</b> RF8 for MDR status</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>5</b> Figure 4</a>
<ul>
<li class="chapter" data-level="5.1" data-path="validation-of-rf8-model-rjaml-cohort-2.html"><a href="validation-of-rf8-model-rjaml-cohort-2.html"><i class="fa fa-check"></i><b>5.1</b> Validation of RF8 model (RJAML-Cohort 2)</a></li>
<li class="chapter" data-level="5.2" data-path="validation-of-rf8-model-fpmtb.html"><a href="validation-of-rf8-model-fpmtb.html"><i class="fa fa-check"></i><b>5.2</b> Validation of RF8 model (FPMTB)</a></li>
<li class="chapter" data-level="5.3" data-path="validation-of-rf8-model-beataml.html"><a href="validation-of-rf8-model-beataml.html"><i class="fa fa-check"></i><b>5.3</b> Validation of RF8 model (BeatAML)</a></li>
<li class="chapter" data-level="5.4" data-path="violin-and-boxplots-for-response-probability-rf8-1.html"><a href="violin-and-boxplots-for-response-probability-rf8-1.html"><i class="fa fa-check"></i><b>5.4</b> Violin and Boxplots for Response Probability (RF8)</a></li>
<li class="chapter" data-level="5.5" data-path="rf8-score-among-clusters.html"><a href="rf8-score-among-clusters.html"><i class="fa fa-check"></i><b>5.5</b> RF8 score among clusters</a></li>
<li class="chapter" data-level="5.6" data-path="integration-of-ex-vivo-drug-response-in-beataml.html"><a href="integration-of-ex-vivo-drug-response-in-beataml.html"><i class="fa fa-check"></i><b>5.6</b> Integration of ex vivo drug-response in BeatAML</a></li>
<li class="chapter" data-level="5.7" data-path="beataml-waves-1-4-heatmap-based-on-drug-profiling.html"><a href="beataml-waves-1-4-heatmap-based-on-drug-profiling.html"><i class="fa fa-check"></i><b>5.7</b> BeatAML Waves 1-4 heatmap based on drug profiling</a></li>
<li class="chapter" data-level="5.8" data-path="boxplot-for-drug-sensitivitity-between-resistant-and-sensitive.html"><a href="boxplot-for-drug-sensitivitity-between-resistant-and-sensitive.html"><i class="fa fa-check"></i><b>5.8</b> Boxplot for drug sensitivitity between resistant and sensitive</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>6</b> Figure 5</a>
<ul>
<li class="chapter" data-level="6.1" data-path="integration-with-aml-genomics.html"><a href="integration-with-aml-genomics.html"><i class="fa fa-check"></i><b>6.1</b> Integration with AML genomics</a></li>
<li class="chapter" data-level="6.2" data-path="mutation-pairs.html"><a href="mutation-pairs.html"><i class="fa fa-check"></i><b>6.2</b> Mutation pairs</a></li>
<li class="chapter" data-level="6.3" data-path="complexheatmap-for-mutation.html"><a href="complexheatmap-for-mutation.html"><i class="fa fa-check"></i><b>6.3</b> Complexheatmap for mutation</a></li>
<li class="chapter" data-level="6.4" data-path="fisher-test.html"><a href="fisher-test.html"><i class="fa fa-check"></i><b>6.4</b> Fisher test</a></li>
<li class="chapter" data-level="6.5" data-path="barplot-in-rjaml-cohort.html"><a href="barplot-in-rjaml-cohort.html"><i class="fa fa-check"></i><b>6.5</b> Barplot in RJAML cohort</a></li>
<li class="chapter" data-level="6.6" data-path="table-one-for-multi-cohorts.html"><a href="table-one-for-multi-cohorts.html"><i class="fa fa-check"></i><b>6.6</b> Table one for multi-cohorts</a></li>
<li class="chapter" data-level="6.7" data-path="barplot-for-real-response.html"><a href="barplot-for-real-response.html"><i class="fa fa-check"></i><b>6.7</b> Barplot for real response</a></li>
<li class="chapter" data-level="6.8" data-path="multivariable-logisticreg.html"><a href="multivariable-logisticreg.html"><i class="fa fa-check"></i><b>6.8</b> Multivariable LogisticReg</a></li>
<li class="chapter" data-level="6.9" data-path="identification-of-rf8-correlated-genes.html"><a href="identification-of-rf8-correlated-genes.html"><i class="fa fa-check"></i><b>6.9</b> Identification of RF8-correlated genes</a></li>
<li class="chapter" data-level="6.10" data-path="differential-expression-analysis-for-mut-vs.-wt.html"><a href="differential-expression-analysis-for-mut-vs.-wt.html"><i class="fa fa-check"></i><b>6.10</b> Differential Expression Analysis for Mut vs. WT</a></li>
<li class="chapter" data-level="6.11" data-path="venn-diagram-for-target-genes.html"><a href="venn-diagram-for-target-genes.html"><i class="fa fa-check"></i><b>6.11</b> Venn Diagram for target genes</a></li>
<li class="chapter" data-level="6.12" data-path="pathway-overlap-analysis-of-genes.html"><a href="pathway-overlap-analysis-of-genes.html"><i class="fa fa-check"></i><b>6.12</b> Pathway overlap analysis of genes</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="figure-6.html"><a href="figure-6.html"><i class="fa fa-check"></i><b>7</b> Figure 6</a>
<ul>
<li class="chapter" data-level="7.1" data-path="relationship-between-rf8-score-vs.-venaza-response-probability.html"><a href="relationship-between-rf8-score-vs.-venaza-response-probability.html"><i class="fa fa-check"></i><b>7.1</b> Relationship between RF8 score vs. VEN/AZA response probability</a></li>
<li class="chapter" data-level="7.2" data-path="monotonic-response-prediction-plot.html"><a href="monotonic-response-prediction-plot.html"><i class="fa fa-check"></i><b>7.2</b> Monotonic Response Prediction Plot</a></li>
<li class="chapter" data-level="7.3" data-path="survival-plot.html"><a href="survival-plot.html"><i class="fa fa-check"></i><b>7.3</b> Survival plot</a></li>
<li class="chapter" data-level="7.4" data-path="brier-score-analysis.html"><a href="brier-score-analysis.html"><i class="fa fa-check"></i><b>7.4</b> Brier Score Analysis</a></li>
<li class="chapter" data-level="7.5" data-path="c-index-comparison.html"><a href="c-index-comparison.html"><i class="fa fa-check"></i><b>7.5</b> C-index Comparison</a></li>
<li class="chapter" data-level="7.6" data-path="km-plots-for-rjaml-cohort-2.html"><a href="km-plots-for-rjaml-cohort-2.html"><i class="fa fa-check"></i><b>7.6</b> KM-Plots for RJAML Cohort 2</a></li>
<li class="chapter" data-level="7.7" data-path="barplot-for-crcri.html"><a href="barplot-for-crcri.html"><i class="fa fa-check"></i><b>7.7</b> Barplot for CR/CRi</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="identification-of-rf8-correlated-genes" class="section level2 hasAnchor" number="6.9">
<h2><span class="header-section-number">6.9</span> Identification of RF8-correlated genes<a href="identification-of-rf8-correlated-genes.html#identification-of-rf8-correlated-genes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb167"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb167-1"><a href="identification-of-rf8-correlated-genes.html#cb167-1" aria-hidden="true" tabindex="-1"></a><span class="co"># library(dplyr)</span></span>
<span id="cb167-2"><a href="identification-of-rf8-correlated-genes.html#cb167-2" aria-hidden="true" tabindex="-1"></a><span class="co"># library(DESeq2)</span></span>
<span id="cb167-3"><a href="identification-of-rf8-correlated-genes.html#cb167-3" aria-hidden="true" tabindex="-1"></a><span class="co"># source(&quot;RunDESeq2.R&quot;)</span></span>
<span id="cb167-4"><a href="identification-of-rf8-correlated-genes.html#cb167-4" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb167-5"><a href="identification-of-rf8-correlated-genes.html#cb167-5" aria-hidden="true" tabindex="-1"></a><span class="co"># # Load TPM and raw count data</span></span>
<span id="cb167-6"><a href="identification-of-rf8-correlated-genes.html#cb167-6" aria-hidden="true" tabindex="-1"></a><span class="co"># tpm &lt;- read.table(&quot;../02.data/ID_matched_TPM_matrix.txt&quot;, header = TRUE, check.names = FALSE)</span></span>
<span id="cb167-7"><a href="identification-of-rf8-correlated-genes.html#cb167-7" aria-hidden="true" tabindex="-1"></a><span class="co"># sample_info &lt;- read.csv(&quot;D:\\01.Projects\\01.project\\35_AZA_VEN\\00.RNAseq_Model\\figure3\\rjaml_rf8_scores.csv&quot;, header = TRUE, check.names = FALSE)</span></span>
<span id="cb167-8"><a href="identification-of-rf8-correlated-genes.html#cb167-8" aria-hidden="true" tabindex="-1"></a><span class="co"># tpm &lt;- tpm[, c(&quot;ID&quot;, &quot;Symbol&quot;, &quot;Type&quot;, as.character(sample_info$PatientID))]</span></span>
<span id="cb167-9"><a href="identification-of-rf8-correlated-genes.html#cb167-9" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb167-10"><a href="identification-of-rf8-correlated-genes.html#cb167-10" aria-hidden="true" tabindex="-1"></a><span class="co"># tpm &lt;- subset(tpm, Type %in% c(&quot;protein_coding&quot;, &quot;miRNA&quot;, &quot;lincRNA&quot;, &quot;antisense&quot;,</span></span>
<span id="cb167-11"><a href="identification-of-rf8-correlated-genes.html#cb167-11" aria-hidden="true" tabindex="-1"></a><span class="co">#                                &quot;processed_transcript&quot;,</span></span>
<span id="cb167-12"><a href="identification-of-rf8-correlated-genes.html#cb167-12" aria-hidden="true" tabindex="-1"></a><span class="co">#                                &quot;transcribed_processed_pseudogene&quot;, &quot;transcribed_unitary_pseudogene&quot;,</span></span>
<span id="cb167-13"><a href="identification-of-rf8-correlated-genes.html#cb167-13" aria-hidden="true" tabindex="-1"></a><span class="co">#                                &quot;transcribed_unprocessed_pseudogene&quot;,&quot;macro_lncRNA&quot;, &quot;TEC&quot;,</span></span>
<span id="cb167-14"><a href="identification-of-rf8-correlated-genes.html#cb167-14" aria-hidden="true" tabindex="-1"></a><span class="co">#                                &quot;3prime_overlapping_ncRNA&quot;,&quot;sense_intronic&quot;, &quot;sense_overlapping&quot;,</span></span>
<span id="cb167-15"><a href="identification-of-rf8-correlated-genes.html#cb167-15" aria-hidden="true" tabindex="-1"></a><span class="co">#                                &quot;unprocessed_pseudogene&quot;))</span></span>
<span id="cb167-16"><a href="identification-of-rf8-correlated-genes.html#cb167-16" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb167-17"><a href="identification-of-rf8-correlated-genes.html#cb167-17" aria-hidden="true" tabindex="-1"></a><span class="co"># tpm &lt;- tpm[which(rowMeans(tpm[, -c(1:3)]) &gt; 0.5), ]</span></span>
<span id="cb167-18"><a href="identification-of-rf8-correlated-genes.html#cb167-18" aria-hidden="true" tabindex="-1"></a><span class="co"># tpm &lt;- tpm[!duplicated(tpm$Symbol), ]</span></span>
<span id="cb167-19"><a href="identification-of-rf8-correlated-genes.html#cb167-19" aria-hidden="true" tabindex="-1"></a><span class="co"># row.names(tpm) &lt;- tpm$Symbol</span></span>
<span id="cb167-20"><a href="identification-of-rf8-correlated-genes.html#cb167-20" aria-hidden="true" tabindex="-1"></a><span class="co"># tpm &lt;- tpm[, -c(1:3)]</span></span>
<span id="cb167-21"><a href="identification-of-rf8-correlated-genes.html#cb167-21" aria-hidden="true" tabindex="-1"></a><span class="co"># tpm &lt;- data.frame(t(tpm))</span></span>
<span id="cb167-22"><a href="identification-of-rf8-correlated-genes.html#cb167-22" aria-hidden="true" tabindex="-1"></a><span class="co"># tpm &lt;- log2(tpm + 1)</span></span>
<span id="cb167-23"><a href="identification-of-rf8-correlated-genes.html#cb167-23" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb167-24"><a href="identification-of-rf8-correlated-genes.html#cb167-24" aria-hidden="true" tabindex="-1"></a><span class="co"># dat &lt;- cbind(sample_info$RF8, tpm)</span></span>
<span id="cb167-25"><a href="identification-of-rf8-correlated-genes.html#cb167-25" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(dat)[1] &lt;- &quot;RF8&quot;</span></span>
<span id="cb167-26"><a href="identification-of-rf8-correlated-genes.html#cb167-26" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb167-27"><a href="identification-of-rf8-correlated-genes.html#cb167-27" aria-hidden="true" tabindex="-1"></a><span class="co"># ## Define a dataframe to store the results</span></span>
<span id="cb167-28"><a href="identification-of-rf8-correlated-genes.html#cb167-28" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb167-29"><a href="identification-of-rf8-correlated-genes.html#cb167-29" aria-hidden="true" tabindex="-1"></a><span class="co"># stat1 &lt;- data.frame(matrix(NA, nrow = (ncol(dat)-1), ncol = 3))</span></span>
<span id="cb167-30"><a href="identification-of-rf8-correlated-genes.html#cb167-30" aria-hidden="true" tabindex="-1"></a><span class="co"># stat1[, 1] &lt;- colnames(dat)[2:ncol(dat)]</span></span>
<span id="cb167-31"><a href="identification-of-rf8-correlated-genes.html#cb167-31" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb167-32"><a href="identification-of-rf8-correlated-genes.html#cb167-32" aria-hidden="true" tabindex="-1"></a><span class="co"># for (i in 2:ncol(dat)) {</span></span>
<span id="cb167-33"><a href="identification-of-rf8-correlated-genes.html#cb167-33" aria-hidden="true" tabindex="-1"></a><span class="co">#   res &lt;- cor.test(dat[,1], dat[,i], method = &quot;pearson&quot;)</span></span>
<span id="cb167-34"><a href="identification-of-rf8-correlated-genes.html#cb167-34" aria-hidden="true" tabindex="-1"></a><span class="co">#   stat1[i-1, 2] &lt;- res$p.value</span></span>
<span id="cb167-35"><a href="identification-of-rf8-correlated-genes.html#cb167-35" aria-hidden="true" tabindex="-1"></a><span class="co">#   stat1[i-1, 3] &lt;- res$estimate</span></span>
<span id="cb167-36"><a href="identification-of-rf8-correlated-genes.html#cb167-36" aria-hidden="true" tabindex="-1"></a><span class="co"># }</span></span>
<span id="cb167-37"><a href="identification-of-rf8-correlated-genes.html#cb167-37" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb167-38"><a href="identification-of-rf8-correlated-genes.html#cb167-38" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(stat1) &lt;- c(&quot;ID&quot;, &quot;P.value&quot;, &quot;rho&quot;)</span></span>
<span id="cb167-39"><a href="identification-of-rf8-correlated-genes.html#cb167-39" aria-hidden="true" tabindex="-1"></a><span class="co"># stat1$fdr &lt;- p.adjust(stat1$P.value, method =&quot;BH&quot;)</span></span>
<span id="cb167-40"><a href="identification-of-rf8-correlated-genes.html#cb167-40" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb167-41"><a href="identification-of-rf8-correlated-genes.html#cb167-41" aria-hidden="true" tabindex="-1"></a><span class="co"># write.csv(stat1, &quot;../03.out/figure5/RF8_correlatedGenes.csv&quot;)</span></span></code></pre></div>
<div class="sourceCode" id="cb168"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb168-1"><a href="identification-of-rf8-correlated-genes.html#cb168-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb168-2"><a href="identification-of-rf8-correlated-genes.html#cb168-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb168-3"><a href="identification-of-rf8-correlated-genes.html#cb168-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrastr)</span>
<span id="cb168-4"><a href="identification-of-rf8-correlated-genes.html#cb168-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(viridis)   </span>
<span id="cb168-5"><a href="identification-of-rf8-correlated-genes.html#cb168-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb168-6"><a href="identification-of-rf8-correlated-genes.html#cb168-6" aria-hidden="true" tabindex="-1"></a>stat1 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../03.out/figure5/RF8_correlatedGenes.csv&quot;</span>, <span class="at">header =</span> T)</span>
<span id="cb168-7"><a href="identification-of-rf8-correlated-genes.html#cb168-7" aria-hidden="true" tabindex="-1"></a>stat1<span class="sc">$</span>rank_score <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log2</span>(stat1<span class="sc">$</span>P.value) <span class="sc">*</span> <span class="fu">abs</span>(stat1<span class="sc">$</span>rho)</span>
<span id="cb168-8"><a href="identification-of-rf8-correlated-genes.html#cb168-8" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> stat1[<span class="fu">order</span>(stat1<span class="sc">$</span>rank_score, <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ]</span>
<span id="cb168-9"><a href="identification-of-rf8-correlated-genes.html#cb168-9" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Rank <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(stat1)</span>
<span id="cb168-10"><a href="identification-of-rf8-correlated-genes.html#cb168-10" aria-hidden="true" tabindex="-1"></a>dat1 <span class="ot">&lt;-</span> <span class="fu">head</span>(dat, <span class="dv">20</span>)</span>
<span id="cb168-11"><a href="identification-of-rf8-correlated-genes.html#cb168-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb168-12"><a href="identification-of-rf8-correlated-genes.html#cb168-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Sort data by rank_score and find the top 500 threshold</span></span>
<span id="cb168-13"><a href="identification-of-rf8-correlated-genes.html#cb168-13" aria-hidden="true" tabindex="-1"></a>top_1000_threshold <span class="ot">&lt;-</span> dat<span class="sc">$</span>rank_score[<span class="fu">order</span>(dat<span class="sc">$</span>rank_score, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)][<span class="dv">1000</span>]</span>
<span id="cb168-14"><a href="identification-of-rf8-correlated-genes.html#cb168-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb168-15"><a href="identification-of-rf8-correlated-genes.html#cb168-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting</span></span>
<span id="cb168-16"><a href="identification-of-rf8-correlated-genes.html#cb168-16" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(dat, <span class="fu">aes</span>(<span class="at">x =</span> Rank, <span class="at">y =</span> rank_score, <span class="at">color =</span> rank_score)) <span class="sc">+</span></span>
<span id="cb168-17"><a href="identification-of-rf8-correlated-genes.html#cb168-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point_rast</span>(<span class="fu">aes</span>(<span class="at">size =</span> <span class="fu">abs</span>(rank_score)), <span class="at">alpha =</span> <span class="fl">0.8</span>, <span class="at">raster.dpi =</span> <span class="dv">350</span>, <span class="at">stroke =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb168-18"><a href="identification-of-rf8-correlated-genes.html#cb168-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_viridis</span>(<span class="at">option =</span> <span class="st">&quot;plasma&quot;</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">name =</span> <span class="st">&quot;Rank Score&quot;</span>) <span class="sc">+</span></span>
<span id="cb168-19"><a href="identification-of-rf8-correlated-genes.html#cb168-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Rank&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Rank score&quot;</span>) <span class="sc">+</span></span>
<span id="cb168-20"><a href="identification-of-rf8-correlated-genes.html#cb168-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">14</span>) <span class="sc">+</span></span>
<span id="cb168-21"><a href="identification-of-rf8-correlated-genes.html#cb168-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb168-22"><a href="identification-of-rf8-correlated-genes.html#cb168-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">&quot;right&quot;</span>,</span>
<span id="cb168-23"><a href="identification-of-rf8-correlated-genes.html#cb168-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>),</span>
<span id="cb168-24"><a href="identification-of-rf8-correlated-genes.html#cb168-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb168-25"><a href="identification-of-rf8-correlated-genes.html#cb168-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.major =</span> <span class="fu">element_line</span>(<span class="at">size =</span> <span class="fl">0.2</span>, <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">color =</span> <span class="st">&quot;grey&quot;</span>),</span>
<span id="cb168-26"><a href="identification-of-rf8-correlated-genes.html#cb168-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb168-27"><a href="identification-of-rf8-correlated-genes.html#cb168-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>)</span>
<span id="cb168-28"><a href="identification-of-rf8-correlated-genes.html#cb168-28" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb168-29"><a href="identification-of-rf8-correlated-genes.html#cb168-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> top_1000_threshold, <span class="at">color =</span> <span class="st">&quot;red&quot;</span>, <span class="at">linetype =</span> <span class="st">&quot;dotted&quot;</span>, <span class="at">lwd =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb168-30"><a href="identification-of-rf8-correlated-genes.html#cb168-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">&quot;text&quot;</span>, <span class="at">x =</span> <span class="fu">max</span>(dat<span class="sc">$</span>Rank) <span class="sc">*</span> <span class="fl">0.95</span>, <span class="at">y =</span> top_1000_threshold, <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">&quot;Top 1000 Genes (&quot;</span>, <span class="fu">round</span>(top_1000_threshold, <span class="dv">2</span>), <span class="st">&quot;)&quot;</span>),</span>
<span id="cb168-31"><a href="identification-of-rf8-correlated-genes.html#cb168-31" aria-hidden="true" tabindex="-1"></a>           <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">&quot;red&quot;</span>, <span class="at">fontface =</span> <span class="st">&quot;bold&quot;</span>, <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb168-32"><a href="identification-of-rf8-correlated-genes.html#cb168-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_size_continuous</span>(<span class="at">range =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">6</span>)) <span class="co"># Adjust point size range for better visibility</span></span>
<span id="cb168-33"><a href="identification-of-rf8-correlated-genes.html#cb168-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb168-34"><a href="identification-of-rf8-correlated-genes.html#cb168-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Adding highlighted points and labels</span></span>
<span id="cb168-35"><a href="identification-of-rf8-correlated-genes.html#cb168-35" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> p <span class="sc">+</span></span>
<span id="cb168-36"><a href="identification-of-rf8-correlated-genes.html#cb168-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> dat1, <span class="fu">aes</span>(<span class="at">size =</span> <span class="fu">abs</span>(rank_score)), <span class="at">shape =</span> <span class="dv">21</span>, <span class="at">stroke =</span> <span class="fl">1.5</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;white&quot;</span>) <span class="sc">+</span></span>
<span id="cb168-37"><a href="identification-of-rf8-correlated-genes.html#cb168-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_label_repel</span>(<span class="at">data =</span> dat1, <span class="fu">aes</span>(<span class="at">label =</span> ID), <span class="at">color =</span> <span class="st">&quot;black&quot;</span>, </span>
<span id="cb168-38"><a href="identification-of-rf8-correlated-genes.html#cb168-38" aria-hidden="true" tabindex="-1"></a>                   <span class="at">box.padding =</span> <span class="fu">unit</span>(<span class="fl">0.4</span>, <span class="st">&quot;lines&quot;</span>), <span class="at">point.padding =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;lines&quot;</span>),</span>
<span id="cb168-39"><a href="identification-of-rf8-correlated-genes.html#cb168-39" aria-hidden="true" tabindex="-1"></a>                   <span class="at">max.overlaps =</span> <span class="dv">100</span>, <span class="at">segment.color =</span> <span class="st">&quot;grey50&quot;</span>, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">fontface =</span> <span class="st">&quot;bold&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;lightyellow&quot;</span>)</span>
<span id="cb168-40"><a href="identification-of-rf8-correlated-genes.html#cb168-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb168-41"><a href="identification-of-rf8-correlated-genes.html#cb168-41" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p1)</span></code></pre></div>
<p><img src="figures/unnamed-chunk-75-1.png" width="672" /></p>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="multivariable-logisticreg.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="differential-expression-analysis-for-mut-vs.-wt.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/05-results5.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
