<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6.12 Pathway overlap analysis of genes | Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning</title>
  <meta name="description" content="This is a book-style code sharing." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="6.12 Pathway overlap analysis of genes | Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a book-style code sharing." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6.12 Pathway overlap analysis of genes | Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning" />
  
  <meta name="twitter:description" content="This is a book-style code sharing." />
  

<meta name="author" content="Peng Jin, Qiqi Jin, Kankan Wang" />


<meta name="date" content="2025-07-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="venn-diagram-for-target-genes.html"/>
<link rel="next" href="figure-6.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Background</a></li>
<li class="chapter" data-level="2" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>2</b> Figure 1</a>
<ul>
<li class="chapter" data-level="2.1" data-path="process-the-data-from-functional_precision_medicine_tumor_board_aml.html"><a href="process-the-data-from-functional_precision_medicine_tumor_board_aml.html"><i class="fa fa-check"></i><b>2.1</b> Process the data from Functional_Precision_Medicine_Tumor_Board_AML</a></li>
<li class="chapter" data-level="2.2" data-path="functional_precision_medicine_tumor_board_aml.html"><a href="functional_precision_medicine_tumor_board_aml.html"><i class="fa fa-check"></i><b>2.2</b> Functional_Precision_Medicine_Tumor_Board_AML</a></li>
<li class="chapter" data-level="2.3" data-path="boxplot-for-drug-sensitivity-between-resistant-and-sensitive.html"><a href="boxplot-for-drug-sensitivity-between-resistant-and-sensitive.html"><i class="fa fa-check"></i><b>2.3</b> Boxplot for drug sensitivity between resistant and sensitive</a></li>
<li class="chapter" data-level="2.4" data-path="de-analysis-functional_medicine.html"><a href="de-analysis-functional_medicine.html"><i class="fa fa-check"></i><b>2.4</b> DE analysis Functional_medicine</a></li>
<li class="chapter" data-level="2.5" data-path="valcanoplot-plot-ex-vivo.html"><a href="valcanoplot-plot-ex-vivo.html"><i class="fa fa-check"></i><b>2.5</b> Valcanoplot plot (Ex vivo)</a></li>
<li class="chapter" data-level="2.6" data-path="heatmap-for-representative-de-genes.html"><a href="heatmap-for-representative-de-genes.html"><i class="fa fa-check"></i><b>2.6</b> Heatmap for representative DE genes</a></li>
<li class="chapter" data-level="2.7" data-path="cibersortx-results-visualization-ex-vivo.html"><a href="cibersortx-results-visualization-ex-vivo.html"><i class="fa fa-check"></i><b>2.7</b> CIBERSORTx Results Visualization (ex vivo)</a></li>
<li class="chapter" data-level="2.8" data-path="cibersortx-results-visualization-in-vivo.html"><a href="cibersortx-results-visualization-in-vivo.html"><i class="fa fa-check"></i><b>2.8</b> CIBERSORTx Results Visualization (in vivo)</a></li>
<li class="chapter" data-level="2.9" data-path="ssgsea-analysis-in-vivo.html"><a href="ssgsea-analysis-in-vivo.html"><i class="fa fa-check"></i><b>2.9</b> ssGSEA analysis in vivo</a></li>
<li class="chapter" data-level="2.10" data-path="go-enrichment-dot-plot.html"><a href="go-enrichment-dot-plot.html"><i class="fa fa-check"></i><b>2.10</b> GO Enrichment Dot Plot</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>3</b> Figure 2</a>
<ul>
<li class="chapter" data-level="3.1" data-path="de-analysis-between-in-vivo-sensitive-and-resistant.html"><a href="de-analysis-between-in-vivo-sensitive-and-resistant.html"><i class="fa fa-check"></i><b>3.1</b> DE analysis between in vivo sensitive and resistant</a></li>
<li class="chapter" data-level="3.2" data-path="de-analysis-rjaml-azaven-cohort.html"><a href="de-analysis-rjaml-azaven-cohort.html"><i class="fa fa-check"></i><b>3.2</b> DE analysis RJAML AZA+VEN cohort</a></li>
<li class="chapter" data-level="3.3" data-path="valcanoplot-plot-ex-vivo-1.html"><a href="valcanoplot-plot-ex-vivo-1.html"><i class="fa fa-check"></i><b>3.3</b> Valcanoplot plot (Ex vivo)</a></li>
<li class="chapter" data-level="3.4" data-path="valcanoplot-plot-in-vivo.html"><a href="valcanoplot-plot-in-vivo.html"><i class="fa fa-check"></i><b>3.4</b> Valcanoplot plot (In vivo)</a></li>
<li class="chapter" data-level="3.5" data-path="venn-plot.html"><a href="venn-plot.html"><i class="fa fa-check"></i><b>3.5</b> Venn plot</a></li>
<li class="chapter" data-level="3.6" data-path="integration-of-degs-ex-vivo-and-in-vivo.html"><a href="integration-of-degs-ex-vivo-and-in-vivo.html"><i class="fa fa-check"></i><b>3.6</b> Integration of DEGs ex vivo and in vivo</a></li>
<li class="chapter" data-level="3.7" data-path="gsea-analysis-for-crispri-drug-sensitivity.html"><a href="gsea-analysis-for-crispri-drug-sensitivity.html"><i class="fa fa-check"></i><b>3.7</b> GSEA analysis for CRISPRi drug sensitivity</a></li>
<li class="chapter" data-level="3.8" data-path="differential-expression-analysis-for-shpink1.html"><a href="differential-expression-analysis-for-shpink1.html"><i class="fa fa-check"></i><b>3.8</b> Differential Expression Analysis for shPINK1</a></li>
<li class="chapter" data-level="3.9" data-path="de-analysis-in-cell-lines.html"><a href="de-analysis-in-cell-lines.html"><i class="fa fa-check"></i><b>3.9</b> DE analysis in cell lines</a></li>
<li class="chapter" data-level="3.10" data-path="valcano-plots-molm13.html"><a href="valcano-plots-molm13.html"><i class="fa fa-check"></i><b>3.10</b> Valcano plots (MOLM13)</a></li>
<li class="chapter" data-level="3.11" data-path="valcano-plots-oa3.html"><a href="valcano-plots-oa3.html"><i class="fa fa-check"></i><b>3.11</b> Valcano plots (OA3)</a></li>
<li class="chapter" data-level="3.12" data-path="gsea-analysis-molm13.html"><a href="gsea-analysis-molm13.html"><i class="fa fa-check"></i><b>3.12</b> GSEA analysis (MOLM13)</a></li>
<li class="chapter" data-level="3.13" data-path="gsea-analysis-oa3.html"><a href="gsea-analysis-oa3.html"><i class="fa fa-check"></i><b>3.13</b> GSEA analysis (OA3)</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>4</b> Figure 3</a>
<ul>
<li class="chapter" data-level="4.1" data-path="benchmark-ml-methods-for-prediction.html"><a href="benchmark-ml-methods-for-prediction.html"><i class="fa fa-check"></i><b>4.1</b> Benchmark ML Methods for Prediction</a></li>
<li class="chapter" data-level="4.2" data-path="predict-class-labels-on-the-test-set-and-plot-roc.html"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html"><i class="fa fa-check"></i><b>4.2</b> Predict Class Labels on the Test Set and Plot ROC</a></li>
<li class="chapter" data-level="4.3" data-path="auc-and-auprc-heatmap-for-the-entire-and-test-sets.html"><a href="auc-and-auprc-heatmap-for-the-entire-and-test-sets.html"><i class="fa fa-check"></i><b>4.3</b> AUC and AUPRC Heatmap for the Entire and Test Sets</a></li>
<li class="chapter" data-level="4.4" data-path="model-performance-plot.html"><a href="model-performance-plot.html"><i class="fa fa-check"></i><b>4.4</b> Model Performance Plot</a></li>
<li class="chapter" data-level="4.5" data-path="feature-importance.html"><a href="feature-importance.html"><i class="fa fa-check"></i><b>4.5</b> Feature importance</a></li>
<li class="chapter" data-level="4.6" data-path="heatmap-for-confusion-matrices-defaut-cutoff-0.html"><a href="heatmap-for-confusion-matrices-defaut-cutoff-0.html"><i class="fa fa-check"></i><b>4.6</b> Heatmap for Confusion Matrices (defaut cutoff 0.5)</a></li>
<li class="chapter" data-level="4.7" data-path="barplot-for-predicted-response-0.html"><a href="barplot-for-predicted-response-0.html"><i class="fa fa-check"></i><b>4.7</b> Barplot for predicted Response (0.5)</a></li>
<li class="chapter" data-level="4.8" data-path="heatmap-for-confusion-matrices-optimal-cutoff-0.html"><a href="heatmap-for-confusion-matrices-optimal-cutoff-0.html"><i class="fa fa-check"></i><b>4.8</b> Heatmap for Confusion Matrices (optimal cutoff 0.630)</a></li>
<li class="chapter" data-level="4.9" data-path="barplot-for-predicted-response-0.html"><a href="barplot-for-predicted-response-0.html#barplot-for-predicted-response-0.630"><i class="fa fa-check"></i><b>4.9</b> Barplot for predicted Response (0.630)</a></li>
<li class="chapter" data-level="4.10" data-path="density-plot-for-response-probability.html"><a href="density-plot-for-response-probability.html"><i class="fa fa-check"></i><b>4.10</b> Density Plot for Response Probability</a></li>
<li class="chapter" data-level="4.11" data-path="violin-and-boxplots-for-response-probability-rf8.html"><a href="violin-and-boxplots-for-response-probability-rf8.html"><i class="fa fa-check"></i><b>4.11</b> Violin and Boxplots for Response Probability (RF8)</a></li>
<li class="chapter" data-level="4.12" data-path="barplot-for-response.html"><a href="barplot-for-response.html"><i class="fa fa-check"></i><b>4.12</b> Barplot for Response</a></li>
<li class="chapter" data-level="4.13" data-path="rf8-for-mdr-status.html"><a href="rf8-for-mdr-status.html"><i class="fa fa-check"></i><b>4.13</b> RF8 for MDR status</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>5</b> Figure 4</a>
<ul>
<li class="chapter" data-level="5.1" data-path="validation-of-rf8-model-rjaml-cohort-2.html"><a href="validation-of-rf8-model-rjaml-cohort-2.html"><i class="fa fa-check"></i><b>5.1</b> Validation of RF8 model (RJAML-Cohort 2)</a></li>
<li class="chapter" data-level="5.2" data-path="validation-of-rf8-model-fpmtb.html"><a href="validation-of-rf8-model-fpmtb.html"><i class="fa fa-check"></i><b>5.2</b> Validation of RF8 model (FPMTB)</a></li>
<li class="chapter" data-level="5.3" data-path="validation-of-rf8-model-beataml.html"><a href="validation-of-rf8-model-beataml.html"><i class="fa fa-check"></i><b>5.3</b> Validation of RF8 model (BeatAML)</a></li>
<li class="chapter" data-level="5.4" data-path="violin-and-boxplots-for-response-probability-rf8-1.html"><a href="violin-and-boxplots-for-response-probability-rf8-1.html"><i class="fa fa-check"></i><b>5.4</b> Violin and Boxplots for Response Probability (RF8)</a></li>
<li class="chapter" data-level="5.5" data-path="rf8-score-among-clusters.html"><a href="rf8-score-among-clusters.html"><i class="fa fa-check"></i><b>5.5</b> RF8 score among clusters</a></li>
<li class="chapter" data-level="5.6" data-path="integration-of-ex-vivo-drug-response-in-beataml.html"><a href="integration-of-ex-vivo-drug-response-in-beataml.html"><i class="fa fa-check"></i><b>5.6</b> Integration of ex vivo drug-response in BeatAML</a></li>
<li class="chapter" data-level="5.7" data-path="beataml-waves-1-4-heatmap-based-on-drug-profiling.html"><a href="beataml-waves-1-4-heatmap-based-on-drug-profiling.html"><i class="fa fa-check"></i><b>5.7</b> BeatAML Waves 1-4 heatmap based on drug profiling</a></li>
<li class="chapter" data-level="5.8" data-path="boxplot-for-drug-sensitivitity-between-resistant-and-sensitive.html"><a href="boxplot-for-drug-sensitivitity-between-resistant-and-sensitive.html"><i class="fa fa-check"></i><b>5.8</b> Boxplot for drug sensitivitity between resistant and sensitive</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>6</b> Figure 5</a>
<ul>
<li class="chapter" data-level="6.1" data-path="integration-with-aml-genomics.html"><a href="integration-with-aml-genomics.html"><i class="fa fa-check"></i><b>6.1</b> Integration with AML genomics</a></li>
<li class="chapter" data-level="6.2" data-path="mutation-pairs.html"><a href="mutation-pairs.html"><i class="fa fa-check"></i><b>6.2</b> Mutation pairs</a></li>
<li class="chapter" data-level="6.3" data-path="complexheatmap-for-mutation.html"><a href="complexheatmap-for-mutation.html"><i class="fa fa-check"></i><b>6.3</b> Complexheatmap for mutation</a></li>
<li class="chapter" data-level="6.4" data-path="fisher-test.html"><a href="fisher-test.html"><i class="fa fa-check"></i><b>6.4</b> Fisher test</a></li>
<li class="chapter" data-level="6.5" data-path="barplot-in-rjaml-cohort.html"><a href="barplot-in-rjaml-cohort.html"><i class="fa fa-check"></i><b>6.5</b> Barplot in RJAML cohort</a></li>
<li class="chapter" data-level="6.6" data-path="table-one-for-multi-cohorts.html"><a href="table-one-for-multi-cohorts.html"><i class="fa fa-check"></i><b>6.6</b> Table one for multi-cohorts</a></li>
<li class="chapter" data-level="6.7" data-path="barplot-for-real-response.html"><a href="barplot-for-real-response.html"><i class="fa fa-check"></i><b>6.7</b> Barplot for real response</a></li>
<li class="chapter" data-level="6.8" data-path="multivariable-logisticreg.html"><a href="multivariable-logisticreg.html"><i class="fa fa-check"></i><b>6.8</b> Multivariable LogisticReg</a></li>
<li class="chapter" data-level="6.9" data-path="identification-of-rf8-correlated-genes.html"><a href="identification-of-rf8-correlated-genes.html"><i class="fa fa-check"></i><b>6.9</b> Identification of RF8-correlated genes</a></li>
<li class="chapter" data-level="6.10" data-path="differential-expression-analysis-for-mut-vs.-wt.html"><a href="differential-expression-analysis-for-mut-vs.-wt.html"><i class="fa fa-check"></i><b>6.10</b> Differential Expression Analysis for Mut vs. WT</a></li>
<li class="chapter" data-level="6.11" data-path="venn-diagram-for-target-genes.html"><a href="venn-diagram-for-target-genes.html"><i class="fa fa-check"></i><b>6.11</b> Venn Diagram for target genes</a></li>
<li class="chapter" data-level="6.12" data-path="pathway-overlap-analysis-of-genes.html"><a href="pathway-overlap-analysis-of-genes.html"><i class="fa fa-check"></i><b>6.12</b> Pathway overlap analysis of genes</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="figure-6.html"><a href="figure-6.html"><i class="fa fa-check"></i><b>7</b> Figure 6</a>
<ul>
<li class="chapter" data-level="7.1" data-path="relationship-between-rf8-score-vs.-venaza-response-probability.html"><a href="relationship-between-rf8-score-vs.-venaza-response-probability.html"><i class="fa fa-check"></i><b>7.1</b> Relationship between RF8 score vs. VEN/AZA response probability</a></li>
<li class="chapter" data-level="7.2" data-path="monotonic-response-prediction-plot.html"><a href="monotonic-response-prediction-plot.html"><i class="fa fa-check"></i><b>7.2</b> Monotonic Response Prediction Plot</a></li>
<li class="chapter" data-level="7.3" data-path="survival-plot.html"><a href="survival-plot.html"><i class="fa fa-check"></i><b>7.3</b> Survival plot</a></li>
<li class="chapter" data-level="7.4" data-path="brier-score-analysis.html"><a href="brier-score-analysis.html"><i class="fa fa-check"></i><b>7.4</b> Brier Score Analysis</a></li>
<li class="chapter" data-level="7.5" data-path="c-index-comparison.html"><a href="c-index-comparison.html"><i class="fa fa-check"></i><b>7.5</b> C-index Comparison</a></li>
<li class="chapter" data-level="7.6" data-path="km-plots-for-rjaml-cohort-2.html"><a href="km-plots-for-rjaml-cohort-2.html"><i class="fa fa-check"></i><b>7.6</b> KM-Plots for RJAML Cohort 2</a></li>
<li class="chapter" data-level="7.7" data-path="barplot-for-crcri.html"><a href="barplot-for-crcri.html"><i class="fa fa-check"></i><b>7.7</b> Barplot for CR/CRi</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="pathway-overlap-analysis-of-genes" class="section level2 hasAnchor" number="6.12">
<h2><span class="header-section-number">6.12</span> Pathway overlap analysis of genes<a href="pathway-overlap-analysis-of-genes.html#pathway-overlap-analysis-of-genes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb174"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb174-1"><a href="pathway-overlap-analysis-of-genes.html#cb174-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb174-2"><a href="pathway-overlap-analysis-of-genes.html#cb174-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb174-3"><a href="pathway-overlap-analysis-of-genes.html#cb174-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb174-4"><a href="pathway-overlap-analysis-of-genes.html#cb174-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb174-5"><a href="pathway-overlap-analysis-of-genes.html#cb174-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb174-6"><a href="pathway-overlap-analysis-of-genes.html#cb174-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-7"><a href="pathway-overlap-analysis-of-genes.html#cb174-7" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123456</span>)</span>
<span id="cb174-8"><a href="pathway-overlap-analysis-of-genes.html#cb174-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Define your gene lists</span></span>
<span id="cb174-9"><a href="pathway-overlap-analysis-of-genes.html#cb174-9" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb174-10"><a href="pathway-overlap-analysis-of-genes.html#cb174-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">RF8_gene =</span> rf8_genes<span class="sc">$</span>ID,</span>
<span id="cb174-11"><a href="pathway-overlap-analysis-of-genes.html#cb174-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">TP53_gene =</span> degs_tp53<span class="sc">$</span>Symbol,</span>
<span id="cb174-12"><a href="pathway-overlap-analysis-of-genes.html#cb174-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">DNMT3A_gene =</span> degs_dnmt3a<span class="sc">$</span>Symbol,</span>
<span id="cb174-13"><a href="pathway-overlap-analysis-of-genes.html#cb174-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">IDH2_gene =</span> degs_idh2<span class="sc">$</span>Symbol,</span>
<span id="cb174-14"><a href="pathway-overlap-analysis-of-genes.html#cb174-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">CEBPA_gene =</span> degs_cebpa<span class="sc">$</span>Symbol</span>
<span id="cb174-15"><a href="pathway-overlap-analysis-of-genes.html#cb174-15" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb174-16"><a href="pathway-overlap-analysis-of-genes.html#cb174-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-17"><a href="pathway-overlap-analysis-of-genes.html#cb174-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Map gene symbols to ENTREZ IDs</span></span>
<span id="cb174-18"><a href="pathway-overlap-analysis-of-genes.html#cb174-18" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">lapply</span>(genes, <span class="cf">function</span>(gene_set) {</span>
<span id="cb174-19"><a href="pathway-overlap-analysis-of-genes.html#cb174-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mapIds</span>(</span>
<span id="cb174-20"><a href="pathway-overlap-analysis-of-genes.html#cb174-20" aria-hidden="true" tabindex="-1"></a>    org.Hs.eg.db,</span>
<span id="cb174-21"><a href="pathway-overlap-analysis-of-genes.html#cb174-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">keys =</span> gene_set,</span>
<span id="cb174-22"><a href="pathway-overlap-analysis-of-genes.html#cb174-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">column =</span> <span class="st">&quot;ENTREZID&quot;</span>,</span>
<span id="cb174-23"><a href="pathway-overlap-analysis-of-genes.html#cb174-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">keytype =</span> <span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb174-24"><a href="pathway-overlap-analysis-of-genes.html#cb174-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">multiVals =</span> <span class="st">&quot;first&quot;</span></span>
<span id="cb174-25"><a href="pathway-overlap-analysis-of-genes.html#cb174-25" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span> <span class="fu">unique</span>()</span>
<span id="cb174-26"><a href="pathway-overlap-analysis-of-genes.html#cb174-26" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb174-27"><a href="pathway-overlap-analysis-of-genes.html#cb174-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-28"><a href="pathway-overlap-analysis-of-genes.html#cb174-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Obtain all gene-pathway associations from the database</span></span>
<span id="cb174-29"><a href="pathway-overlap-analysis-of-genes.html#cb174-29" aria-hidden="true" tabindex="-1"></a>all_pathway_genes <span class="ot">&lt;-</span> AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb174-30"><a href="pathway-overlap-analysis-of-genes.html#cb174-30" aria-hidden="true" tabindex="-1"></a>  org.Hs.eg.db, </span>
<span id="cb174-31"><a href="pathway-overlap-analysis-of-genes.html#cb174-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">keys =</span> <span class="fu">keys</span>(org.Hs.eg.db, <span class="st">&quot;GO&quot;</span>), </span>
<span id="cb174-32"><a href="pathway-overlap-analysis-of-genes.html#cb174-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">&quot;ENTREZID&quot;</span>, <span class="st">&quot;GO&quot;</span>, <span class="st">&quot;ONTOLOGY&quot;</span>), </span>
<span id="cb174-33"><a href="pathway-overlap-analysis-of-genes.html#cb174-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">keytype =</span> <span class="st">&quot;GO&quot;</span></span>
<span id="cb174-34"><a href="pathway-overlap-analysis-of-genes.html#cb174-34" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(GO))</span>
<span id="cb174-35"><a href="pathway-overlap-analysis-of-genes.html#cb174-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-36"><a href="pathway-overlap-analysis-of-genes.html#cb174-36" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate total gene counts per pathway and filter for at least 5 genes</span></span>
<span id="cb174-37"><a href="pathway-overlap-analysis-of-genes.html#cb174-37" aria-hidden="true" tabindex="-1"></a>total_genes_per_pathway <span class="ot">&lt;-</span> all_pathway_genes <span class="sc">%&gt;%</span></span>
<span id="cb174-38"><a href="pathway-overlap-analysis-of-genes.html#cb174-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(GO) <span class="sc">%&gt;%</span></span>
<span id="cb174-39"><a href="pathway-overlap-analysis-of-genes.html#cb174-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">Total_Genes =</span> <span class="fu">n_distinct</span>(ENTREZID)) <span class="sc">%&gt;%</span></span>
<span id="cb174-40"><a href="pathway-overlap-analysis-of-genes.html#cb174-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Total_Genes <span class="sc">&gt;=</span> <span class="dv">5</span>)</span>
<span id="cb174-41"><a href="pathway-overlap-analysis-of-genes.html#cb174-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-42"><a href="pathway-overlap-analysis-of-genes.html#cb174-42" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to calculate overlap ratio for each gene list</span></span>
<span id="cb174-43"><a href="pathway-overlap-analysis-of-genes.html#cb174-43" aria-hidden="true" tabindex="-1"></a>calculate_overlap_ratio <span class="ot">&lt;-</span> <span class="cf">function</span>(gene_list, pathway_genes, total_counts) {</span>
<span id="cb174-44"><a href="pathway-overlap-analysis-of-genes.html#cb174-44" aria-hidden="true" tabindex="-1"></a>  pathway_overlap <span class="ot">&lt;-</span> pathway_genes <span class="sc">%&gt;%</span></span>
<span id="cb174-45"><a href="pathway-overlap-analysis-of-genes.html#cb174-45" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(ENTREZID <span class="sc">%in%</span> gene_list) <span class="sc">%&gt;%</span></span>
<span id="cb174-46"><a href="pathway-overlap-analysis-of-genes.html#cb174-46" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(GO, ONTOLOGY) <span class="sc">%&gt;%</span></span>
<span id="cb174-47"><a href="pathway-overlap-analysis-of-genes.html#cb174-47" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarize</span>(<span class="at">Overlap_Count =</span> <span class="fu">n_distinct</span>(ENTREZID), <span class="at">.groups =</span> <span class="st">&quot;drop&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb174-48"><a href="pathway-overlap-analysis-of-genes.html#cb174-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(total_counts, <span class="at">by =</span> <span class="st">&quot;GO&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb174-49"><a href="pathway-overlap-analysis-of-genes.html#cb174-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Overlap_Ratio =</span> Overlap_Count <span class="sc">/</span> Total_Genes) <span class="sc">%&gt;%</span></span>
<span id="cb174-50"><a href="pathway-overlap-analysis-of-genes.html#cb174-50" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(Overlap_Ratio))</span>
<span id="cb174-51"><a href="pathway-overlap-analysis-of-genes.html#cb174-51" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb174-52"><a href="pathway-overlap-analysis-of-genes.html#cb174-52" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert to data frame to avoid issues with tibble</span></span>
<span id="cb174-53"><a href="pathway-overlap-analysis-of-genes.html#cb174-53" aria-hidden="true" tabindex="-1"></a>  pathway_overlap <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pathway_overlap)</span>
<span id="cb174-54"><a href="pathway-overlap-analysis-of-genes.html#cb174-54" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb174-55"><a href="pathway-overlap-analysis-of-genes.html#cb174-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(pathway_overlap)</span>
<span id="cb174-56"><a href="pathway-overlap-analysis-of-genes.html#cb174-56" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb174-57"><a href="pathway-overlap-analysis-of-genes.html#cb174-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-58"><a href="pathway-overlap-analysis-of-genes.html#cb174-58" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate overlap ratios for each gene list</span></span>
<span id="cb174-59"><a href="pathway-overlap-analysis-of-genes.html#cb174-59" aria-hidden="true" tabindex="-1"></a>overlap_results <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">names</span>(genes), <span class="cf">function</span>(list_name) {</span>
<span id="cb174-60"><a href="pathway-overlap-analysis-of-genes.html#cb174-60" aria-hidden="true" tabindex="-1"></a>  overlap_data <span class="ot">&lt;-</span> <span class="fu">calculate_overlap_ratio</span>(genes[[list_name]], all_pathway_genes, total_genes_per_pathway)</span>
<span id="cb174-61"><a href="pathway-overlap-analysis-of-genes.html#cb174-61" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb174-62"><a href="pathway-overlap-analysis-of-genes.html#cb174-62" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Safely select and rename columns</span></span>
<span id="cb174-63"><a href="pathway-overlap-analysis-of-genes.html#cb174-63" aria-hidden="true" tabindex="-1"></a>  overlap_data <span class="ot">&lt;-</span> overlap_data[, <span class="fu">c</span>(<span class="st">&quot;GO&quot;</span>, <span class="st">&quot;Overlap_Ratio&quot;</span>)]</span>
<span id="cb174-64"><a href="pathway-overlap-analysis-of-genes.html#cb174-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(overlap_data)[<span class="dv">2</span>] <span class="ot">&lt;-</span> list_name  <span class="co"># Dynamically assign the new column name</span></span>
<span id="cb174-65"><a href="pathway-overlap-analysis-of-genes.html#cb174-65" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb174-66"><a href="pathway-overlap-analysis-of-genes.html#cb174-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(overlap_data)</span>
<span id="cb174-67"><a href="pathway-overlap-analysis-of-genes.html#cb174-67" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb174-68"><a href="pathway-overlap-analysis-of-genes.html#cb174-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-69"><a href="pathway-overlap-analysis-of-genes.html#cb174-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-70"><a href="pathway-overlap-analysis-of-genes.html#cb174-70" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine results from all gene lists into a single data frame</span></span>
<span id="cb174-71"><a href="pathway-overlap-analysis-of-genes.html#cb174-71" aria-hidden="true" tabindex="-1"></a>combined_overlap_results <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(<span class="cf">function</span>(x, y) <span class="fu">merge</span>(x, y, <span class="at">by =</span> <span class="st">&quot;GO&quot;</span>, <span class="at">all =</span> <span class="cn">TRUE</span>), overlap_results)</span>
<span id="cb174-72"><a href="pathway-overlap-analysis-of-genes.html#cb174-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-73"><a href="pathway-overlap-analysis-of-genes.html#cb174-73" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the results as a CSV file</span></span>
<span id="cb174-74"><a href="pathway-overlap-analysis-of-genes.html#cb174-74" aria-hidden="true" tabindex="-1"></a><span class="co"># write.csv(combined_overlap_results, &quot;../03.out/figure5/pathway_overlap_analysis_result.csv&quot;, row.names = FALSE)</span></span>
<span id="cb174-75"><a href="pathway-overlap-analysis-of-genes.html#cb174-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-76"><a href="pathway-overlap-analysis-of-genes.html#cb174-76" aria-hidden="true" tabindex="-1"></a><span class="co"># Scatter plot function</span></span>
<span id="cb174-77"><a href="pathway-overlap-analysis-of-genes.html#cb174-77" aria-hidden="true" tabindex="-1"></a>plot_scatter <span class="ot">&lt;-</span> <span class="cf">function</span>(data, x_col, y_col, color, x_label, y_label) {</span>
<span id="cb174-78"><a href="pathway-overlap-analysis-of-genes.html#cb174-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggscatter</span>(</span>
<span id="cb174-79"><a href="pathway-overlap-analysis-of-genes.html#cb174-79" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> data,</span>
<span id="cb174-80"><a href="pathway-overlap-analysis-of-genes.html#cb174-80" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> x_col,</span>
<span id="cb174-81"><a href="pathway-overlap-analysis-of-genes.html#cb174-81" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> y_col,</span>
<span id="cb174-82"><a href="pathway-overlap-analysis-of-genes.html#cb174-82" aria-hidden="true" tabindex="-1"></a>    <span class="at">add =</span> <span class="st">&quot;reg.line&quot;</span>,                      <span class="co"># Add regression line</span></span>
<span id="cb174-83"><a href="pathway-overlap-analysis-of-genes.html#cb174-83" aria-hidden="true" tabindex="-1"></a>    <span class="at">conf.int =</span> <span class="cn">TRUE</span>,                       <span class="co"># Add confidence interval</span></span>
<span id="cb174-84"><a href="pathway-overlap-analysis-of-genes.html#cb174-84" aria-hidden="true" tabindex="-1"></a>    <span class="at">add.params =</span> <span class="fu">list</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;lightgray&quot;</span>), <span class="co"># Regression line params</span></span>
<span id="cb174-85"><a href="pathway-overlap-analysis-of-genes.html#cb174-85" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> color,                         <span class="co"># Scatter point color</span></span>
<span id="cb174-86"><a href="pathway-overlap-analysis-of-genes.html#cb174-86" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">2</span>                               <span class="co"># Point size</span></span>
<span id="cb174-87"><a href="pathway-overlap-analysis-of-genes.html#cb174-87" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb174-88"><a href="pathway-overlap-analysis-of-genes.html#cb174-88" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_cor</span>(<span class="at">method =</span> <span class="st">&quot;pearson&quot;</span>) <span class="sc">+</span></span>
<span id="cb174-89"><a href="pathway-overlap-analysis-of-genes.html#cb174-89" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> x_label, <span class="at">y =</span> y_label) <span class="sc">+</span>       <span class="co"># Axis labels</span></span>
<span id="cb174-90"><a href="pathway-overlap-analysis-of-genes.html#cb174-90" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span>                      <span class="co"># Use a classic theme</span></span>
<span id="cb174-91"><a href="pathway-overlap-analysis-of-genes.html#cb174-91" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb174-92"><a href="pathway-overlap-analysis-of-genes.html#cb174-92" aria-hidden="true" tabindex="-1"></a>      <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>),  <span class="co"># Customize facet labels</span></span>
<span id="cb174-93"><a href="pathway-overlap-analysis-of-genes.html#cb174-93" aria-hidden="true" tabindex="-1"></a>      <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>),<span class="co"># Center-align title</span></span>
<span id="cb174-94"><a href="pathway-overlap-analysis-of-genes.html#cb174-94" aria-hidden="true" tabindex="-1"></a>      <span class="co"># plot.margin = margin(5, 5, 5, 5),                    # Reduce plot margins</span></span>
<span id="cb174-95"><a href="pathway-overlap-analysis-of-genes.html#cb174-95" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.spacing =</span> <span class="fu">unit</span>(<span class="fl">0.5</span>, <span class="st">&quot;lines&quot;</span>)                   <span class="co"># Reduce spacing between facets</span></span>
<span id="cb174-96"><a href="pathway-overlap-analysis-of-genes.html#cb174-96" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb174-97"><a href="pathway-overlap-analysis-of-genes.html#cb174-97" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb174-98"><a href="pathway-overlap-analysis-of-genes.html#cb174-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-99"><a href="pathway-overlap-analysis-of-genes.html#cb174-99" aria-hidden="true" tabindex="-1"></a><span class="co"># Scatter plots between different gene overlap ratios</span></span>
<span id="cb174-100"><a href="pathway-overlap-analysis-of-genes.html#cb174-100" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">plot_scatter</span>(combined_overlap_results, <span class="st">&quot;TP53_gene&quot;</span>, <span class="st">&quot;RF8_gene&quot;</span>, <span class="st">&quot;#1B9E77&quot;</span>,</span>
<span id="cb174-101"><a href="pathway-overlap-analysis-of-genes.html#cb174-101" aria-hidden="true" tabindex="-1"></a>                   <span class="st">&quot;% Pathway Overlap </span><span class="sc">\n</span><span class="st">(TP53 mutation-affected genes)&quot;</span>, </span>
<span id="cb174-102"><a href="pathway-overlap-analysis-of-genes.html#cb174-102" aria-hidden="true" tabindex="-1"></a>                   <span class="st">&quot;% Pathway Overlap (RF8-correlated genes)&quot;</span>)</span>
<span id="cb174-103"><a href="pathway-overlap-analysis-of-genes.html#cb174-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-104"><a href="pathway-overlap-analysis-of-genes.html#cb174-104" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">plot_scatter</span>(combined_overlap_results, <span class="st">&quot;DNMT3A_gene&quot;</span>, <span class="st">&quot;RF8_gene&quot;</span>, <span class="st">&quot;#D95F02&quot;</span>,</span>
<span id="cb174-105"><a href="pathway-overlap-analysis-of-genes.html#cb174-105" aria-hidden="true" tabindex="-1"></a>                   <span class="st">&quot;% Pathway Overlap </span><span class="sc">\n</span><span class="st">(DNMT3A mutation-affected genes)&quot;</span>, </span>
<span id="cb174-106"><a href="pathway-overlap-analysis-of-genes.html#cb174-106" aria-hidden="true" tabindex="-1"></a>                   <span class="st">&quot;% Pathway Overlap (RF8-correlated genes)&quot;</span>)</span>
<span id="cb174-107"><a href="pathway-overlap-analysis-of-genes.html#cb174-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-108"><a href="pathway-overlap-analysis-of-genes.html#cb174-108" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">plot_scatter</span>(combined_overlap_results, <span class="st">&quot;IDH2_gene&quot;</span>, <span class="st">&quot;RF8_gene&quot;</span>, <span class="st">&quot;#7570B3&quot;</span>,</span>
<span id="cb174-109"><a href="pathway-overlap-analysis-of-genes.html#cb174-109" aria-hidden="true" tabindex="-1"></a>                   <span class="st">&quot;% Pathway Overlap </span><span class="sc">\n</span><span class="st">(IDH2 mutation-affected genes)&quot;</span>, </span>
<span id="cb174-110"><a href="pathway-overlap-analysis-of-genes.html#cb174-110" aria-hidden="true" tabindex="-1"></a>                   <span class="st">&quot;% Pathway Overlap (RF8-correlated genes)&quot;</span>)</span>
<span id="cb174-111"><a href="pathway-overlap-analysis-of-genes.html#cb174-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-112"><a href="pathway-overlap-analysis-of-genes.html#cb174-112" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> <span class="fu">plot_scatter</span>(combined_overlap_results, <span class="st">&quot;CEBPA_gene&quot;</span>, <span class="st">&quot;RF8_gene&quot;</span>, <span class="st">&quot;#E6AB02&quot;</span>,</span>
<span id="cb174-113"><a href="pathway-overlap-analysis-of-genes.html#cb174-113" aria-hidden="true" tabindex="-1"></a>                   <span class="st">&quot;% Pathway Overlap </span><span class="sc">\n</span><span class="st">(CEBPA mutation-affected genes)&quot;</span>, </span>
<span id="cb174-114"><a href="pathway-overlap-analysis-of-genes.html#cb174-114" aria-hidden="true" tabindex="-1"></a>                   <span class="st">&quot;% Pathway Overlap (RF8-correlated genes)&quot;</span>)</span>
<span id="cb174-115"><a href="pathway-overlap-analysis-of-genes.html#cb174-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-116"><a href="pathway-overlap-analysis-of-genes.html#cb174-116" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine plots using patchwork</span></span>
<span id="cb174-117"><a href="pathway-overlap-analysis-of-genes.html#cb174-117" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> (p1 <span class="sc">|</span> p2) <span class="sc">/</span> (p3 <span class="sc">|</span> p4)</span>
<span id="cb174-118"><a href="pathway-overlap-analysis-of-genes.html#cb174-118" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(plots)</span></code></pre></div>
<p><img src="figures/unnamed-chunk-79-1.png" width="672" /></p>
<div class="sourceCode" id="cb175"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb175-1"><a href="pathway-overlap-analysis-of-genes.html#cb175-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggcorrplot)</span>
<span id="cb175-2"><a href="pathway-overlap-analysis-of-genes.html#cb175-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare the correlation matrix from the combined_overlap_results</span></span>
<span id="cb175-3"><a href="pathway-overlap-analysis-of-genes.html#cb175-3" aria-hidden="true" tabindex="-1"></a>cor_matrix <span class="ot">&lt;-</span> <span class="fu">cor</span>(combined_overlap_results[,<span class="sc">-</span><span class="dv">1</span>], <span class="at">use =</span> <span class="st">&quot;pairwise.complete.obs&quot;</span>)</span>
<span id="cb175-4"><a href="pathway-overlap-analysis-of-genes.html#cb175-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb175-5"><a href="pathway-overlap-analysis-of-genes.html#cb175-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the correlation heatmap</span></span>
<span id="cb175-6"><a href="pathway-overlap-analysis-of-genes.html#cb175-6" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggcorrplot</span>(</span>
<span id="cb175-7"><a href="pathway-overlap-analysis-of-genes.html#cb175-7" aria-hidden="true" tabindex="-1"></a>  cor_matrix,</span>
<span id="cb175-8"><a href="pathway-overlap-analysis-of-genes.html#cb175-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">&quot;circle&quot;</span>,       <span class="co"># Use circle style for visualization</span></span>
<span id="cb175-9"><a href="pathway-overlap-analysis-of-genes.html#cb175-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">type =</span> <span class="st">&quot;upper&quot;</span>,          <span class="co"># Show only the upper triangle</span></span>
<span id="cb175-10"><a href="pathway-overlap-analysis-of-genes.html#cb175-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">lab =</span> <span class="cn">TRUE</span>,              <span class="co"># Add correlation coefficient labels</span></span>
<span id="cb175-11"><a href="pathway-overlap-analysis-of-genes.html#cb175-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">lab_size =</span> <span class="dv">4</span>,            <span class="co"># Label size</span></span>
<span id="cb175-12"><a href="pathway-overlap-analysis-of-genes.html#cb175-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">colors =</span> <span class="fu">c</span>(<span class="st">&quot;#6D9EC1&quot;</span>, <span class="st">&quot;white&quot;</span>, <span class="st">&quot;#E46726&quot;</span>),  <span class="co"># Color gradient (blue-white-red)</span></span>
<span id="cb175-13"><a href="pathway-overlap-analysis-of-genes.html#cb175-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">title =</span> <span class="st">&quot;Correlation of Pathway Overlap Ratios&quot;</span>,</span>
<span id="cb175-14"><a href="pathway-overlap-analysis-of-genes.html#cb175-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">outline.color =</span> <span class="st">&quot;black&quot;</span>  <span class="co"># Outline color for circles</span></span>
<span id="cb175-15"><a href="pathway-overlap-analysis-of-genes.html#cb175-15" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb175-16"><a href="pathway-overlap-analysis-of-genes.html#cb175-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb175-17"><a href="pathway-overlap-analysis-of-genes.html#cb175-17" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p)</span></code></pre></div>
<p><img src="figures/unnamed-chunk-80-1.png" width="672" /></p>

</div>
<!-- </div> -->
            </section>

          </div>
        </div>
      </div>
<a href="venn-diagram-for-target-genes.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="figure-6.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/05-results5.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
