<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4.2 Predict Class Labels on the Test Set and Plot ROC | Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning</title>
  <meta name="description" content="This is a book-style code sharing." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="4.2 Predict Class Labels on the Test Set and Plot ROC | Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a book-style code sharing." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4.2 Predict Class Labels on the Test Set and Plot ROC | Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning" />
  
  <meta name="twitter:description" content="This is a book-style code sharing." />
  

<meta name="author" content="Peng Jin, Qiqi Jin, Kankan Wang" />


<meta name="date" content="2025-07-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="benchmark-ml-methods-for-prediction.html"/>
<link rel="next" href="auc-and-auprc-heatmap-for-the-entire-and-test-sets.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Background</a></li>
<li class="chapter" data-level="2" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>2</b> Figure 1</a>
<ul>
<li class="chapter" data-level="2.1" data-path="process-the-data-from-functional_precision_medicine_tumor_board_aml.html"><a href="process-the-data-from-functional_precision_medicine_tumor_board_aml.html"><i class="fa fa-check"></i><b>2.1</b> Process the data from Functional_Precision_Medicine_Tumor_Board_AML</a></li>
<li class="chapter" data-level="2.2" data-path="functional_precision_medicine_tumor_board_aml.html"><a href="functional_precision_medicine_tumor_board_aml.html"><i class="fa fa-check"></i><b>2.2</b> Functional_Precision_Medicine_Tumor_Board_AML</a></li>
<li class="chapter" data-level="2.3" data-path="boxplot-for-drug-sensitivity-between-resistant-and-sensitive.html"><a href="boxplot-for-drug-sensitivity-between-resistant-and-sensitive.html"><i class="fa fa-check"></i><b>2.3</b> Boxplot for drug sensitivity between resistant and sensitive</a></li>
<li class="chapter" data-level="2.4" data-path="de-analysis-functional_medicine.html"><a href="de-analysis-functional_medicine.html"><i class="fa fa-check"></i><b>2.4</b> DE analysis Functional_medicine</a></li>
<li class="chapter" data-level="2.5" data-path="valcanoplot-plot-ex-vivo.html"><a href="valcanoplot-plot-ex-vivo.html"><i class="fa fa-check"></i><b>2.5</b> Valcanoplot plot (Ex vivo)</a></li>
<li class="chapter" data-level="2.6" data-path="heatmap-for-representative-de-genes.html"><a href="heatmap-for-representative-de-genes.html"><i class="fa fa-check"></i><b>2.6</b> Heatmap for representative DE genes</a></li>
<li class="chapter" data-level="2.7" data-path="cibersortx-results-visualization-ex-vivo.html"><a href="cibersortx-results-visualization-ex-vivo.html"><i class="fa fa-check"></i><b>2.7</b> CIBERSORTx Results Visualization (ex vivo)</a></li>
<li class="chapter" data-level="2.8" data-path="cibersortx-results-visualization-in-vivo.html"><a href="cibersortx-results-visualization-in-vivo.html"><i class="fa fa-check"></i><b>2.8</b> CIBERSORTx Results Visualization (in vivo)</a></li>
<li class="chapter" data-level="2.9" data-path="ssgsea-analysis-in-vivo.html"><a href="ssgsea-analysis-in-vivo.html"><i class="fa fa-check"></i><b>2.9</b> ssGSEA analysis in vivo</a></li>
<li class="chapter" data-level="2.10" data-path="go-enrichment-dot-plot.html"><a href="go-enrichment-dot-plot.html"><i class="fa fa-check"></i><b>2.10</b> GO Enrichment Dot Plot</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>3</b> Figure 2</a>
<ul>
<li class="chapter" data-level="3.1" data-path="de-analysis-between-in-vivo-sensitive-and-resistant.html"><a href="de-analysis-between-in-vivo-sensitive-and-resistant.html"><i class="fa fa-check"></i><b>3.1</b> DE analysis between in vivo sensitive and resistant</a></li>
<li class="chapter" data-level="3.2" data-path="de-analysis-rjaml-azaven-cohort.html"><a href="de-analysis-rjaml-azaven-cohort.html"><i class="fa fa-check"></i><b>3.2</b> DE analysis RJAML AZA+VEN cohort</a></li>
<li class="chapter" data-level="3.3" data-path="valcanoplot-plot-ex-vivo-1.html"><a href="valcanoplot-plot-ex-vivo-1.html"><i class="fa fa-check"></i><b>3.3</b> Valcanoplot plot (Ex vivo)</a></li>
<li class="chapter" data-level="3.4" data-path="valcanoplot-plot-in-vivo.html"><a href="valcanoplot-plot-in-vivo.html"><i class="fa fa-check"></i><b>3.4</b> Valcanoplot plot (In vivo)</a></li>
<li class="chapter" data-level="3.5" data-path="venn-plot.html"><a href="venn-plot.html"><i class="fa fa-check"></i><b>3.5</b> Venn plot</a></li>
<li class="chapter" data-level="3.6" data-path="integration-of-degs-ex-vivo-and-in-vivo.html"><a href="integration-of-degs-ex-vivo-and-in-vivo.html"><i class="fa fa-check"></i><b>3.6</b> Integration of DEGs ex vivo and in vivo</a></li>
<li class="chapter" data-level="3.7" data-path="gsea-analysis-for-crispri-drug-sensitivity.html"><a href="gsea-analysis-for-crispri-drug-sensitivity.html"><i class="fa fa-check"></i><b>3.7</b> GSEA analysis for CRISPRi drug sensitivity</a></li>
<li class="chapter" data-level="3.8" data-path="differential-expression-analysis-for-shpink1.html"><a href="differential-expression-analysis-for-shpink1.html"><i class="fa fa-check"></i><b>3.8</b> Differential Expression Analysis for shPINK1</a></li>
<li class="chapter" data-level="3.9" data-path="de-analysis-in-cell-lines.html"><a href="de-analysis-in-cell-lines.html"><i class="fa fa-check"></i><b>3.9</b> DE analysis in cell lines</a></li>
<li class="chapter" data-level="3.10" data-path="valcano-plots-molm13.html"><a href="valcano-plots-molm13.html"><i class="fa fa-check"></i><b>3.10</b> Valcano plots (MOLM13)</a></li>
<li class="chapter" data-level="3.11" data-path="valcano-plots-oa3.html"><a href="valcano-plots-oa3.html"><i class="fa fa-check"></i><b>3.11</b> Valcano plots (OA3)</a></li>
<li class="chapter" data-level="3.12" data-path="gsea-analysis-molm13.html"><a href="gsea-analysis-molm13.html"><i class="fa fa-check"></i><b>3.12</b> GSEA analysis (MOLM13)</a></li>
<li class="chapter" data-level="3.13" data-path="gsea-analysis-oa3.html"><a href="gsea-analysis-oa3.html"><i class="fa fa-check"></i><b>3.13</b> GSEA analysis (OA3)</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>4</b> Figure 3</a>
<ul>
<li class="chapter" data-level="4.1" data-path="benchmark-ml-methods-for-prediction.html"><a href="benchmark-ml-methods-for-prediction.html"><i class="fa fa-check"></i><b>4.1</b> Benchmark ML Methods for Prediction</a></li>
<li class="chapter" data-level="4.2" data-path="predict-class-labels-on-the-test-set-and-plot-roc.html"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html"><i class="fa fa-check"></i><b>4.2</b> Predict Class Labels on the Test Set and Plot ROC</a></li>
<li class="chapter" data-level="4.3" data-path="auc-and-auprc-heatmap-for-the-entire-and-test-sets.html"><a href="auc-and-auprc-heatmap-for-the-entire-and-test-sets.html"><i class="fa fa-check"></i><b>4.3</b> AUC and AUPRC Heatmap for the Entire and Test Sets</a></li>
<li class="chapter" data-level="4.4" data-path="model-performance-plot.html"><a href="model-performance-plot.html"><i class="fa fa-check"></i><b>4.4</b> Model Performance Plot</a></li>
<li class="chapter" data-level="4.5" data-path="feature-importance.html"><a href="feature-importance.html"><i class="fa fa-check"></i><b>4.5</b> Feature importance</a></li>
<li class="chapter" data-level="4.6" data-path="heatmap-for-confusion-matrices-defaut-cutoff-0.html"><a href="heatmap-for-confusion-matrices-defaut-cutoff-0.html"><i class="fa fa-check"></i><b>4.6</b> Heatmap for Confusion Matrices (defaut cutoff 0.5)</a></li>
<li class="chapter" data-level="4.7" data-path="barplot-for-predicted-response-0.html"><a href="barplot-for-predicted-response-0.html"><i class="fa fa-check"></i><b>4.7</b> Barplot for predicted Response (0.5)</a></li>
<li class="chapter" data-level="4.8" data-path="heatmap-for-confusion-matrices-optimal-cutoff-0.html"><a href="heatmap-for-confusion-matrices-optimal-cutoff-0.html"><i class="fa fa-check"></i><b>4.8</b> Heatmap for Confusion Matrices (optimal cutoff 0.630)</a></li>
<li class="chapter" data-level="4.9" data-path="barplot-for-predicted-response-0.html"><a href="barplot-for-predicted-response-0.html#barplot-for-predicted-response-0.630"><i class="fa fa-check"></i><b>4.9</b> Barplot for predicted Response (0.630)</a></li>
<li class="chapter" data-level="4.10" data-path="density-plot-for-response-probability.html"><a href="density-plot-for-response-probability.html"><i class="fa fa-check"></i><b>4.10</b> Density Plot for Response Probability</a></li>
<li class="chapter" data-level="4.11" data-path="violin-and-boxplots-for-response-probability-rf8.html"><a href="violin-and-boxplots-for-response-probability-rf8.html"><i class="fa fa-check"></i><b>4.11</b> Violin and Boxplots for Response Probability (RF8)</a></li>
<li class="chapter" data-level="4.12" data-path="barplot-for-response.html"><a href="barplot-for-response.html"><i class="fa fa-check"></i><b>4.12</b> Barplot for Response</a></li>
<li class="chapter" data-level="4.13" data-path="rf8-for-mdr-status.html"><a href="rf8-for-mdr-status.html"><i class="fa fa-check"></i><b>4.13</b> RF8 for MDR status</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>5</b> Figure 4</a>
<ul>
<li class="chapter" data-level="5.1" data-path="validation-of-rf8-model-rjaml-cohort-2.html"><a href="validation-of-rf8-model-rjaml-cohort-2.html"><i class="fa fa-check"></i><b>5.1</b> Validation of RF8 model (RJAML-Cohort 2)</a></li>
<li class="chapter" data-level="5.2" data-path="validation-of-rf8-model-fpmtb.html"><a href="validation-of-rf8-model-fpmtb.html"><i class="fa fa-check"></i><b>5.2</b> Validation of RF8 model (FPMTB)</a></li>
<li class="chapter" data-level="5.3" data-path="validation-of-rf8-model-beataml.html"><a href="validation-of-rf8-model-beataml.html"><i class="fa fa-check"></i><b>5.3</b> Validation of RF8 model (BeatAML)</a></li>
<li class="chapter" data-level="5.4" data-path="violin-and-boxplots-for-response-probability-rf8-1.html"><a href="violin-and-boxplots-for-response-probability-rf8-1.html"><i class="fa fa-check"></i><b>5.4</b> Violin and Boxplots for Response Probability (RF8)</a></li>
<li class="chapter" data-level="5.5" data-path="rf8-score-among-clusters.html"><a href="rf8-score-among-clusters.html"><i class="fa fa-check"></i><b>5.5</b> RF8 score among clusters</a></li>
<li class="chapter" data-level="5.6" data-path="integration-of-ex-vivo-drug-response-in-beataml.html"><a href="integration-of-ex-vivo-drug-response-in-beataml.html"><i class="fa fa-check"></i><b>5.6</b> Integration of ex vivo drug-response in BeatAML</a></li>
<li class="chapter" data-level="5.7" data-path="beataml-waves-1-4-heatmap-based-on-drug-profiling.html"><a href="beataml-waves-1-4-heatmap-based-on-drug-profiling.html"><i class="fa fa-check"></i><b>5.7</b> BeatAML Waves 1-4 heatmap based on drug profiling</a></li>
<li class="chapter" data-level="5.8" data-path="boxplot-for-drug-sensitivitity-between-resistant-and-sensitive.html"><a href="boxplot-for-drug-sensitivitity-between-resistant-and-sensitive.html"><i class="fa fa-check"></i><b>5.8</b> Boxplot for drug sensitivitity between resistant and sensitive</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>6</b> Figure 5</a>
<ul>
<li class="chapter" data-level="6.1" data-path="integration-with-aml-genomics.html"><a href="integration-with-aml-genomics.html"><i class="fa fa-check"></i><b>6.1</b> Integration with AML genomics</a></li>
<li class="chapter" data-level="6.2" data-path="mutation-pairs.html"><a href="mutation-pairs.html"><i class="fa fa-check"></i><b>6.2</b> Mutation pairs</a></li>
<li class="chapter" data-level="6.3" data-path="complexheatmap-for-mutation.html"><a href="complexheatmap-for-mutation.html"><i class="fa fa-check"></i><b>6.3</b> Complexheatmap for mutation</a></li>
<li class="chapter" data-level="6.4" data-path="fisher-test.html"><a href="fisher-test.html"><i class="fa fa-check"></i><b>6.4</b> Fisher test</a></li>
<li class="chapter" data-level="6.5" data-path="barplot-in-rjaml-cohort.html"><a href="barplot-in-rjaml-cohort.html"><i class="fa fa-check"></i><b>6.5</b> Barplot in RJAML cohort</a></li>
<li class="chapter" data-level="6.6" data-path="table-one-for-multi-cohorts.html"><a href="table-one-for-multi-cohorts.html"><i class="fa fa-check"></i><b>6.6</b> Table one for multi-cohorts</a></li>
<li class="chapter" data-level="6.7" data-path="barplot-for-real-response.html"><a href="barplot-for-real-response.html"><i class="fa fa-check"></i><b>6.7</b> Barplot for real response</a></li>
<li class="chapter" data-level="6.8" data-path="multivariable-logisticreg.html"><a href="multivariable-logisticreg.html"><i class="fa fa-check"></i><b>6.8</b> Multivariable LogisticReg</a></li>
<li class="chapter" data-level="6.9" data-path="identification-of-rf8-correlated-genes.html"><a href="identification-of-rf8-correlated-genes.html"><i class="fa fa-check"></i><b>6.9</b> Identification of RF8-correlated genes</a></li>
<li class="chapter" data-level="6.10" data-path="differential-expression-analysis-for-mut-vs.-wt.html"><a href="differential-expression-analysis-for-mut-vs.-wt.html"><i class="fa fa-check"></i><b>6.10</b> Differential Expression Analysis for Mut vs. WT</a></li>
<li class="chapter" data-level="6.11" data-path="venn-diagram-for-target-genes.html"><a href="venn-diagram-for-target-genes.html"><i class="fa fa-check"></i><b>6.11</b> Venn Diagram for target genes</a></li>
<li class="chapter" data-level="6.12" data-path="pathway-overlap-analysis-of-genes.html"><a href="pathway-overlap-analysis-of-genes.html"><i class="fa fa-check"></i><b>6.12</b> Pathway overlap analysis of genes</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="figure-6.html"><a href="figure-6.html"><i class="fa fa-check"></i><b>7</b> Figure 6</a>
<ul>
<li class="chapter" data-level="7.1" data-path="relationship-between-rf8-score-vs.-venaza-response-probability.html"><a href="relationship-between-rf8-score-vs.-venaza-response-probability.html"><i class="fa fa-check"></i><b>7.1</b> Relationship between RF8 score vs. VEN/AZA response probability</a></li>
<li class="chapter" data-level="7.2" data-path="monotonic-response-prediction-plot.html"><a href="monotonic-response-prediction-plot.html"><i class="fa fa-check"></i><b>7.2</b> Monotonic Response Prediction Plot</a></li>
<li class="chapter" data-level="7.3" data-path="survival-plot.html"><a href="survival-plot.html"><i class="fa fa-check"></i><b>7.3</b> Survival plot</a></li>
<li class="chapter" data-level="7.4" data-path="brier-score-analysis.html"><a href="brier-score-analysis.html"><i class="fa fa-check"></i><b>7.4</b> Brier Score Analysis</a></li>
<li class="chapter" data-level="7.5" data-path="c-index-comparison.html"><a href="c-index-comparison.html"><i class="fa fa-check"></i><b>7.5</b> C-index Comparison</a></li>
<li class="chapter" data-level="7.6" data-path="km-plots-for-rjaml-cohort-2.html"><a href="km-plots-for-rjaml-cohort-2.html"><i class="fa fa-check"></i><b>7.6</b> KM-Plots for RJAML Cohort 2</a></li>
<li class="chapter" data-level="7.7" data-path="barplot-for-crcri.html"><a href="barplot-for-crcri.html"><i class="fa fa-check"></i><b>7.7</b> Barplot for CR/CRi</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="predict-class-labels-on-the-test-set-and-plot-roc" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Predict Class Labels on the Test Set and Plot ROC<a href="predict-class-labels-on-the-test-set-and-plot-roc.html#predict-class-labels-on-the-test-set-and-plot-roc" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="do">#########################################################</span></span>
<span id="cb68-2"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb68-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Predict Class Labels on the Test Set and Plot ROC</span></span>
<span id="cb68-3"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb68-3" aria-hidden="true" tabindex="-1"></a><span class="do">#########################################################</span></span>
<span id="cb68-4"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb68-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-5"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb68-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pROC)</span>
<span id="cb68-6"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb68-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(PRROC)</span>
<span id="cb68-7"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb68-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-8"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb68-8" aria-hidden="true" tabindex="-1"></a>rf_tuned <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../03.out/figure3/AZA_VEN_RandomForest8.rds&quot;</span>)</span>
<span id="cb68-9"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb68-9" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">&quot;../02.data/TrainingTest_RJAML_Cohort1_84_genes.xlsx&quot;</span>)</span>
<span id="cb68-10"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb68-10" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> data[, <span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb68-11"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb68-11" aria-hidden="true" tabindex="-1"></a>data<span class="sc">$</span>Response <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(data<span class="sc">$</span>Response)</span>
<span id="cb68-12"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb68-12" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">12345</span>)</span>
<span id="cb68-13"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb68-13" aria-hidden="true" tabindex="-1"></a>trainIndex <span class="ot">&lt;-</span> <span class="fu">createDataPartition</span>(data<span class="sc">$</span>Response, <span class="at">p =</span> <span class="fl">0.7</span>, <span class="at">list =</span> <span class="cn">FALSE</span>)</span>
<span id="cb68-14"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb68-14" aria-hidden="true" tabindex="-1"></a>trainData <span class="ot">&lt;-</span> data[trainIndex, ]</span>
<span id="cb68-15"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb68-15" aria-hidden="true" tabindex="-1"></a>testData <span class="ot">&lt;-</span> data[<span class="sc">-</span>trainIndex, ]</span>
<span id="cb68-16"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb68-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-17"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb68-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Predict class labels and probabilities on the test set</span></span>
<span id="cb68-18"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb68-18" aria-hidden="true" tabindex="-1"></a>test_predictions <span class="ot">&lt;-</span> <span class="fu">predict</span>(rf_tuned, <span class="at">newdata =</span> testData)</span>
<span id="cb68-19"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb68-19" aria-hidden="true" tabindex="-1"></a>test_probabilities <span class="ot">&lt;-</span> <span class="fu">predict</span>(rf_tuned, <span class="at">newdata =</span> testData, <span class="at">type =</span> <span class="st">&quot;prob&quot;</span>)</span>
<span id="cb68-20"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb68-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-21"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb68-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate confusion matrix</span></span>
<span id="cb68-22"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb68-22" aria-hidden="true" tabindex="-1"></a><span class="fu">confusionMatrix</span>(test_predictions, testData<span class="sc">$</span>Response)</span></code></pre></div>
<pre><code>## Confusion Matrix and Statistics
## 
##           Reference
## Prediction CR NonCR
##      CR     3     1
##      NonCR 20     8
##                                           
##                Accuracy : 0.3438          
##                  95% CI : (0.1857, 0.5319)
##     No Information Rate : 0.7188          
##     P-Value [Acc &gt; NIR] : 1               
##                                           
##                   Kappa : 0.0118          
##                                           
##  Mcnemar&#39;s Test P-Value : 8.568e-05       
##                                           
##             Sensitivity : 0.13043         
##             Specificity : 0.88889         
##          Pos Pred Value : 0.75000         
##          Neg Pred Value : 0.28571         
##              Prevalence : 0.71875         
##          Detection Rate : 0.09375         
##    Detection Prevalence : 0.12500         
##       Balanced Accuracy : 0.50966         
##                                           
##        &#39;Positive&#39; Class : CR              
## </code></pre>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add predicted probabilities for CR</span></span>
<span id="cb70-2"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb70-2" aria-hidden="true" tabindex="-1"></a>testData<span class="sc">$</span>RF8.prob.CR <span class="ot">&lt;-</span> test_probabilities[, <span class="st">&quot;CR&quot;</span>]</span>
<span id="cb70-3"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb70-3" aria-hidden="true" tabindex="-1"></a><span class="co"># write.csv(testData, &quot;../03.out/figure3/RF8_RJAML_cohort1_test.csv&quot;)</span></span>
<span id="cb70-4"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb70-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-5"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb70-5" aria-hidden="true" tabindex="-1"></a>testData <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../03.out/figure3/RF8_RJAML_cohort1_test.csv&quot;</span>, <span class="at">header =</span> T)</span>
<span id="cb70-6"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb70-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot ROC curve for the test set</span></span>
<span id="cb70-7"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb70-7" aria-hidden="true" tabindex="-1"></a>roc_obj <span class="ot">&lt;-</span> <span class="fu">roc</span>(testData<span class="sc">$</span>Response, testData<span class="sc">$</span>RF8.prob.CR)</span>
<span id="cb70-8"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb70-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-9"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb70-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Save ROC plot to PDF</span></span>
<span id="cb70-10"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb70-10" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">plot</span>(<span class="fu">smooth</span>(roc_obj), <span class="at">legacy.axes =</span> <span class="cn">TRUE</span>, <span class="at">lwd =</span> <span class="dv">2</span>,                          </span>
<span id="cb70-11"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb70-11" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">&quot;Test set (N = 32)&quot;</span>,</span>
<span id="cb70-12"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb70-12" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">&quot;False Positive Rate (1 - Specificity)&quot;</span>, </span>
<span id="cb70-13"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb70-13" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">&quot;True Positive Rate (Sensitivity)&quot;</span>, </span>
<span id="cb70-14"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb70-14" aria-hidden="true" tabindex="-1"></a>     <span class="at">cex.main =</span> <span class="fl">1.5</span>, <span class="at">cex.lab =</span> <span class="fl">1.2</span>, <span class="at">cex.axis =</span> <span class="fl">1.1</span>)</span>
<span id="cb70-15"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb70-15" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="at">by =</span> <span class="fl">0.1</span>), <span class="at">v =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="at">by =</span> <span class="fl">0.1</span>), <span class="at">col =</span> <span class="st">&quot;gray80&quot;</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb70-16"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb70-16" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;bottomright&quot;</span>, <span class="at">legend =</span> <span class="fu">c</span>(<span class="fu">paste</span>(<span class="st">&quot;AUC =&quot;</span>, <span class="fu">round</span>(roc_obj<span class="sc">$</span>auc, <span class="dv">2</span>))),</span>
<span id="cb70-17"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb70-17" aria-hidden="true" tabindex="-1"></a>       <span class="at">col =</span> <span class="st">&quot;#1f77b4&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">cex =</span> <span class="fl">1.2</span>)</span></code></pre></div>
<p><img src="figures/unnamed-chunk-41-1.png" width="672" /></p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p1)</span></code></pre></div>
<pre><code>## 
## Call:
## smooth.roc(roc = roc_obj)
## 
## Data: testData$RF8.prob.CR in 23 controls (testData$Response CR) &gt; 9 cases (testData$Response NonCR).
## Smoothing: binormal 
## Area under the curve: 0.8475</code></pre>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute Precision-Recall curve and AUPRC for the test set</span></span>
<span id="cb73-2"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb73-2" aria-hidden="true" tabindex="-1"></a>testData<span class="sc">$</span>Response <span class="ot">&lt;-</span> <span class="fu">factor</span>(testData<span class="sc">$</span>Response, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;NonCR&quot;</span>, <span class="st">&quot;CR&quot;</span>))</span>
<span id="cb73-3"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb73-3" aria-hidden="true" tabindex="-1"></a>response_binary <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(testData<span class="sc">$</span>Response) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb73-4"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb73-4" aria-hidden="true" tabindex="-1"></a>pr_curve <span class="ot">&lt;-</span> <span class="fu">pr.curve</span>(<span class="at">scores.class0 =</span> testData<span class="sc">$</span>RF8.prob.CR, <span class="at">weights.class0 =</span> response_binary, <span class="at">curve =</span> <span class="cn">TRUE</span>)</span>
<span id="cb73-5"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb73-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-6"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb73-6" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">plot</span>(pr_curve, <span class="at">main =</span> <span class="st">&quot;Test set (N = 32)&quot;</span>, <span class="at">color =</span> <span class="st">&quot;#1f77b4&quot;</span>, <span class="at">auc.main =</span> <span class="cn">FALSE</span>)</span>
<span id="cb73-7"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb73-7" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="at">by =</span> <span class="fl">0.1</span>), <span class="at">v =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="at">by =</span> <span class="fl">0.1</span>), <span class="at">col =</span> <span class="st">&quot;gray80&quot;</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb73-8"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb73-8" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;bottomright&quot;</span>, <span class="at">legend =</span> <span class="fu">c</span>(<span class="fu">paste</span>(<span class="st">&quot;AUPRC =&quot;</span>, <span class="fu">round</span>(pr_curve<span class="sc">$</span>auc.integral, <span class="dv">2</span>))),</span>
<span id="cb73-9"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb73-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">col =</span> <span class="st">&quot;#1f77b4&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">cex =</span> <span class="fl">1.2</span>)</span></code></pre></div>
<p><img src="figures/unnamed-chunk-41-2.png" width="672" /></p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p2)</span></code></pre></div>
<pre><code>## NULL</code></pre>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="co"># #########################################################</span></span>
<span id="cb76-2"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-2" aria-hidden="true" tabindex="-1"></a><span class="co"># ## Predict Class Labels on the Training Set and Plot ROC</span></span>
<span id="cb76-3"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-3" aria-hidden="true" tabindex="-1"></a><span class="co"># #########################################################</span></span>
<span id="cb76-4"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-4" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb76-5"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-5" aria-hidden="true" tabindex="-1"></a><span class="co"># # Predict class labels and probabilities on the training set</span></span>
<span id="cb76-6"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-6" aria-hidden="true" tabindex="-1"></a><span class="co"># train_predictions &lt;- predict(rf_tuned, newdata = trainData)</span></span>
<span id="cb76-7"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-7" aria-hidden="true" tabindex="-1"></a><span class="co"># train_probabilities &lt;- predict(rf_tuned, newdata = trainData, type = &quot;prob&quot;)</span></span>
<span id="cb76-8"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-8" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb76-9"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-9" aria-hidden="true" tabindex="-1"></a><span class="co"># # Generate confusion matrix</span></span>
<span id="cb76-10"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-10" aria-hidden="true" tabindex="-1"></a><span class="co"># confusionMatrix(train_predictions, trainData$Response)</span></span>
<span id="cb76-11"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-11" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb76-12"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-12" aria-hidden="true" tabindex="-1"></a><span class="co"># # Add predicted probabilities for CR</span></span>
<span id="cb76-13"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-13" aria-hidden="true" tabindex="-1"></a><span class="co"># trainData$RF8.prob.CR &lt;- train_probabilities[, &quot;CR&quot;]</span></span>
<span id="cb76-14"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-14" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb76-15"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-15" aria-hidden="true" tabindex="-1"></a><span class="co"># # Plot ROC curve for the training set</span></span>
<span id="cb76-16"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-16" aria-hidden="true" tabindex="-1"></a><span class="co"># roc_obj_train &lt;- roc(trainData$Response, trainData$RF8.prob.CR)</span></span>
<span id="cb76-17"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-17" aria-hidden="true" tabindex="-1"></a><span class="co"># roc_obj_train$auc</span></span>
<span id="cb76-18"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-18" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb76-19"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-19" aria-hidden="true" tabindex="-1"></a><span class="co"># # Save ROC plot to PDF</span></span>
<span id="cb76-20"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-20" aria-hidden="true" tabindex="-1"></a><span class="co"># pdf(&quot;figure3/RF8_Training_ROC.pdf&quot;, width = 4.2, height = 4)</span></span>
<span id="cb76-21"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-21" aria-hidden="true" tabindex="-1"></a><span class="co"># plot(roc_obj_train, legacy.axes = TRUE, lwd = 2,                          </span></span>
<span id="cb76-22"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-22" aria-hidden="true" tabindex="-1"></a><span class="co">#      main = &quot;Training set (N = 78)&quot;,</span></span>
<span id="cb76-23"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-23" aria-hidden="true" tabindex="-1"></a><span class="co">#      xlab = &quot;False Positive Rate (1 - Specificity)&quot;, </span></span>
<span id="cb76-24"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-24" aria-hidden="true" tabindex="-1"></a><span class="co">#      ylab = &quot;True Positive Rate (Sensitivity)&quot;, </span></span>
<span id="cb76-25"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-25" aria-hidden="true" tabindex="-1"></a><span class="co">#      cex.main = 1.5, cex.lab = 1.2, cex.axis = 1.1)</span></span>
<span id="cb76-26"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-26" aria-hidden="true" tabindex="-1"></a><span class="co"># abline(h = seq(0, 1, by = 0.1), v = seq(0, 1, by = 0.1), col = &quot;gray80&quot;, lty = 2)</span></span>
<span id="cb76-27"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-27" aria-hidden="true" tabindex="-1"></a><span class="co"># legend(&quot;bottomright&quot;, legend = c(paste(&quot;AUC =&quot;, round(roc_obj_train$auc, 2))),</span></span>
<span id="cb76-28"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-28" aria-hidden="true" tabindex="-1"></a><span class="co">#        col = &quot;#1f77b4&quot;, lwd = 2, bty = &quot;n&quot;, cex = 1.2)</span></span>
<span id="cb76-29"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-29" aria-hidden="true" tabindex="-1"></a><span class="co"># dev.off()</span></span>
<span id="cb76-30"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-31"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-31" aria-hidden="true" tabindex="-1"></a><span class="do">#########################################################</span></span>
<span id="cb76-32"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-32" aria-hidden="true" tabindex="-1"></a><span class="do">## Predict Class Labels on the Entire Dataset and Plot ROC</span></span>
<span id="cb76-33"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-33" aria-hidden="true" tabindex="-1"></a><span class="do">#########################################################</span></span>
<span id="cb76-34"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-35"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-35" aria-hidden="true" tabindex="-1"></a><span class="co"># # Predict class labels and probabilities on the entire set</span></span>
<span id="cb76-36"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-36" aria-hidden="true" tabindex="-1"></a><span class="co"># entire_predictions &lt;- predict(rf_tuned, newdata = data_selected)</span></span>
<span id="cb76-37"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-37" aria-hidden="true" tabindex="-1"></a><span class="co"># entire_probabilities &lt;- predict(rf_tuned, newdata = data_selected, type = &quot;prob&quot;)</span></span>
<span id="cb76-38"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-38" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb76-39"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-39" aria-hidden="true" tabindex="-1"></a><span class="co"># # Generate confusion matrix</span></span>
<span id="cb76-40"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-40" aria-hidden="true" tabindex="-1"></a><span class="co"># confusionMatrix(entire_predictions, data_selected$Response)</span></span>
<span id="cb76-41"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-41" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb76-42"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-42" aria-hidden="true" tabindex="-1"></a><span class="co"># # Add predicted probabilities and class labels</span></span>
<span id="cb76-43"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-43" aria-hidden="true" tabindex="-1"></a><span class="co"># data_selected$RF8.prob.CR &lt;- entire_probabilities[, &quot;CR&quot;]</span></span>
<span id="cb76-44"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-44" aria-hidden="true" tabindex="-1"></a><span class="co"># data_selected$Predicted_Response &lt;- entire_predictions</span></span>
<span id="cb76-45"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-45" aria-hidden="true" tabindex="-1"></a><span class="co"># write.csv(data_selected, &quot;../03.out/figure3/RF8_RJAML_cohort1_entire.csv&quot;)</span></span>
<span id="cb76-46"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-47"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-47" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot ROC curve for the entire dataset</span></span>
<span id="cb76-48"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-48" aria-hidden="true" tabindex="-1"></a>data_selected <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../03.out/figure3/RF8_RJAML_cohort1_entire.csv&quot;</span>, <span class="at">header =</span> T)</span>
<span id="cb76-49"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-49" aria-hidden="true" tabindex="-1"></a>roc_obj_entire <span class="ot">&lt;-</span> <span class="fu">roc</span>(data_selected<span class="sc">$</span>Response, data_selected<span class="sc">$</span>RF8.prob.CR)</span>
<span id="cb76-50"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb76-50" aria-hidden="true" tabindex="-1"></a>roc_obj_entire<span class="sc">$</span>auc</span></code></pre></div>
<pre><code>## Area under the curve: 0.976</code></pre>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save ROC plot to PDF</span></span>
<span id="cb78-2"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb78-2" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">plot</span>(roc_obj_entire, <span class="at">legacy.axes =</span> <span class="cn">TRUE</span>, <span class="at">lwd =</span> <span class="dv">2</span>,                          </span>
<span id="cb78-3"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb78-3" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">&quot;Entire set (N = 110)&quot;</span>,</span>
<span id="cb78-4"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb78-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">&quot;False Positive Rate (1 - Specificity)&quot;</span>, </span>
<span id="cb78-5"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb78-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">&quot;True Positive Rate (Sensitivity)&quot;</span>, </span>
<span id="cb78-6"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb78-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">cex.main =</span> <span class="fl">1.5</span>, <span class="at">cex.lab =</span> <span class="fl">1.2</span>, <span class="at">cex.axis =</span> <span class="fl">1.1</span>)</span>
<span id="cb78-7"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb78-7" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p3)</span></code></pre></div>
<pre><code>## 
## Call:
## roc.default(response = data_selected$Response, predictor = data_selected$RF8.prob.CR)
## 
## Data: data_selected$RF8.prob.CR in 77 controls (data_selected$Response CR) &gt; 33 cases (data_selected$Response NonCR).
## Area under the curve: 0.976</code></pre>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb80-1" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> <span class="fu">plot</span>(roc_obj_entire, <span class="at">print.thres =</span> <span class="st">&quot;best&quot;</span>, <span class="at">col =</span> <span class="st">&quot;#1f77b4&quot;</span>, <span class="at">print.thres.cex =</span> <span class="fl">1.2</span>, <span class="at">add =</span> <span class="cn">TRUE</span>)</span>
<span id="cb80-2"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb80-2" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="at">by =</span> <span class="fl">0.1</span>), <span class="at">v =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="at">by =</span> <span class="fl">0.1</span>), <span class="at">col =</span> <span class="st">&quot;gray80&quot;</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb80-3"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb80-3" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;bottomright&quot;</span>, <span class="at">legend =</span> <span class="fu">c</span>(<span class="fu">paste</span>(<span class="st">&quot;AUC =&quot;</span>, <span class="fu">round</span>(roc_obj_entire<span class="sc">$</span>auc, <span class="dv">2</span>))),</span>
<span id="cb80-4"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb80-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">col =</span> <span class="st">&quot;#1f77b4&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">cex =</span> <span class="fl">1.2</span>)</span></code></pre></div>
<p><img src="figures/unnamed-chunk-41-3.png" width="672" /></p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p4)</span></code></pre></div>
<pre><code>## 
## Call:
## roc.default(response = data_selected$Response, predictor = data_selected$RF8.prob.CR)
## 
## Data: data_selected$RF8.prob.CR in 77 controls (data_selected$Response CR) &gt; 33 cases (data_selected$Response NonCR).
## Area under the curve: 0.976</code></pre>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute Precision-Recall curve and AUPRC for the entire dataset</span></span>
<span id="cb83-2"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb83-2" aria-hidden="true" tabindex="-1"></a>data_selected<span class="sc">$</span>Response <span class="ot">&lt;-</span> <span class="fu">factor</span>(data_selected<span class="sc">$</span>Response, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;NonCR&quot;</span>, <span class="st">&quot;CR&quot;</span>))</span>
<span id="cb83-3"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb83-3" aria-hidden="true" tabindex="-1"></a>response_binary_entire <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(data_selected<span class="sc">$</span>Response) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb83-4"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb83-4" aria-hidden="true" tabindex="-1"></a>pr_curve_entire <span class="ot">&lt;-</span> <span class="fu">pr.curve</span>(<span class="at">scores.class0 =</span> data_selected<span class="sc">$</span>RF8.prob.CR, <span class="at">weights.class0 =</span> response_binary_entire, <span class="at">curve =</span> <span class="cn">TRUE</span>)</span>
<span id="cb83-5"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb83-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-6"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb83-6" aria-hidden="true" tabindex="-1"></a>p5 <span class="ot">&lt;-</span> <span class="fu">plot</span>(pr_curve_entire, <span class="at">main =</span> <span class="st">&quot;Entire set (N = 110)&quot;</span>, <span class="at">color =</span> <span class="st">&quot;#1f77b4&quot;</span>, <span class="at">auc.main =</span> <span class="cn">FALSE</span>)</span>
<span id="cb83-7"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb83-7" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="at">by =</span> <span class="fl">0.1</span>), <span class="at">v =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="at">by =</span> <span class="fl">0.1</span>), <span class="at">col =</span> <span class="st">&quot;gray80&quot;</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb83-8"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb83-8" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;bottomright&quot;</span>, <span class="at">legend =</span> <span class="fu">c</span>(<span class="fu">paste</span>(<span class="st">&quot;AUPRC =&quot;</span>, <span class="fu">round</span>(pr_curve_entire<span class="sc">$</span>auc.integral, <span class="dv">2</span>))),</span>
<span id="cb83-9"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb83-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">col =</span> <span class="st">&quot;#1f77b4&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">cex =</span> <span class="fl">1.2</span>)</span></code></pre></div>
<p><img src="figures/unnamed-chunk-41-4.png" width="672" /></p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p5)</span></code></pre></div>
<pre><code>## NULL</code></pre>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="benchmark-ml-methods-for-prediction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="auc-and-auprc-heatmap-for-the-entire-and-test-sets.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/03-results3.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
