<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6.6 Table one for multi-cohorts | Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning</title>
  <meta name="description" content="This is a book-style code sharing." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="6.6 Table one for multi-cohorts | Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a book-style code sharing." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6.6 Table one for multi-cohorts | Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning" />
  
  <meta name="twitter:description" content="This is a book-style code sharing." />
  

<meta name="author" content="Peng Jin, Qiqi Jin, Kankan Wang" />


<meta name="date" content="2025-07-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="barplot-in-rjaml-cohort.html"/>
<link rel="next" href="barplot-for-real-response.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Background</a></li>
<li class="chapter" data-level="2" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>2</b> Figure 1</a>
<ul>
<li class="chapter" data-level="2.1" data-path="process-the-data-from-functional_precision_medicine_tumor_board_aml.html"><a href="process-the-data-from-functional_precision_medicine_tumor_board_aml.html"><i class="fa fa-check"></i><b>2.1</b> Process the data from Functional_Precision_Medicine_Tumor_Board_AML</a></li>
<li class="chapter" data-level="2.2" data-path="functional_precision_medicine_tumor_board_aml.html"><a href="functional_precision_medicine_tumor_board_aml.html"><i class="fa fa-check"></i><b>2.2</b> Functional_Precision_Medicine_Tumor_Board_AML</a></li>
<li class="chapter" data-level="2.3" data-path="boxplot-for-drug-sensitivity-between-resistant-and-sensitive.html"><a href="boxplot-for-drug-sensitivity-between-resistant-and-sensitive.html"><i class="fa fa-check"></i><b>2.3</b> Boxplot for drug sensitivity between resistant and sensitive</a></li>
<li class="chapter" data-level="2.4" data-path="de-analysis-functional_medicine.html"><a href="de-analysis-functional_medicine.html"><i class="fa fa-check"></i><b>2.4</b> DE analysis Functional_medicine</a></li>
<li class="chapter" data-level="2.5" data-path="valcanoplot-plot-ex-vivo.html"><a href="valcanoplot-plot-ex-vivo.html"><i class="fa fa-check"></i><b>2.5</b> Valcanoplot plot (Ex vivo)</a></li>
<li class="chapter" data-level="2.6" data-path="heatmap-for-representative-de-genes.html"><a href="heatmap-for-representative-de-genes.html"><i class="fa fa-check"></i><b>2.6</b> Heatmap for representative DE genes</a></li>
<li class="chapter" data-level="2.7" data-path="cibersortx-results-visualization-ex-vivo.html"><a href="cibersortx-results-visualization-ex-vivo.html"><i class="fa fa-check"></i><b>2.7</b> CIBERSORTx Results Visualization (ex vivo)</a></li>
<li class="chapter" data-level="2.8" data-path="cibersortx-results-visualization-in-vivo.html"><a href="cibersortx-results-visualization-in-vivo.html"><i class="fa fa-check"></i><b>2.8</b> CIBERSORTx Results Visualization (in vivo)</a></li>
<li class="chapter" data-level="2.9" data-path="ssgsea-analysis-in-vivo.html"><a href="ssgsea-analysis-in-vivo.html"><i class="fa fa-check"></i><b>2.9</b> ssGSEA analysis in vivo</a></li>
<li class="chapter" data-level="2.10" data-path="go-enrichment-dot-plot.html"><a href="go-enrichment-dot-plot.html"><i class="fa fa-check"></i><b>2.10</b> GO Enrichment Dot Plot</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>3</b> Figure 2</a>
<ul>
<li class="chapter" data-level="3.1" data-path="de-analysis-between-in-vivo-sensitive-and-resistant.html"><a href="de-analysis-between-in-vivo-sensitive-and-resistant.html"><i class="fa fa-check"></i><b>3.1</b> DE analysis between in vivo sensitive and resistant</a></li>
<li class="chapter" data-level="3.2" data-path="de-analysis-rjaml-azaven-cohort.html"><a href="de-analysis-rjaml-azaven-cohort.html"><i class="fa fa-check"></i><b>3.2</b> DE analysis RJAML AZA+VEN cohort</a></li>
<li class="chapter" data-level="3.3" data-path="valcanoplot-plot-ex-vivo-1.html"><a href="valcanoplot-plot-ex-vivo-1.html"><i class="fa fa-check"></i><b>3.3</b> Valcanoplot plot (Ex vivo)</a></li>
<li class="chapter" data-level="3.4" data-path="valcanoplot-plot-in-vivo.html"><a href="valcanoplot-plot-in-vivo.html"><i class="fa fa-check"></i><b>3.4</b> Valcanoplot plot (In vivo)</a></li>
<li class="chapter" data-level="3.5" data-path="venn-plot.html"><a href="venn-plot.html"><i class="fa fa-check"></i><b>3.5</b> Venn plot</a></li>
<li class="chapter" data-level="3.6" data-path="integration-of-degs-ex-vivo-and-in-vivo.html"><a href="integration-of-degs-ex-vivo-and-in-vivo.html"><i class="fa fa-check"></i><b>3.6</b> Integration of DEGs ex vivo and in vivo</a></li>
<li class="chapter" data-level="3.7" data-path="gsea-analysis-for-crispri-drug-sensitivity.html"><a href="gsea-analysis-for-crispri-drug-sensitivity.html"><i class="fa fa-check"></i><b>3.7</b> GSEA analysis for CRISPRi drug sensitivity</a></li>
<li class="chapter" data-level="3.8" data-path="differential-expression-analysis-for-shpink1.html"><a href="differential-expression-analysis-for-shpink1.html"><i class="fa fa-check"></i><b>3.8</b> Differential Expression Analysis for shPINK1</a></li>
<li class="chapter" data-level="3.9" data-path="de-analysis-in-cell-lines.html"><a href="de-analysis-in-cell-lines.html"><i class="fa fa-check"></i><b>3.9</b> DE analysis in cell lines</a></li>
<li class="chapter" data-level="3.10" data-path="valcano-plots-molm13.html"><a href="valcano-plots-molm13.html"><i class="fa fa-check"></i><b>3.10</b> Valcano plots (MOLM13)</a></li>
<li class="chapter" data-level="3.11" data-path="valcano-plots-oa3.html"><a href="valcano-plots-oa3.html"><i class="fa fa-check"></i><b>3.11</b> Valcano plots (OA3)</a></li>
<li class="chapter" data-level="3.12" data-path="gsea-analysis-molm13.html"><a href="gsea-analysis-molm13.html"><i class="fa fa-check"></i><b>3.12</b> GSEA analysis (MOLM13)</a></li>
<li class="chapter" data-level="3.13" data-path="gsea-analysis-oa3.html"><a href="gsea-analysis-oa3.html"><i class="fa fa-check"></i><b>3.13</b> GSEA analysis (OA3)</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>4</b> Figure 3</a>
<ul>
<li class="chapter" data-level="4.1" data-path="benchmark-ml-methods-for-prediction.html"><a href="benchmark-ml-methods-for-prediction.html"><i class="fa fa-check"></i><b>4.1</b> Benchmark ML Methods for Prediction</a></li>
<li class="chapter" data-level="4.2" data-path="predict-class-labels-on-the-test-set-and-plot-roc.html"><a href="predict-class-labels-on-the-test-set-and-plot-roc.html"><i class="fa fa-check"></i><b>4.2</b> Predict Class Labels on the Test Set and Plot ROC</a></li>
<li class="chapter" data-level="4.3" data-path="auc-and-auprc-heatmap-for-the-entire-and-test-sets.html"><a href="auc-and-auprc-heatmap-for-the-entire-and-test-sets.html"><i class="fa fa-check"></i><b>4.3</b> AUC and AUPRC Heatmap for the Entire and Test Sets</a></li>
<li class="chapter" data-level="4.4" data-path="model-performance-plot.html"><a href="model-performance-plot.html"><i class="fa fa-check"></i><b>4.4</b> Model Performance Plot</a></li>
<li class="chapter" data-level="4.5" data-path="feature-importance.html"><a href="feature-importance.html"><i class="fa fa-check"></i><b>4.5</b> Feature importance</a></li>
<li class="chapter" data-level="4.6" data-path="heatmap-for-confusion-matrices-defaut-cutoff-0.html"><a href="heatmap-for-confusion-matrices-defaut-cutoff-0.html"><i class="fa fa-check"></i><b>4.6</b> Heatmap for Confusion Matrices (defaut cutoff 0.5)</a></li>
<li class="chapter" data-level="4.7" data-path="barplot-for-predicted-response-0.html"><a href="barplot-for-predicted-response-0.html"><i class="fa fa-check"></i><b>4.7</b> Barplot for predicted Response (0.5)</a></li>
<li class="chapter" data-level="4.8" data-path="heatmap-for-confusion-matrices-optimal-cutoff-0.html"><a href="heatmap-for-confusion-matrices-optimal-cutoff-0.html"><i class="fa fa-check"></i><b>4.8</b> Heatmap for Confusion Matrices (optimal cutoff 0.630)</a></li>
<li class="chapter" data-level="4.9" data-path="barplot-for-predicted-response-0.html"><a href="barplot-for-predicted-response-0.html#barplot-for-predicted-response-0.630"><i class="fa fa-check"></i><b>4.9</b> Barplot for predicted Response (0.630)</a></li>
<li class="chapter" data-level="4.10" data-path="density-plot-for-response-probability.html"><a href="density-plot-for-response-probability.html"><i class="fa fa-check"></i><b>4.10</b> Density Plot for Response Probability</a></li>
<li class="chapter" data-level="4.11" data-path="violin-and-boxplots-for-response-probability-rf8.html"><a href="violin-and-boxplots-for-response-probability-rf8.html"><i class="fa fa-check"></i><b>4.11</b> Violin and Boxplots for Response Probability (RF8)</a></li>
<li class="chapter" data-level="4.12" data-path="barplot-for-response.html"><a href="barplot-for-response.html"><i class="fa fa-check"></i><b>4.12</b> Barplot for Response</a></li>
<li class="chapter" data-level="4.13" data-path="rf8-for-mdr-status.html"><a href="rf8-for-mdr-status.html"><i class="fa fa-check"></i><b>4.13</b> RF8 for MDR status</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>5</b> Figure 4</a>
<ul>
<li class="chapter" data-level="5.1" data-path="validation-of-rf8-model-rjaml-cohort-2.html"><a href="validation-of-rf8-model-rjaml-cohort-2.html"><i class="fa fa-check"></i><b>5.1</b> Validation of RF8 model (RJAML-Cohort 2)</a></li>
<li class="chapter" data-level="5.2" data-path="validation-of-rf8-model-fpmtb.html"><a href="validation-of-rf8-model-fpmtb.html"><i class="fa fa-check"></i><b>5.2</b> Validation of RF8 model (FPMTB)</a></li>
<li class="chapter" data-level="5.3" data-path="validation-of-rf8-model-beataml.html"><a href="validation-of-rf8-model-beataml.html"><i class="fa fa-check"></i><b>5.3</b> Validation of RF8 model (BeatAML)</a></li>
<li class="chapter" data-level="5.4" data-path="violin-and-boxplots-for-response-probability-rf8-1.html"><a href="violin-and-boxplots-for-response-probability-rf8-1.html"><i class="fa fa-check"></i><b>5.4</b> Violin and Boxplots for Response Probability (RF8)</a></li>
<li class="chapter" data-level="5.5" data-path="rf8-score-among-clusters.html"><a href="rf8-score-among-clusters.html"><i class="fa fa-check"></i><b>5.5</b> RF8 score among clusters</a></li>
<li class="chapter" data-level="5.6" data-path="integration-of-ex-vivo-drug-response-in-beataml.html"><a href="integration-of-ex-vivo-drug-response-in-beataml.html"><i class="fa fa-check"></i><b>5.6</b> Integration of ex vivo drug-response in BeatAML</a></li>
<li class="chapter" data-level="5.7" data-path="beataml-waves-1-4-heatmap-based-on-drug-profiling.html"><a href="beataml-waves-1-4-heatmap-based-on-drug-profiling.html"><i class="fa fa-check"></i><b>5.7</b> BeatAML Waves 1-4 heatmap based on drug profiling</a></li>
<li class="chapter" data-level="5.8" data-path="boxplot-for-drug-sensitivitity-between-resistant-and-sensitive.html"><a href="boxplot-for-drug-sensitivitity-between-resistant-and-sensitive.html"><i class="fa fa-check"></i><b>5.8</b> Boxplot for drug sensitivitity between resistant and sensitive</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>6</b> Figure 5</a>
<ul>
<li class="chapter" data-level="6.1" data-path="integration-with-aml-genomics.html"><a href="integration-with-aml-genomics.html"><i class="fa fa-check"></i><b>6.1</b> Integration with AML genomics</a></li>
<li class="chapter" data-level="6.2" data-path="mutation-pairs.html"><a href="mutation-pairs.html"><i class="fa fa-check"></i><b>6.2</b> Mutation pairs</a></li>
<li class="chapter" data-level="6.3" data-path="complexheatmap-for-mutation.html"><a href="complexheatmap-for-mutation.html"><i class="fa fa-check"></i><b>6.3</b> Complexheatmap for mutation</a></li>
<li class="chapter" data-level="6.4" data-path="fisher-test.html"><a href="fisher-test.html"><i class="fa fa-check"></i><b>6.4</b> Fisher test</a></li>
<li class="chapter" data-level="6.5" data-path="barplot-in-rjaml-cohort.html"><a href="barplot-in-rjaml-cohort.html"><i class="fa fa-check"></i><b>6.5</b> Barplot in RJAML cohort</a></li>
<li class="chapter" data-level="6.6" data-path="table-one-for-multi-cohorts.html"><a href="table-one-for-multi-cohorts.html"><i class="fa fa-check"></i><b>6.6</b> Table one for multi-cohorts</a></li>
<li class="chapter" data-level="6.7" data-path="barplot-for-real-response.html"><a href="barplot-for-real-response.html"><i class="fa fa-check"></i><b>6.7</b> Barplot for real response</a></li>
<li class="chapter" data-level="6.8" data-path="multivariable-logisticreg.html"><a href="multivariable-logisticreg.html"><i class="fa fa-check"></i><b>6.8</b> Multivariable LogisticReg</a></li>
<li class="chapter" data-level="6.9" data-path="identification-of-rf8-correlated-genes.html"><a href="identification-of-rf8-correlated-genes.html"><i class="fa fa-check"></i><b>6.9</b> Identification of RF8-correlated genes</a></li>
<li class="chapter" data-level="6.10" data-path="differential-expression-analysis-for-mut-vs.-wt.html"><a href="differential-expression-analysis-for-mut-vs.-wt.html"><i class="fa fa-check"></i><b>6.10</b> Differential Expression Analysis for Mut vs. WT</a></li>
<li class="chapter" data-level="6.11" data-path="venn-diagram-for-target-genes.html"><a href="venn-diagram-for-target-genes.html"><i class="fa fa-check"></i><b>6.11</b> Venn Diagram for target genes</a></li>
<li class="chapter" data-level="6.12" data-path="pathway-overlap-analysis-of-genes.html"><a href="pathway-overlap-analysis-of-genes.html"><i class="fa fa-check"></i><b>6.12</b> Pathway overlap analysis of genes</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="figure-6.html"><a href="figure-6.html"><i class="fa fa-check"></i><b>7</b> Figure 6</a>
<ul>
<li class="chapter" data-level="7.1" data-path="relationship-between-rf8-score-vs.-venaza-response-probability.html"><a href="relationship-between-rf8-score-vs.-venaza-response-probability.html"><i class="fa fa-check"></i><b>7.1</b> Relationship between RF8 score vs. VEN/AZA response probability</a></li>
<li class="chapter" data-level="7.2" data-path="monotonic-response-prediction-plot.html"><a href="monotonic-response-prediction-plot.html"><i class="fa fa-check"></i><b>7.2</b> Monotonic Response Prediction Plot</a></li>
<li class="chapter" data-level="7.3" data-path="survival-plot.html"><a href="survival-plot.html"><i class="fa fa-check"></i><b>7.3</b> Survival plot</a></li>
<li class="chapter" data-level="7.4" data-path="brier-score-analysis.html"><a href="brier-score-analysis.html"><i class="fa fa-check"></i><b>7.4</b> Brier Score Analysis</a></li>
<li class="chapter" data-level="7.5" data-path="c-index-comparison.html"><a href="c-index-comparison.html"><i class="fa fa-check"></i><b>7.5</b> C-index Comparison</a></li>
<li class="chapter" data-level="7.6" data-path="km-plots-for-rjaml-cohort-2.html"><a href="km-plots-for-rjaml-cohort-2.html"><i class="fa fa-check"></i><b>7.6</b> KM-Plots for RJAML Cohort 2</a></li>
<li class="chapter" data-level="7.7" data-path="barplot-for-crcri.html"><a href="barplot-for-crcri.html"><i class="fa fa-check"></i><b>7.7</b> Barplot for CR/CRi</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Precision Prediction of Venetoclax-Azacitidine Treatment Efficacy in Acute Myeloid Leukemia via Integrative Drug Screening and Machine Learning</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="table-one-for-multi-cohorts" class="section level2 hasAnchor" number="6.6">
<h2><span class="header-section-number">6.6</span> Table one for multi-cohorts<a href="table-one-for-multi-cohorts.html#table-one-for-multi-cohorts" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="table-one-for-multi-cohorts.html#cb161-1" aria-hidden="true" tabindex="-1"></a><span class="do">####################################</span></span>
<span id="cb161-2"><a href="table-one-for-multi-cohorts.html#cb161-2" aria-hidden="true" tabindex="-1"></a><span class="do">### Table one for multi-cohorts</span></span>
<span id="cb161-3"><a href="table-one-for-multi-cohorts.html#cb161-3" aria-hidden="true" tabindex="-1"></a><span class="do">####################################</span></span>
<span id="cb161-4"><a href="table-one-for-multi-cohorts.html#cb161-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tableone)</span>
<span id="cb161-5"><a href="table-one-for-multi-cohorts.html#cb161-5" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;../03.out/figure5/multi-cohorts_4gene_Clinicaldata.csv&quot;</span>,</span>
<span id="cb161-6"><a href="table-one-for-multi-cohorts.html#cb161-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">header =</span> T, <span class="at">check.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb161-7"><a href="table-one-for-multi-cohorts.html#cb161-7" aria-hidden="true" tabindex="-1"></a>myVars <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Age&quot;</span>, <span class="st">&quot;Gender&quot;</span>, <span class="st">&quot;Secondary_AML&quot;</span>, <span class="st">&quot;Complex_Karyotype&quot;</span>, </span>
<span id="cb161-8"><a href="table-one-for-multi-cohorts.html#cb161-8" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;FLT3-ITD&quot;</span>, <span class="st">&quot;NPM1&quot;</span>, <span class="st">&quot;KRAS&quot;</span>, <span class="st">&quot;NRAS&quot;</span>, <span class="st">&quot;CEBPA&quot;</span>, <span class="st">&quot;DNMT3A&quot;</span>,</span>
<span id="cb161-9"><a href="table-one-for-multi-cohorts.html#cb161-9" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;IDH2&quot;</span>, <span class="st">&quot;TP53&quot;</span>)</span>
<span id="cb161-10"><a href="table-one-for-multi-cohorts.html#cb161-10" aria-hidden="true" tabindex="-1"></a>catVars <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Gender&quot;</span>, <span class="st">&quot;Secondary_AML&quot;</span>, <span class="st">&quot;Complex_Karyotype&quot;</span>,</span>
<span id="cb161-11"><a href="table-one-for-multi-cohorts.html#cb161-11" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;FLT3-ITD&quot;</span>, <span class="st">&quot;NPM1&quot;</span>, <span class="st">&quot;KRAS&quot;</span>, <span class="st">&quot;NRAS&quot;</span>, <span class="st">&quot;CEBPA&quot;</span>, <span class="st">&quot;DNMT3A&quot;</span>,</span>
<span id="cb161-12"><a href="table-one-for-multi-cohorts.html#cb161-12" aria-hidden="true" tabindex="-1"></a>             <span class="st">&quot;IDH2&quot;</span>, <span class="st">&quot;TP53&quot;</span>)</span>
<span id="cb161-13"><a href="table-one-for-multi-cohorts.html#cb161-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb161-14"><a href="table-one-for-multi-cohorts.html#cb161-14" aria-hidden="true" tabindex="-1"></a>tab <span class="ot">&lt;-</span> <span class="fu">CreateTableOne</span>(<span class="at">vars =</span> myVars, <span class="at">strata =</span> <span class="fu">c</span>(<span class="st">&quot;Response_group&quot;</span>, <span class="st">&quot;Cohort&quot;</span>),</span>
<span id="cb161-15"><a href="table-one-for-multi-cohorts.html#cb161-15" aria-hidden="true" tabindex="-1"></a>                      <span class="at">addOverall =</span> <span class="cn">TRUE</span>, <span class="at">data =</span> data, <span class="at">factorVars =</span> catVars)</span>
<span id="cb161-16"><a href="table-one-for-multi-cohorts.html#cb161-16" aria-hidden="true" tabindex="-1"></a>tab2 <span class="ot">&lt;-</span> <span class="fu">print</span>(tab, <span class="at">showAllLevels =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>##                        Stratified by Response_group:Cohort
##                         level  Overall       CR/CRi:Daniel A. Pollyea_2018_NM
##   n                              427            29                           
##   Age (mean (SD))              68.22 (9.76)  74.41 (5.60)                    
##   Gender (%)            Female   199 (46.6)     18 ( 62.1)                   
##                         Male     228 (53.4)     11 ( 37.9)                   
##   Secondary_AML (%)     No       295 (69.1)     19 ( 65.5)                   
##                         Yes      132 (30.9)     10 ( 34.5)                   
##   Complex_Karyotype (%)           33 ( 7.7)     29 (100.0)                   
##                         No       308 (72.1)      0 (  0.0)                   
##                         Yes       86 (20.1)      0 (  0.0)                   
##   FLT3-ITD (%)          Mut       57 (13.3)      4 ( 13.8)                   
##                         WT       370 (86.7)     25 ( 86.2)                   
##   NPM1 (%)              Mut       92 (21.5)      8 ( 27.6)                   
##                         WT       335 (78.5)     21 ( 72.4)                   
##   KRAS (%)                        33 ( 7.7)     29 (100.0)                   
##                         Mut       19 ( 4.4)      0 (  0.0)                   
##                         WT       375 (87.8)      0 (  0.0)                   
##   NRAS (%)              Mut       40 ( 9.4)      1 (  3.4)                   
##                         WT       387 (90.6)     28 ( 96.6)                   
##   CEBPA (%)             Mut       32 ( 7.5)      1 (  3.4)                   
##                         WT       395 (92.5)     28 ( 96.6)                   
##   DNMT3A (%)            Mut       95 (22.2)      5 ( 17.2)                   
##                         WT       332 (77.8)     24 ( 82.8)                   
##   IDH2 (%)              Mut       70 (16.4)      5 ( 17.2)                   
##                         WT       357 (83.6)     24 ( 82.8)                   
##   TP53 (%)              Mut       57 (13.3)      2 (  6.9)                   
##                         WT       370 (86.7)     27 ( 93.1)                   
##                        Stratified by Response_group:Cohort
##                         Non-response:Daniel A. Pollyea_2018_NM CR/CRi:DKFZ_Cohort1
##   n                         4                                     40              
##   Age (mean (SD))       74.50 (7.05)                           71.12 (8.35)       
##   Gender (%)                1 ( 25.0)                             14 (35.0)       
##                             3 ( 75.0)                             26 (65.0)       
##   Secondary_AML (%)         4 (100.0)                             31 (77.5)       
##                             0 (  0.0)                              9 (22.5)       
##   Complex_Karyotype (%)     4 (100.0)                              0 ( 0.0)       
##                             0 (  0.0)                             32 (80.0)       
##                             0 (  0.0)                              8 (20.0)       
##   FLT3-ITD (%)              1 ( 25.0)                              5 (12.5)       
##                             3 ( 75.0)                             35 (87.5)       
##   NPM1 (%)                  0 (  0.0)                             10 (25.0)       
##                             4 (100.0)                             30 (75.0)       
##   KRAS (%)                  4 (100.0)                              0 ( 0.0)       
##                             0 (  0.0)                              3 ( 7.5)       
##                             0 (  0.0)                             37 (92.5)       
##   NRAS (%)                  0 (  0.0)                              4 (10.0)       
##                             4 (100.0)                             36 (90.0)       
##   CEBPA (%)                 0 (  0.0)                              2 ( 5.0)       
##                             4 (100.0)                             38 (95.0)       
##   DNMT3A (%)                0 (  0.0)                              8 (20.0)       
##                             4 (100.0)                             32 (80.0)       
##   IDH2 (%)                  0 (  0.0)                              9 (22.5)       
##                             4 (100.0)                             31 (77.5)       
##   TP53 (%)                  1 ( 25.0)                              3 ( 7.5)       
##                             3 ( 75.0)                             37 (92.5)       
##                        Stratified by Response_group:Cohort
##                         Non-response:DKFZ_Cohort1 CR/CRi:ECLA   Non-response:ECLA
##   n                        14                        42            23            
##   Age (mean (SD))       67.71 (7.31)              64.48 (11.08) 69.96 (10.17)    
##   Gender (%)                3 ( 21.4)                19 (45.2)      8 ( 34.8)    
##                            11 ( 78.6)                23 (54.8)     15 ( 65.2)    
##   Secondary_AML (%)         6 ( 42.9)                39 (92.9)     21 ( 91.3)    
##                             8 ( 57.1)                 3 ( 7.1)      2 (  8.7)    
##   Complex_Karyotype (%)     0 (  0.0)                 0 ( 0.0)      0 (  0.0)    
##                             6 ( 42.9)                36 (85.7)     16 ( 69.6)    
##                             8 ( 57.1)                 6 (14.3)      7 ( 30.4)    
##   FLT3-ITD (%)              1 (  7.1)                 3 ( 7.1)      2 (  8.7)    
##                            13 ( 92.9)                39 (92.9)     21 ( 91.3)    
##   NPM1 (%)                  1 (  7.1)                 6 (14.3)      0 (  0.0)    
##                            13 ( 92.9)                36 (85.7)     23 (100.0)    
##   KRAS (%)                  0 (  0.0)                 0 ( 0.0)      0 (  0.0)    
##                             1 (  7.1)                 3 ( 7.1)      2 (  8.7)    
##                            13 ( 92.9)                39 (92.9)     21 ( 91.3)    
##   NRAS (%)                  1 (  7.1)                 5 (11.9)      2 (  8.7)    
##                            13 ( 92.9)                37 (88.1)     21 ( 91.3)    
##   CEBPA (%)                 0 (  0.0)                 5 (11.9)      4 ( 17.4)    
##                            14 (100.0)                37 (88.1)     19 ( 82.6)    
##   DNMT3A (%)                2 ( 14.3)                12 (28.6)      4 ( 17.4)    
##                            12 ( 85.7)                30 (71.4)     19 ( 82.6)    
##   IDH2 (%)                  1 (  7.1)                 7 (16.7)      2 (  8.7)    
##                            13 ( 92.9)                35 (83.3)     21 ( 91.3)    
##   TP53 (%)                  4 ( 28.6)                 8 (19.0)      6 ( 26.1)    
##                            10 ( 71.4)                34 (81.0)     17 ( 73.9)    
##                        Stratified by Response_group:Cohort
##                         CR/CRi:PSS_CD  Non-response:PSS_CD CR/CRi:RJAML 
##   n                        85             15                 119        
##   Age (mean (SD))       69.75 (13.14)  66.73 (10.90)       66.66 (7.00) 
##   Gender (%)               47 ( 55.3)      4 ( 26.7)          61 (51.3) 
##                            38 ( 44.7)     11 ( 73.3)          58 (48.7) 
##   Secondary_AML (%)        67 ( 78.8)     13 ( 86.7)          67 (56.3) 
##                            18 ( 21.2)      2 ( 13.3)          52 (43.7) 
##   Complex_Karyotype (%)     0 (  0.0)      0 (  0.0)           0 ( 0.0) 
##                            62 ( 72.9)      9 ( 60.0)         102 (85.7) 
##                            23 ( 27.1)      6 ( 40.0)          17 (14.3) 
##   FLT3-ITD (%)             15 ( 17.6)      2 ( 13.3)          19 (16.0) 
##                            70 ( 82.4)     13 ( 86.7)         100 (84.0) 
##   NPM1 (%)                 27 ( 31.8)      2 ( 13.3)          32 (26.9) 
##                            58 ( 68.2)     13 ( 86.7)          87 (73.1) 
##   KRAS (%)                  0 (  0.0)      0 (  0.0)           0 ( 0.0) 
##                             0 (  0.0)      3 ( 20.0)           5 ( 4.2) 
##                            85 (100.0)     12 ( 80.0)         114 (95.8) 
##   NRAS (%)                  5 (  5.9)      3 ( 20.0)          13 (10.9) 
##                            80 ( 94.1)     12 ( 80.0)         106 (89.1) 
##   CEBPA (%)                 5 (  5.9)      0 (  0.0)          12 (10.1) 
##                            80 ( 94.1)     15 (100.0)         107 (89.9) 
##   DNMT3A (%)               22 ( 25.9)      3 ( 20.0)          32 (26.9) 
##                            63 ( 74.1)     12 ( 80.0)          87 (73.1) 
##   IDH2 (%)                 15 ( 17.6)      0 (  0.0)          23 (19.3) 
##                            70 ( 82.4)     15 (100.0)          96 (80.7) 
##   TP53 (%)                  8 (  9.4)      2 ( 13.3)          11 ( 9.2) 
##                            77 ( 90.6)     13 ( 86.7)         108 (90.8) 
##                        Stratified by Response_group:Cohort
##                         Non-response:RJAML p      test
##   n                        56                         
##   Age (mean (SD))       66.07 (8.38)       &lt;0.001     
##   Gender (%)               24 (42.9)        0.050     
##                            32 (57.1)                  
##   Secondary_AML (%)        28 (50.0)       &lt;0.001     
##                            28 (50.0)                  
##   Complex_Karyotype (%)     0 ( 0.0)       &lt;0.001     
##                            45 (80.4)                  
##                            11 (19.6)                  
##   FLT3-ITD (%)              5 ( 8.9)        0.759     
##                            51 (91.1)                  
##   NPM1 (%)                  6 (10.7)        0.006     
##                            50 (89.3)                  
##   KRAS (%)                  0 ( 0.0)       &lt;0.001     
##                             2 ( 3.6)                  
##                            54 (96.4)                  
##   NRAS (%)                  6 (10.7)        0.768     
##                            50 (89.3)                  
##   CEBPA (%)                 3 ( 5.4)        0.352     
##                            53 (94.6)                  
##   DNMT3A (%)                7 (12.5)        0.462     
##                            49 (87.5)                  
##   IDH2 (%)                  8 (14.3)        0.562     
##                            48 (85.7)                  
##   TP53 (%)                 12 (21.4)        0.068     
##                            44 (78.6)</code></pre>
<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="table-one-for-multi-cohorts.html#cb163-1" aria-hidden="true" tabindex="-1"></a><span class="co"># write.csv(tab2, file = &quot;../03.out/figure5/multi-cohorts_4gene_Clinicaldata_out.csv&quot;)</span></span></code></pre></div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="barplot-in-rjaml-cohort.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="barplot-for-real-response.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/05-results5.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
